---
title: "day1_gsea_example"
output:
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
---

Prepare the data. Here we read three types of data:

1. the phenotype/condition data
2. the gene expression matrix
3. the gene set

The gene expression data is saved in `.gct` format and experimental condition/phenotype
is saves in `.cls` format. Both formats are very simple, and you can try to write your own
code to parse them.

Gene sets are saved in "`.gmt` format. It is also a simple format. You can try to parse
it by your own.

```{r}
library(CePa)
condition = read.cls("P53.cls", treatment = "MUT", control = "WT")$label
expr = read.gct("P53_collapsed_symbols.gct")

ln = strsplit(readLines("c2.symbols.gmt"), "\t")
gs = lapply(ln, function(x) x[-(1:2)])
names(gs) = sapply(ln, function(x) x[1])

geneset = gs[["p53hypoxiaPathway"]]
```

- `expr`
- `condition`
- `geneset`

This gene set is very small:

```{r}
length(geneset)
```

Note here gene IDs in the expression matrix and in the gene sets are all gene
symbols, thus no more adjustment needs to be done here.


The gene-level difference score is set as signal-to-noise ratios, which is:

- mean in group 1
- mean in group 2
- sd in group 1
- sd in group 2

$$ \frac{\mu_1 - \mu_2}{\sigma_1 + \sigma_2} $$

We calculate the gene-level difference score in `s`:

```{r}
s = apply(expr, 1, function(x) {
	x1 = x[condition == "WT"]
	x2 = x[condition == "MUT"]
	(mean(x1) - mean(x2))/(sd(x1) + sd(x2))
})
```

- s: gene-level difference score

Sort the gene scores from the highest to the lowest (to make it into a ranked list):

```{r}
s = sort(s, decreasing = TRUE)
```

Next we first implement the original GSEA method, which was proposed in Mootha et al., 2003.

```{r}
## original GSEA
l_set = names(s) %in% geneset
f1 = cumsum(l_set)/sum(l_set)

l_other = !names(s) %in% geneset
f2 = cumsum(l_other)/sum(l_other)
```

Here `f1` is the cumulative probability function of genes in the set and `f2` is the cumulative probability function of genes not in the set.

We first plot the CDF of two distributions.

```{r}
n = length(s)
plot(1:n, f1, type = "l", col = "red")
lines(1:n, f2, col = "blue")
```

The reason why the blue locates almost on the diagonal is the gene set is very small.

Next the difference of cumulative probability (`f1 - f2`) at each position of the ranked gene list.
Let's call it "the GSEA plot".

```{r}
plot(f1 - f2, type = "l")
abline(h = 0, lty = 2, col = "grey")
points(which(l_set), rep(0, sum(l_set)), pch = "|", col = "red")
```

The enrichment score (ES) defined as `max(f1 - f2)` is:

```{r}
es = max(f1 - f2)
es
```

And the position in the "GSEA plot":

```{r}
plot(f1 - f2, type = "l")
abline(h = 0, lty = 2, col = "grey")
points(which(l_set), rep(0, sum(l_set)), pch = "|", col = "red")
abline(v = which.max(f1 - f2), lty = 3, col = "blue")
```

The statistic `es` actually is the Kolmogorov-Smirnov statistics, thus, we can
directly apply the KS test:

```{r}
ks.test(which(l_set), which(l_other))
```

However, we can see the p-value is not significant, this is because KS test is not
a powerful test. Next we construct the null distribution by sample permutation.

In the next code chunk, the calculation of ES score is wrapped into a function,
also we use `rowMeans()` and `rowSds()` to speed up the calculation of gene-level
scores.

```{r}
library(matrixStats)
# expr: the complete expression matrix
# condition: the condition labels of samples
# cmp: a vector of two, cmp[1] - cmp[2] > 0 means up-regulation
# geneset: A vector of genes
calculate_es = function(expr, condition, cmp, geneset) {

	m1 = expr[, condition == cmp[1]]  # only samples in group 1
	m2 = expr[, condition == cmp[2]]  # only samples in group 2

	s = (rowMeans(m1) - rowMeans(m2))/(rowSds(m1) + rowSds(m2)) # a gene-level difference socre (S2N ratio) 

	s = sort(s, decreasing = TRUE)  # ranked gene list

	l_set = names(s) %in% geneset
	f1 = cumsum(l_set)/sum(l_set)   # CDF for genes in the set

	l_other = !l_set
	f2 = cumsum(l_other)/sum(l_other)  # CDF for genes not in the set

	max(f1 - f2)
}
```

The ES score calculated by `calculate_es()`:

```{r}
es = calculate_es(expr, condition, cmp = c("WT", "MUT"), geneset = geneset)
es
```

We randomly permute sample labels or we randomly permute `condition`. We do it
1000 times. The ES scores in null distributions are saved in `es_rand`.

```{r}
set.seed(123)
es_rand = numeric(1000)
for(i in 1:1000) {
	es_rand[i] = calculate_es(expr, sample(condition), 
	    cmp = c("WT", "MUT"), geneset = geneset)
}
```

p-value is calculated as the proportion of `es` being equal to or larger than in values in `es_rand`.

```{r}
sum(es_rand >= es)/1000
```

0.129

The null distribution of ES:

```{r}
hist(es_rand)
abline(v = es, col = "red")
```



Next we implement the improved GSEA (Subramanian et al., PNAS, 2005) where gene-level scores are taken as the weight.

We directly modify `calculate_es()` to `calculate_es_v2()` where there is only two lines new, which we highlight in the code chunk:

```{r}
calculate_es_v2 = function(expr, condition, cmp, geneset, plot = FALSE, power = 1) {

    m1 = expr[, condition == cmp[1]]
    m2 = expr[, condition == cmp[2]]

    s = (rowMeans(m1) - rowMeans(m2))/(rowSds(m1) + rowSds(m2))

    s = sort(s, decreasing = TRUE)

    l_set = names(s) %in% geneset
    # f1 = cumsum(l_set)/sum(l_set)  # <<-- the original line
    s_set = abs(s)^power   # <<-- here
    s_set[!l_set] = 0
    f1 = cumsum(s_set)/sum(s_set)  ## <<- here

    l_other = !l_set
    f2 = cumsum(l_other)/sum(l_other)

    if(plot) {
        plot(f1 - f2, type = "l")
        abline(h = 0, lty = 2, col = "grey")
        points(which(l_set), rep(0, sum(l_set)), pch = "|", col = "red")
        abline(v = which.max(f1 - f2), lty = 3, col = "blue")
    }

    max(f1 - f2)
}
```

Now we calculate the new ES score and make the GSEA plot:

```{r}
es = calculate_es_v2(expr, condition, cmp = c("WT", "MUT"), plot = TRUE, 
    geneset = geneset)
```

We can also check when `power = 0` and `power = 2`:

```{r, results = "none", fig.width = 10}
par(mfrow = c(1, 2))
calculate_es_v2(expr, condition, cmp = c("WT", "MUT"), plot = TRUE, power = 0, 
    geneset = geneset)  # same as the original GSEA
title("power = 0")
calculate_es_v2(expr, condition, cmp = c("WT", "MUT"), plot = TRUE, power = 2, 
    geneset = geneset)
title("power = 2")
par(mfrow = c(1, 1))
```

Similarly, we randomly permute samples to obtain the null distribution of ES:

```{r}
es_rand = numeric(1000)
for(i in 1:1000) {
	es_rand[i] = calculate_es_v2(expr, sample(condition), 
	    cmp = c("WT", "MUT"), geneset = geneset)
}
```

The new p-value:

```{r}
sum(es_rand >= es)/1000
```

1/1000, < 0.001

And the null distribution of ES:

```{r}
hist(es_rand, xlim = c(0, 1))
abline(v = es, col = "red")
```

We can see the improved GSEA is more powerful than the original GSEA, because the 
original GSEA equally weights genes and the improved GSEA weights genes based on their differential expression,
which increases the effect of diff genes. Let's plot the weight of genes:

```{r}
plot(abs(s))
```


---------------------------------------------------

Null distribution can also be constructed by gene permutation. It is very
easy to implement:

```{r}
# s: a vector of pre-calcualted gene-level scores
# s should be sorted
calculate_es_v2_gene_perm = function(s, perm = FALSE, power = 1) {
	
	if(perm) {
	    # s is still sorted, but the gene labels are randomly shuffled
		names(s) = sample(names(s))  ## <<- here
	}

	l_set = names(s) %in% geneset
	s_set = abs(s)^power
	s_set[!l_set] = 0
	f1 = cumsum(s_set)/sum(s_set)

	l_other = !l_set
	f2 = cumsum(l_other)/sum(l_other)

	max(f1 - f2)
}
```

Good thing of gene permutation is the gene-level scores only need to be calculated
once and can be repeatedly used.


```{r}
# pre-calculate gene-level scores
m1 = expr[, condition == "WT"]
m2 = expr[, condition == "MUT"]

s = (rowMeans(m1) - rowMeans(m2))/(rowSds(m1) + rowSds(m2))
s = sort(s, decreasing = TRUE)  # must be pre-sorted
```

We calculate the null distribution of ES from gene permutation:

```{r}
es = calculate_es_v2_gene_perm(s)
es_rand = numeric(1000)
for(i in 1:1000) {
	es_rand[i] = calculate_es_v2_gene_perm(s, perm = TRUE)
}

sum(es_rand >= es)/1000
```

also < 0.001

The null distribution of ES from gene permutation:

```{r}
hist(es_rand, xlim = c(0, 1))
abline(v = es, col = "red")
```
