<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 2-05: Use clusterProfiler for GSEA • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 2-05: Use clusterProfiler for GSEA">
<meta property="og:description" content="GSEAtraining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 2-05: Use clusterProfiler for GSEA</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2024-01-25</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic2_05_clusterProfiler_gsea.Rmd" class="external-link"><code>vignettes/topic2_05_clusterProfiler_gsea.Rmd</code></a></small>
      <div class="hidden name"><code>topic2_05_clusterProfiler_gsea.Rmd</code></div>

    </div>

    
    
<p><strong>Note in clusterProfiler, p-values are calculated by gene permutations!</strong></p>
<p>We use a dataset <code>gene_diff_score</code> which is a vector of a certain metric for differential expression of genes.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_diff_score</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"gene_diff_score.rds"</span>, package <span class="op">=</span> <span class="st">"GSEAtraining"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_diff_score</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    CDKN1A       BAX      MDM2      DDB2     STAT6     NTSR2 </span></span>
<span><span class="co">## 0.8425507 0.8232428 0.6306318 0.5945239 0.5791271 0.5712063</span></span></code></pre>
<p>We need to convert genes from symbols to EntreZ IDs.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span></span>
<span><span class="co">##     as.data.frame, basename, cbind, colnames, dirname, do.call,</span></span>
<span><span class="co">##     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span></span>
<span><span class="co">##     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span></span>
<span><span class="co">##     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span></span>
<span><span class="co">##     tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'S4Vectors' was built under R version 4.3.2</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     I, expand.grid, unname</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene_diff_score</span><span class="op">)</span>, keytype <span class="op">=</span> <span class="st">"SYMBOL"</span>, column <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">map</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  CDKN1A     BAX    MDM2    DDB2   STAT6   NTSR2 </span></span>
<span><span class="co">##  "1026"   "581"  "4193"  "1643"  "6778" "23620"</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># names(gene_diff_score) = map[ names(gene_diff_score) ]</span></span></code></pre></div>
<p>For this dataset, the mapping is 1:1. But you need to be careful when the mapping is not 1:1.</p>
<p>Here we use the <code><a href="../reference/convert_to_entrez_id.html">convert_to_entrez_id()</a></code> function from the <strong>GSEAtraining</strong> package, which automatically takes care of multiple mappings:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/GSEAtraining" class="external-link">GSEAtraining</a></span><span class="op">)</span></span>
<span><span class="va">gene_diff_score</span> <span class="op">=</span> <span class="fu"><a href="../reference/convert_to_entrez_id.html">convert_to_entrez_id</a></span><span class="op">(</span><span class="va">gene_diff_score</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   gene id might be SYMBOL (p =  1.000 )</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_diff_score</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          10         100        1000       10000       10001       10002 </span></span>
<span><span class="co">##  0.02683074  0.12681435 -0.13737449  0.13335944 -0.18805099  0.12855168</span></span></code></pre>
<p>Load the <strong>clusterProfiler</strong> package.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span></code></pre></div>
<p>Next we perform GSEA on different gene set sources.</p>
<p>Before we start, <code>gene_diff_score</code> must be pre-sorted.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_diff_score</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">gene_diff_score</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="go-enrichment">GO enrichment<a class="anchor" aria-label="anchor" href="#go-enrichment"></a>
</h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">gene_diff_score</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                                               Description</span></span>
<span><span class="co">## GO:0007186 GO:0007186              G protein-coupled receptor signaling pathway</span></span>
<span><span class="co">## GO:1902850 GO:1902850 microtubule cytoskeleton organization involved in mitosis</span></span>
<span><span class="co">## GO:0098655 GO:0098655                 monoatomic cation transmembrane transport</span></span>
<span><span class="co">## GO:0030001 GO:0030001                                       metal ion transport</span></span>
<span><span class="co">## GO:0045087 GO:0045087                                    innate immune response</span></span>
<span><span class="co">## GO:0098662 GO:0098662                  inorganic cation transmembrane transport</span></span>
<span><span class="co">##            setSize enrichmentScore       NES       pvalue     p.adjust</span></span>
<span><span class="co">## GO:0007186     472       0.3655642  1.867258 1.636225e-10 8.262937e-07</span></span>
<span><span class="co">## GO:1902850      69      -0.4809040 -2.172523 1.370112e-06 1.383813e-03</span></span>
<span><span class="co">## GO:0098655     426       0.3312117  1.684990 5.572633e-07 1.383813e-03</span></span>
<span><span class="co">## GO:0030001     484       0.3237042  1.658101 1.116764e-06 1.383813e-03</span></span>
<span><span class="co">## GO:0045087     469       0.3202448  1.636321 1.328397e-06 1.383813e-03</span></span>
<span><span class="co">## GO:0098662     420       0.3312151  1.685456 1.848061e-06 1.555452e-03</span></span>
<span><span class="co">##                  qvalue rank                   leading_edge</span></span>
<span><span class="co">## GO:0007186 7.344067e-07 1687 tags=39%, list=25%, signal=31%</span></span>
<span><span class="co">## GO:1902850 1.229928e-03 1730 tags=51%, list=26%, signal=38%</span></span>
<span><span class="co">## GO:0098655 1.229928e-03 1934 tags=39%, list=29%, signal=30%</span></span>
<span><span class="co">## GO:0030001 1.229928e-03 1917 tags=38%, list=29%, signal=29%</span></span>
<span><span class="co">## GO:0045087 1.229928e-03 1450 tags=30%, list=22%, signal=26%</span></span>
<span><span class="co">## GO:0098662 1.382480e-03 1934 tags=39%, list=29%, signal=30%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     core_enrichment</span></span>
<span><span class="co">## GO:0007186 23620/26033/150/9248/23295/2151/6870/3060/3362/4544/2911/6357/6714/6753/3351/6608/815/1113/6366/3360/26532/7432/2842/8685/181/5148/1237/19/9934/680/51704/729230/5573/247/8477/8601/112/107/2775/6370/2919/2099/1812/3358/4157/2783/6344/5144/5741/887/146/1909/5138/1814/5368/4985/134/186/2642/10636/348/29909/88/8851/611/23683/5132/335/1608/4161/3357/3363/11248/6002/6362/9718/7434/2841/2185/6375/2833/3630/623/10692/22844/2219/2780/4935/5330/5294/6360/153/8644/8620/5332/2788/6354/6356/4829/954/135948/113/2792/2253/4987/56288/11069/3248/10266/1258/10672/7275/2843/6364/10268/115/6900/2695/6846/776/4158/1672/3640/26648/6014/2918/345/8484/1816/5140/3000/3558/5995/7409/8111/22986/11247/56848/152/5141/7849/3717/7301/6363/5697/6622/2853/1910/109/1269/9138/7932/2798/6295/9826/9590/1230/6754/1606/6001/6915/6368/916/5996/9368/6755/5443/4852/719/2914/8527/5367/2870/6011/6010/9283/4308/108/3356/6387/2691/10798</span></span>
<span><span class="co">## GO:1902850                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      991/5347/2316/7415/3303/4926/3925/23636/4134/1058/22974/2801/7283/9493/3833/8480/387/22919/4204/332/55968/4085/9525/10252/23291/8766/3837/891/10253/7272/9183/10735/4751/9371/29899</span></span>
<span><span class="co">## GO:0098655                                                                               581/8912/2906/8514/2040/150/7225/596/5371/3786/5027/6524/6553/6568/4544/6556/23479/527/845/6366/859/783/3756/6539/10786/6540/8913/2745/51606/81614/1812/3358/25800/6530/1145/2783/486/5144/55117/9104/6334/1909/1814/5336/3758/1520/4891/90134/26133/5319/88/5581/6262/3357/9254/9512/3208/967/6528/2185/6375/3738/3780/478/3777/623/3755/6335/5330/5294/9446/5664/6564/5170/825/6263/9472/5332/23327/3749/6569/6340/9478/1020/777/931/7222/9498/4987/1258/5587/10021/3764/1535/526/10268/10063/776/5024/528/3458/5250/5026/10058/5534/10246/6534/610/11194/4734/5774/56848/491/4318/7386/6363/6622/482/4842/1910/8671/309/64116/6531/6546/9368/1340/3772/9296/1327/2903/4308/535/3709/3356/2905/3741/1497/6236/9001/6548/4283/8989/5348/10050/6574/774/57468/6557/7779/5142/6834/9114/1236/495/3782/9187/3768/2056/2815/5354/3785/5788/6833/6520/117/9550</span></span>
<span><span class="co">## GO:0030001   581/8912/2906/8514/2040/150/7225/596/5371/3786/5027/3060/6524/6553/6568/4544/3718/6556/1017/23479/815/9058/920/845/6366/7432/816/859/783/3756/1272/6539/10786/6540/8913/3263/9962/5187/2745/81614/1812/3358/25800/6530/2783/486/5144/55117/9104/6334/7369/1909/1814/5336/3758/4985/134/145389/4891/90134/26133/5319/88/5581/6262/3357/9254/9512/3208/967/6528/2185/6375/3738/3780/2702/478/3777/623/3755/7178/6335/5330/5294/3077/9446/5664/5170/825/6263/9472/5332/23327/3749/6569/6340/475/2512/5743/9478/1020/777/931/7222/9498/4987/10266/5587/10021/3764/1535/10268/10063/776/5024/3458/3273/5026/10058/8884/5534/10246/6534/610/11194/4734/5774/56848/491/3603/6363/6622/482/4842/1910/8671/309/64116/6531/6546/1230/9368/4504/3772/1213/2903/2288/4308/3709/3356/6387/1447/2905/3741/6236/9001/6548/6527/4283/8989/5348/10050/6574/468/774/57468/6557/7779/5142/1236/495/3782/9187/3768/2056/2815/5354/3785/5788/6833/6520/1141</span></span>
<span><span class="co">## GO:0045087                                                                                                                                                                                      3140/5802/3439/22841/5371/10475/1668/1436/3684/10346/259197/10584/3565/6223/23643/653361/3718/6357/6556/6714/1113/200316/7124/923/6366/4063/8685/6039/10666/9641/2813/3663/8942/3428/337/3441/4814/3263/27350/729/6370/5648/4321/2099/383/2213/713/71/9261/3662/7369/5336/6441/1520/7376/5063/3448/5319/348/10623/5581/6733/7726/4282/640/3456/6362/9051/64135/10333/2207/3002/6375/3630/10155/2219/914/5294/841/2671/735/6360/8764/5170/7294/639/10865/3055/7096/5988/6354/3805/1660/6356/4829/3848/10225/26191/5608/7128/3592/6318/9450/9958/5587/4360/3551/1535/6364/6036/6846/1672/9437/7535/3458/4688/8876/4435/9447/3113/7409/5770/3556/3146/3329/2268/10747/8767/7454/1380/3430/3717/7301/6363/6622/731/23138/11326/733/4332/3442/60489/6672</span></span>
<span><span class="co">## GO:0098662                                                                                          581/8912/2906/8514/2040/150/7225/596/5371/3786/5027/6524/6553/6568/4544/6556/23479/343/527/845/6366/859/783/3756/6539/10786/6540/8913/2745/51606/81614/1812/3358/25800/6530/2783/486/5144/55117/9104/6334/1909/1814/5336/3758/4891/90134/26133/5319/88/5581/6262/3357/9254/9512/3208/967/6528/2185/6375/3738/3780/478/3777/623/3755/6335/5330/5294/9446/5664/6564/5170/825/6263/9472/5332/23327/3749/6569/6340/9478/1020/777/931/7222/9498/4987/5587/10021/3764/1535/526/10268/10063/776/5024/528/3458/5250/5026/10058/5534/10246/6534/610/11194/4734/5774/56848/491/7386/6363/6622/482/4842/1910/8671/309/64116/6531/6546/9368/1340/3772/9296/1327/2903/4308/535/3709/3356/2905/3741/1497/6236/9001/6548/4283/8989/5348/10050/6574/774/57468/6557/7779/5142/6834/9114/1236/495/3782/9187/3768/2056/2815/5354/3359/3785/5788/6833/6520/117/9550</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="kegg-enrichment">KEGG enrichment<a class="anchor" aria-label="anchor" href="#kegg-enrichment"></a>
</h2>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseKEGG.html" class="external-link">gseKEGG</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">gene_diff_score</span>, organism <span class="op">=</span> <span class="st">"hsa"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                ID                             Description setSize</span></span>
<span><span class="co">## hsa04080 hsa04080 Neuroactive ligand-receptor interaction     233</span></span>
<span><span class="co">## hsa03010 hsa03010                                Ribosome      85</span></span>
<span><span class="co">## hsa04060 hsa04060  Cytokine-cytokine receptor interaction     192</span></span>
<span><span class="co">## hsa04940 hsa04940                Type I diabetes mellitus      36</span></span>
<span><span class="co">## hsa05332 hsa05332               Graft-versus-host disease      29</span></span>
<span><span class="co">## hsa04514 hsa04514                 Cell adhesion molecules      98</span></span>
<span><span class="co">##          enrichmentScore      NES       pvalue     p.adjust       qvalue rank</span></span>
<span><span class="co">## hsa04080       0.4123718 1.970068 1.511487e-08 4.912331e-06 4.057148e-06 2404</span></span>
<span><span class="co">## hsa03010       0.5118898 2.167550 1.909148e-07 3.102366e-05 2.562278e-05 2348</span></span>
<span><span class="co">## hsa04060       0.3894784 1.830022 2.013424e-06 2.181209e-04 1.801484e-04 1812</span></span>
<span><span class="co">## hsa04940       0.5967598 2.125719 2.023140e-05 1.643801e-03 1.357633e-03 1530</span></span>
<span><span class="co">## hsa05332       0.6066844 2.025408 8.653227e-05 4.687165e-03 3.871181e-03 1762</span></span>
<span><span class="co">## hsa04514       0.4276091 1.842488 7.552459e-05 4.687165e-03 3.871181e-03 1804</span></span>
<span><span class="co">##                            leading_edge</span></span>
<span><span class="co">## hsa04080 tags=52%, list=36%, signal=35%</span></span>
<span><span class="co">## hsa03010 tags=60%, list=35%, signal=40%</span></span>
<span><span class="co">## hsa04060 tags=45%, list=27%, signal=34%</span></span>
<span><span class="co">## hsa04940 tags=56%, list=23%, signal=43%</span></span>
<span><span class="co">## hsa05332 tags=62%, list=26%, signal=46%</span></span>
<span><span class="co">## hsa04514 tags=45%, list=27%, signal=33%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              core_enrichment</span></span>
<span><span class="co">## hsa04080 23620/2906/150/2901/9248/2151/6870/7425/5027/3060/3362/4544/2911/6753/3351/3360/7432/1907/9934/680/1812/3358/1145/4157/6344/5741/887/8973/146/1909/1814/5368/4985/134/186/2562/2642/551/4161/3357/3363/7434/623/4295/153/8620/2689/4829/885/2558/3001/4987/56288/2554/2695/4158/3640/5024/2918/8484/1816/6013/2688/1511/5026/2560/152/1142/5697/7200/2569/1910/1269/3953/2798/6754/6915/7068/6755/5443/4852/719/2914/5367/2903/3356/2905/2691/5173/4887/3827/1141/117/2742/2899/2561/5739/6019/1813/1392/1908/727/2690/3972/2912/2897/2692/3375/796/554/4889/1144/6752/9002/5724/3952/4988/8001/5729/1136/2890/5025</span></span>
<span><span class="co">## hsa03010                                                                                                                                                                                                                                                                                                                                                    6223/6171/6192/6232/6175/6137/6210/6143/6176/6169/6182/6128/9349/6122/6139/6168/6187/6224/6147/6160/6124/6209/9801/6218/6155/6150/6152/6194/6158/6228/6235/6233/6129/6234/6173/23521/6201/6156/6193/6132/65005/6222/6191/6157/6204/6142/6144/9045/6203/6183/4736</span></span>
<span><span class="co">## hsa04060                                                                                                                                                                       355/3439/9518/9466/1436/3552/3623/3565/8744/6357/7124/1896/920/6366/353500/1237/729230/3441/6372/6370/7293/2919/3598/3568/10563/3448/2920/1439/1440/10850/3456/6362/4050/6375/3560/2833/1271/6360/1441/8764/5008/943/2689/6354/3570/6356/3566/3553/4055/3592/7066/6364/6846/3458/1270/3595/93/3558/2688/91/3556/7046/3554/8600/3603/7850/6363/3574/11009/3600/3953/3442/8718/1230/6368/8793/9173/3590/6387/3624/3455/4283/3562/8795/2662/8794</span></span>
<span><span class="co">## hsa04940                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  355/3552/7124/3109/3111/3002/3630/5551/3553/5798/3592/3115/3119/5799/3458/3558/3113/3329/3108/3122</span></span>
<span><span class="co">## hsa05332                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          355/3552/7124/3109/3111/3002/5551/3553/3115/115653/3119/3458/3558/3113/3108/3122/3812/3125</span></span>
<span><span class="co">## hsa04514                                                                                                                                                                                                                                                                                                                                                                                         51208/5802/3684/4897/23562/923/920/1272/10666/3109/8174/4099/947/5175/29851/3111/3683/6404/3680/914/5789/5133/9378/22865/1003/3115/3897/3119/6900/7122/3385/933/3113/3695/6401/3108/6383/3122/6402/7412/9019/3125/4359/1013</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="reactome-enrichment">Reactome enrichment<a class="anchor" aria-label="anchor" href="#reactome-enrichment"></a>
</h2>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">ReactomePA</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ReactomePA v1.44.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use ReactomePA in published research, please cite:</span></span>
<span><span class="co">## Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization. Molecular BioSystems 2016, 12(2):477-479</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/gsePathway.html" class="external-link">gsePathway</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">gene_diff_score</span>, organism <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                          ID                          Description setSize</span></span>
<span><span class="co">## R-HSA-373076   R-HSA-373076 Class A/1 (Rhodopsin-like receptors)     198</span></span>
<span><span class="co">## R-HSA-1640170 R-HSA-1640170                           Cell Cycle     347</span></span>
<span><span class="co">## R-HSA-69278     R-HSA-69278                  Cell Cycle, Mitotic     275</span></span>
<span><span class="co">## R-HSA-500792   R-HSA-500792                  GPCR ligand binding     268</span></span>
<span><span class="co">## R-HSA-68886     R-HSA-68886                              M Phase     171</span></span>
<span><span class="co">## R-HSA-388396   R-HSA-388396           GPCR downstream signalling     403</span></span>
<span><span class="co">##               enrichmentScore       NES       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## R-HSA-373076        0.4403472  2.071012 1.415276e-09 9.291284e-07 7.538205e-07</span></span>
<span><span class="co">## R-HSA-1640170      -0.3244639 -1.887086 8.608944e-10 9.291284e-07 7.538205e-07</span></span>
<span><span class="co">## R-HSA-69278        -0.3427274 -1.967048 2.582519e-09 1.130282e-06 9.170207e-07</span></span>
<span><span class="co">## R-HSA-500792        0.4002429  1.935977 1.764099e-08 5.790656e-06 4.698075e-06</span></span>
<span><span class="co">## R-HSA-68886        -0.3745608 -2.005034 9.406336e-08 2.470104e-05 2.004045e-05</span></span>
<span><span class="co">## R-HSA-388396        0.3456868  1.739450 1.372912e-07 3.004388e-05 2.437520e-05</span></span>
<span><span class="co">##               rank                   leading_edge</span></span>
<span><span class="co">## R-HSA-373076  1666 tags=43%, list=25%, signal=33%</span></span>
<span><span class="co">## R-HSA-1640170 1750 tags=39%, list=26%, signal=30%</span></span>
<span><span class="co">## R-HSA-69278   1565 tags=38%, list=23%, signal=30%</span></span>
<span><span class="co">## R-HSA-500792  1681 tags=40%, list=25%, signal=31%</span></span>
<span><span class="co">## R-HSA-68886   1565 tags=40%, list=23%, signal=31%</span></span>
<span><span class="co">## R-HSA-388396  1681 tags=37%, list=25%, signal=30%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   core_enrichment</span></span>
<span><span class="co">## R-HSA-373076                                                                                                                                                                                                                                                                                                                                    23620/150/2151/6870/3060/3362/4544/6357/6753/3351/6366/3360/1907/1237/9934/680/729230/8477/6372/6370/2919/1812/3358/4157/887/146/1909/1814/10563/5368/4985/134/186/2920/611/551/4161/3357/10850/3363/9718/2841/6375/2833/623/10692/4935/6360/4295/153/8620/6354/6356/4829/885/4987/6364/6846/4158/3640/8484/1816/5995/8111/152/6363/5697/7200/2853/1910/1269/2798/1230/6754/6915/6368/6755/5443/4852/719/5367/6010/9283/3356/6387</span></span>
<span><span class="co">## R-HSA-1640170                                                    1022/4171/5693/991/5347/6847/6119/472/5687/5520/10714/5433/6502/4926/8317/11130/7443/10213/7514/140609/7014/23636/9184/1875/1032/701/23511/7027/4438/1063/1457/7529/1058/22974/23198/5594/4000/9491/2801/4218/890/1736/8815/1616/5932/7283/699/9493/5436/5934/10733/5432/5690/994/5700/5431/5982/91754/5685/1460/7153/2237/5981/5711/8480/10844/3364/22919/5519/3161/2810/332/5529/5695/4085/1859/5706/9232/990/25836/5905/203068/5933/11339/23291/5683/54107/2010/23279/7341/6117/5721/5928/546/1031/3837/10121/7324/891/7531/10133/902/996/1719/5528/7156/1030/5686/5702/8556/5500/5720/7465/23649/995/23133/3065/5753/9183/6118/11065/10735/5537/1111/5689/5437/9663/4751/1029/2033/11004/5527/5526/5810/9631</span></span>
<span><span class="co">## R-HSA-69278                                                                                                                                                                                                                4926/8317/11130/7443/10213/7514/140609/23636/9184/1875/1032/701/23511/7027/1063/1457/1058/22974/23198/5594/4000/9491/2801/4218/890/8815/7283/699/9493/5934/10733/5690/994/5700/5982/91754/5685/1460/7153/2237/5981/5711/8480/10844/22919/5519/3161/332/5529/5695/4085/1859/5706/9232/990/25836/5905/203068/5933/23291/5683/54107/2010/23279/7341/6117/5721/5928/1031/3837/10121/7324/891/7531/10133/902/996/1719/5528/1030/5686/5702/8556/5500/5720/7465/23649/995/23133/3065/5753/9183/6118/11065/10735/5689/9663/4751/1029/2033/11004/5527/5526/9631</span></span>
<span><span class="co">## R-HSA-500792                                                                                                                                                                                                                        23620/150/2151/6870/3060/3362/4544/2911/6357/6753/3351/6608/6366/3360/7432/1907/1237/9934/680/729230/8477/6372/6370/2919/1812/3358/4157/2783/6344/5741/887/146/1909/1814/10563/5368/50846/4985/134/186/2642/2920/611/551/4161/3357/6469/10850/3363/7482/9718/7434/2841/6375/2833/7476/623/10692/4935/6360/4295/153/8620/2788/6354/6356/4829/885/7475/2792/4987/10266/6364/10268/2695/6846/4158/3640/2918/8484/1816/5995/8111/152/6363/5697/7200/2853/1910/1269/2798/1230/6754/6915/6368/6755/5443/4852/719/2914/5367/6010/9283/3356/6387/2691</span></span>
<span><span class="co">## R-HSA-68886                                                                                                                                                                                                                                                                                                                                                                                                    4926/11130/7443/10213/7514/140609/23636/9184/701/23511/1063/1457/1058/23198/5594/4000/9491/2801/8815/7283/699/9493/10733/5690/5700/91754/5685/1460/5711/8480/10844/22919/5519/332/5529/5695/4085/5706/9232/25836/5905/203068/5683/2010/23279/7341/5721/3837/10121/7324/891/7531/996/5528/5686/5702/5720/23133/9183/11065/10735/5689/9663/4751/11004/5527/5526/9631</span></span>
<span><span class="co">## R-HSA-388396  23620/5136/150/7225/2151/6870/3060/3362/4544/2911/6357/6714/6753/3351/815/6366/3360/7432/1907/816/1237/9934/680/729230/5573/8477/8601/112/107/6372/6370/2919/1812/3358/4157/2783/6344/5144/5741/887/146/1909/5138/1814/10563/5368/5321/4985/134/186/5568/9639/2642/2920/10636/611/5581/1608/551/4161/3357/10850/3363/6002/5153/7434/2841/6375/2833/623/10692/2780/4935/5330/5294/63940/6360/4295/153/5170/8620/5332/2788/4829/885/113/1020/7222/2792/4987/10266/10672/2843/7410/6364/10268/115/2695/6846/4158/3640/2918/5580/8484/1816/5140/5995/7409/8111/5534/814/152/5141/6363/5697/7200/2853/1910/109/388/1269/9138/4168/2798/9826/1230/6754/1606/10276/6001/6915/6368/5996/6755/5443/4852/719/2914/8527/5367/2870/6010/9283/108/8503/3709/3356/6387/10235/2691</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="do-enrichment">DO enrichment<a class="anchor" aria-label="anchor" href="#do-enrichment"></a>
</h2>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">DOSE</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DOSE v3.26.2  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use DOSE in published research, please cite:</span></span>
<span><span class="co">## Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/gseDO.html" class="external-link">gseDO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">gene_diff_score</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                      Description setSize enrichmentScore</span></span>
<span><span class="co">## DOID:2841   DOID:2841                           asthma     252       0.3901582</span></span>
<span><span class="co">## DOID:1176   DOID:1176                bronchial disease     260       0.3782700</span></span>
<span><span class="co">## DOID:10952 DOID:10952                        nephritis     242       0.3680361</span></span>
<span><span class="co">## DOID:7148   DOID:7148             rheumatoid arthritis     434       0.3221059</span></span>
<span><span class="co">## DOID:3498   DOID:3498 pancreatic ductal adenocarcinoma      97      -0.3900281</span></span>
<span><span class="co">## DOID:417     DOID:417               autoimmune disease     420       0.3126948</span></span>
<span><span class="co">##                  NES       pvalue     p.adjust       qvalue rank</span></span>
<span><span class="co">## DOID:2841   1.902793 5.772259e-08 6.355257e-05 5.188957e-05 1653</span></span>
<span><span class="co">## DOID:1176   1.850731 3.302009e-07 1.817756e-04 1.484166e-04 1627</span></span>
<span><span class="co">## DOID:10952  1.781216 2.123469e-06 6.702163e-04 5.472199e-04 1613</span></span>
<span><span class="co">## DOID:7148   1.637489 2.434937e-06 6.702163e-04 5.472199e-04 1704</span></span>
<span><span class="co">## DOID:3498  -1.941271 1.723829e-05 3.795870e-03 3.099262e-03 1061</span></span>
<span><span class="co">## DOID:417    1.587569 2.595639e-05 4.762997e-03 3.888904e-03 1762</span></span>
<span><span class="co">##                              leading_edge</span></span>
<span><span class="co">## DOID:2841  tags=37%, list=25%, signal=29%</span></span>
<span><span class="co">## DOID:1176  tags=36%, list=24%, signal=29%</span></span>
<span><span class="co">## DOID:10952 tags=36%, list=24%, signal=28%</span></span>
<span><span class="co">## DOID:7148  tags=34%, list=25%, signal=27%</span></span>
<span><span class="co">## DOID:3498  tags=45%, list=16%, signal=39%</span></span>
<span><span class="co">## DOID:417   tags=34%, list=26%, signal=27%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  core_enrichment</span></span>
<span><span class="co">## DOID:2841                                                                                                                                                                                                                                                                                              581/6778/355/6775/2100/3439/596/3552/5027/3623/3565/1215/6556/7124/7432/4524/259/1237/729230/2919/383/3598/3568/4056/5321/6441/134/186/6440/1439/1440/3315/4282/10850/6362/2181/3111/10333/6375/3560/2833/6404/2212/623/26762/5294/864/8764/28/5551/7096/6354/6580/6356/5743/3566/3553/3592/6318/3115/3119/1535/6364/4589/3067/6036/1672/3458/3240/3558/3113/6317/3329/4318/3603/6363/3574/4842/1910/3162/3600/7056/3631/3702/6915/6772/3122/6439/719/5367/9173/5265/2155</span></span>
<span><span class="co">## DOID:1176                                                                                                                                                                                                                                                                                         581/6778/355/6775/2100/3439/596/3552/5027/3623/3565/1215/6556/7124/7432/4524/259/1237/729230/2919/383/3598/3568/4056/5321/6441/134/186/6440/1439/1440/3315/4282/10850/6362/2181/3111/10333/6375/4317/3560/2833/6404/2212/623/26762/5294/864/8764/28/5551/7096/6354/6580/6340/6356/5743/3566/3553/3592/6318/3115/3119/1535/6364/4589/3067/6036/1672/3458/3240/3558/3113/6317/3329/4318/3603/6363/3574/4842/1910/3162/3600/7056/3631/3702/6915/6772/3122/6439/719/5367/9173/5265</span></span>
<span><span class="co">## DOID:10952                                                                                                                                                                                                                                                                                                                                1026/355/4653/6775/3439/7225/1436/3552/5027/3684/3565/1017/1113/7124/920/3164/4524/3818/19/8654/3663/729/7293/4321/2213/3568/10563/4056/2153/7490/186/348/5175/5076/3315/4282/1286/6469/6362/3683/1843/2185/6375/2833/2702/2212/4311/623/1285/119/26762/841/5551/7096/6354/6696/6356/5133/5743/231/3553/931/3119/4043/6364/2215/4868/345/2204/3458/285/894/6635/7454/4318/22925/7436/4842/6401/55806/7450/6402/3815/1773/719/4548/4489</span></span>
<span><span class="co">## DOID:7148  1026/4193/355/6775/2100/596/5371/9518/1668/1436/4772/3552/5027/3623/3684/4118/3565/8744/653361/4544/6357/6556/8190/1113/7124/920/6366/4524/9641/3109/3663/729230/1836/337/5609/9420/6370/383/2213/3481/4481/10563/5321/6441/1520/8828/348/335/29851/3315/4282/3456/6362/9051/5783/64135/967/8651/4050/6528/7434/3002/6573/6375/4317/3560/2833/2212/1513/7178/914/5294/3077/8764/639/4689/5008/943/8454/6354/3570/6696/6356/1544/4286/5133/9047/5743/3566/3553/4055/26191/7128/931/10266/3119/6364/2215/843/2204/7535/3385/3458/5580/4688/3240/3558/11278/3146/4350/3554/3329/56848/2034/4318/8600/3603/3717/3574/3162/11009/1269/3600/6401/3953/896/1280/7056/5657/3108/8718/1230/6368/4170/8793/1773/5979/2821/4855/2870/5265/4249/3709/671/3356/6387/3820/7412/3624</span></span>
<span><span class="co">## DOID:3498                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              2316/7052/2805/3993/7040/6774/2044/3569/4221/811/3918/8743/1956/5292/5327/1545/5653/857/387/1015/10874/5268/6768/6279/977/11315/1524/10551/2064/966/5655/8574/101/51100/5879/3655/3065/6286/3066/1969/6850/2033/7428/2152</span></span>
<span><span class="co">## DOID:417                                        1026/581/355/6775/2100/3439/596/1436/3552/5027/3060/3684/3565/653361/6556/7124/923/920/6366/4524/4063/27040/3818/213/9021/5172/10666/1118/3663/43/7306/3428/337/2315/8174/1588/7293/7299/4321/2099/383/1145/2213/10563/5368/6441/4985/348/335/29851/10544/5069/2705/3315/4282/4593/640/3456/64135/1843/6528/2185/3002/6375/3560/2833/3738/230/3630/6285/3777/2212/368/5521/841/4295/639/6097/5551/3570/6356/2936/5133/5743/3566/8034/3553/26191/5445/3592/4987/3115/23385/3119/7295/2215/6846/4158/6649/345/2204/3458/10750/3240/3558/2688/3146/8678/3329/814/491/4318/1380/3603/3574/1638/3162/3600/6401/7056/5657/326/4345/6772/3122/5443/6402/1773/4548/5367/4855/5265/6387/3726/2691/4780/7412/868/4283/7035/10125/3562/3125</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="msigdb-enrichment">MSigDB enrichment<a class="anchor" aria-label="anchor" href="#msigdb-enrichment"></a>
</h2>
<p>There is no built-in function specific for MSigDB gene sets, but there is a universal function <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">GSEA()</a></code> which accepts manually-specified gene sets. The gene sets object is simply a two-column data frame:</p>
<ul>
<li>the first column is the gene set ID</li>
<li>the second column is the gene ID</li>
</ul>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span>
<span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="va">map</span> <span class="op">=</span> <span class="va">gene_sets</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"entrez_gene"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">map</span><span class="op">$</span><span class="va">entrez_gene</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">map</span><span class="op">$</span><span class="va">entrez_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">GSEA</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">gene_diff_score</span>, TERM2GENE <span class="op">=</span> <span class="va">map</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                        ID</span></span>
<span><span class="co">## HALLMARK_G2M_CHECKPOINT           HALLMARK_G2M_CHECKPOINT</span></span>
<span><span class="co">## HALLMARK_MITOTIC_SPINDLE         HALLMARK_MITOTIC_SPINDLE</span></span>
<span><span class="co">## HALLMARK_ALLOGRAFT_REJECTION HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS                 HALLMARK_E2F_TARGETS</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_DN           HALLMARK_UV_RESPONSE_DN</span></span>
<span><span class="co">## HALLMARK_PROTEIN_SECRETION     HALLMARK_PROTEIN_SECRETION</span></span>
<span><span class="co">##                                               Description setSize</span></span>
<span><span class="co">## HALLMARK_G2M_CHECKPOINT           HALLMARK_G2M_CHECKPOINT     142</span></span>
<span><span class="co">## HALLMARK_MITOTIC_SPINDLE         HALLMARK_MITOTIC_SPINDLE     116</span></span>
<span><span class="co">## HALLMARK_ALLOGRAFT_REJECTION HALLMARK_ALLOGRAFT_REJECTION     168</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS                 HALLMARK_E2F_TARGETS     129</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_DN           HALLMARK_UV_RESPONSE_DN     122</span></span>
<span><span class="co">## HALLMARK_PROTEIN_SECRETION     HALLMARK_PROTEIN_SECRETION      80</span></span>
<span><span class="co">##                              enrichmentScore       NES       pvalue</span></span>
<span><span class="co">## HALLMARK_G2M_CHECKPOINT           -0.4110969 -2.125053 3.860772e-08</span></span>
<span><span class="co">## HALLMARK_MITOTIC_SPINDLE          -0.3665070 -1.842667 2.212711e-05</span></span>
<span><span class="co">## HALLMARK_ALLOGRAFT_REJECTION       0.3851576  1.782814 2.481982e-05</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS              -0.3557647 -1.803866 3.383854e-05</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_DN           -0.3540220 -1.782161 8.660680e-05</span></span>
<span><span class="co">## HALLMARK_PROTEIN_SECRETION        -0.3611803 -1.698696 9.509585e-04</span></span>
<span><span class="co">##                                  p.adjust       qvalue rank</span></span>
<span><span class="co">## HALLMARK_G2M_CHECKPOINT      1.930386e-06 1.544309e-06 1751</span></span>
<span><span class="co">## HALLMARK_MITOTIC_SPINDLE     4.136637e-04 3.309310e-04 1717</span></span>
<span><span class="co">## HALLMARK_ALLOGRAFT_REJECTION 4.136637e-04 3.309310e-04 1530</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS         4.229818e-04 3.383854e-04 1740</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_DN      8.660680e-04 6.928544e-04 1557</span></span>
<span><span class="co">## HALLMARK_PROTEIN_SECRETION   7.924654e-03 6.339724e-03 1369</span></span>
<span><span class="co">##                                                leading_edge</span></span>
<span><span class="co">## HALLMARK_G2M_CHECKPOINT      tags=46%, list=26%, signal=35%</span></span>
<span><span class="co">## HALLMARK_MITOTIC_SPINDLE     tags=42%, list=26%, signal=32%</span></span>
<span><span class="co">## HALLMARK_ALLOGRAFT_REJECTION tags=38%, list=23%, signal=30%</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS         tags=40%, list=26%, signal=30%</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_DN      tags=38%, list=23%, signal=30%</span></span>
<span><span class="co">## HALLMARK_PROTEIN_SECRETION   tags=32%, list=20%, signal=26%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                  core_enrichment</span></span>
<span><span class="co">## HALLMARK_G2M_CHECKPOINT      2176/4171/4678/991/5347/6558/7040/4926/8317/2130/4288/7514/3925/4691/9184/7027/1063/6541/1058/22974/4853/11104/1164/8451/1033/3151/890/1736/3609/699/9493/5887/10951/1432/10733/994/7153/3364/3161/332/4085/4212/9232/990/5933/546/1031/3837/6599/8065/996/4082/2935/3799/23649/7272/8819/6118/1756/11065/1111/4751/11004/1810/7290</span></span>
<span><span class="co">## HALLMARK_MITOTIC_SPINDLE                                                                                  6709/998/5347/2316/9771/2317/4926/23095/8476/10006/6654/1063/382/23637/23647/22974/4853/11104/4627/394/9344/7414/10928/613/699/9493/7153/5981/1739/10844/81/6093/22919/332/2017/10435/11135/7531/8936/996/667/4082/3799/7272/10013/4751/11004/832/9371</span></span>
<span><span class="co">## HALLMARK_ALLOGRAFT_REJECTION          355/6775/9466/3565/6223/6357/7124/920/3109/729230/43/8477/2213/3662/10563/1520/3111/3683/8651/4050/10333/3002/3560/2833/914/4689/5551/7096/6354/6356/3848/10225/3566/3553/3001/3059/3592/9450/3551/822/9437/7535/3458/7163/3558/894/4830/2268/8767/7454/4318/3603/3717/6363/3574/3600/896/3108/3702/1230/924/916/6772/3122</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS                                                                   4171/4678/991/5347/6119/10714/7884/4288/7514/3925/701/10527/7037/1164/1033/10635/7283/3609/5395/5631/23468/10733/1786/994/5982/7153/5981/7913/6749/3364/3161/332/4085/9232/9833/203068/5902/6117/10606/7398/1031/2935/7465/23649/9319/9183/6118/1111/1029/5511/11004/5810</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_DN                                                                                                    657/2869/8476/4325/140609/50937/7049/9112/3397/6541/4853/5376/5999/23576/4092/1432/4131/8405/120/273/857/1739/4781/1859/25836/11099/302/4052/546/3488/2064/5737/1301/29970/51460/5468/323/594/2202/2650/8871/23469/9975/136/2152/5195</span></span>
<span><span class="co">## HALLMARK_PROTEIN_SECRETION                                                                                                                                                                                                    9367/9871/10059/5594/6197/23673/5868/1956/538/1182/8774/8027/9525/664/372/11079/26286/27236/8773/9805/7251/667/9183/3920/9341/3998</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="example-for-other-organism">Example for other organism<a class="anchor" aria-label="anchor" href="#example-for-other-organism"></a>
</h2>
<ol style="list-style-type: decimal">
<li>GO enrichment</li>
</ol>
<p><code><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO()</a></code> accepts an <code>OrgDb</code> object as the source of the GO gene sets. We take pig as an example.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Ss.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keys</a></span><span class="op">(</span><span class="va">org.Ss.eg.db</span>, keytype <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="va">scores</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">scores</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Ss.eg.db</span>, pvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID</span></span>
<span><span class="co">## GO:0007584 GO:0007584</span></span>
<span><span class="co">## GO:0071880 GO:0071880</span></span>
<span><span class="co">## GO:0030521 GO:0030521</span></span>
<span><span class="co">## GO:0007156 GO:0007156</span></span>
<span><span class="co">## GO:0007340 GO:0007340</span></span>
<span><span class="co">## GO:0006303 GO:0006303</span></span>
<span><span class="co">##                                                                   Description</span></span>
<span><span class="co">## GO:0007584                                               response to nutrient</span></span>
<span><span class="co">## GO:0071880 adenylate cyclase-activating adrenergic receptor signaling pathway</span></span>
<span><span class="co">## GO:0030521                                androgen receptor signaling pathway</span></span>
<span><span class="co">## GO:0007156    homophilic cell adhesion via plasma membrane adhesion molecules</span></span>
<span><span class="co">## GO:0007340                                                  acrosome reaction</span></span>
<span><span class="co">## GO:0006303           double-strand break repair via nonhomologous end joining</span></span>
<span><span class="co">##            setSize enrichmentScore       NES       pvalue  p.adjust    qvalue</span></span>
<span><span class="co">## GO:0007584      16       0.6676030  2.008815 0.0009276183 0.6940372 0.6882389</span></span>
<span><span class="co">## GO:0071880      11      -0.7330745 -1.951329 0.0005951761 0.6940372 0.6882389</span></span>
<span><span class="co">## GO:0030521      10       0.7487575  1.916931 0.0015791367 0.6940372 0.6882389</span></span>
<span><span class="co">## GO:0007156      25      -0.5728791 -1.909460 0.0017790080 0.6940372 0.6882389</span></span>
<span><span class="co">## GO:0007340      16      -0.6443482 -1.904359 0.0023734148 0.6940372 0.6882389</span></span>
<span><span class="co">## GO:0006303      20      -0.5764523 -1.830950 0.0030407513 0.6940372 0.6882389</span></span>
<span><span class="co">##             rank                   leading_edge</span></span>
<span><span class="co">## GO:0007584  7694 tags=56%, list=17%, signal=47%</span></span>
<span><span class="co">## GO:0071880  9442 tags=64%, list=21%, signal=50%</span></span>
<span><span class="co">## GO:0030521  3439  tags=50%, list=8%, signal=46%</span></span>
<span><span class="co">## GO:0007156  7480 tags=44%, list=16%, signal=37%</span></span>
<span><span class="co">## GO:0007340 10471 tags=56%, list=23%, signal=43%</span></span>
<span><span class="co">## GO:0006303  5191 tags=40%, list=11%, signal=35%</span></span>
<span><span class="co">##                                                                                                 core_enrichment</span></span>
<span><span class="co">## GO:0007584                           100621622/397530/397227/100525560/100513621/396689/100127359/397145/396655</span></span>
<span><span class="co">## GO:0071880                                         397421/552898/100144471/100520233/100519931/397355/100524631</span></span>
<span><span class="co">## GO:0030521                                                          100621622/100625057/751859/100512591/397582</span></span>
<span><span class="co">## GO:0007156 414736/100523408/100625833/100156315/100038012/641355/100155738/100511695/100522817/397247/100520035</span></span>
<span><span class="co">## GO:0007340                        396891/100511655/404696/100153434/397111/100620649/100415935/406180/100622575</span></span>
<span><span class="co">## GO:0006303                      100524635/100156734/100155891/100523873/100625894/100153781/100522033/100516672</span></span></code></pre>
<ol start="2" style="list-style-type: decimal">
<li>KEGG enrichment</li>
</ol>
<p>The KEGG code of a specific organism can be found at <a href="https://rest.kegg.jp/list/organism" class="external-link uri">https://rest.kegg.jp/list/organism</a></p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseKEGG.html" class="external-link">gseKEGG</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">scores</span>, organism <span class="op">=</span> <span class="st">"ssc"</span>, pvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                ID                          Description setSize enrichmentScore</span></span>
<span><span class="co">## ssc03450 ssc03450           Non-homologous end-joining      13      -0.6797457</span></span>
<span><span class="co">## ssc05340 ssc05340             Primary immunodeficiency      36      -0.4276687</span></span>
<span><span class="co">## ssc00051 ssc00051      Fructose and mannose metabolism      32       0.4317999</span></span>
<span><span class="co">## ssc04620 ssc04620 Toll-like receptor signaling pathway     108      -0.3310094</span></span>
<span><span class="co">## ssc03020 ssc03020                       RNA polymerase      33       0.4214929</span></span>
<span><span class="co">## ssc00350 ssc00350                  Tyrosine metabolism      37       0.4054984</span></span>
<span><span class="co">##                NES      pvalue  p.adjust    qvalue  rank</span></span>
<span><span class="co">## ssc03450 -1.918880 0.001519221 0.5074199 0.5037418  3324</span></span>
<span><span class="co">## ssc05340 -1.591436 0.012883167 0.7706707 0.7650844 11651</span></span>
<span><span class="co">## ssc00051  1.527517 0.017413926 0.7706707 0.7650844  7916</span></span>
<span><span class="co">## ssc04620 -1.520184 0.006778414 0.7706707 0.7650844  6263</span></span>
<span><span class="co">## ssc03020  1.511670 0.025418710 0.7706707 0.7650844  9034</span></span>
<span><span class="co">## ssc00350  1.496839 0.032303562 0.7706707 0.7650844  5105</span></span>
<span><span class="co">##                            leading_edge</span></span>
<span><span class="co">## ssc03450  tags=31%, list=7%, signal=29%</span></span>
<span><span class="co">## ssc05340 tags=36%, list=26%, signal=27%</span></span>
<span><span class="co">## ssc00051 tags=38%, list=17%, signal=31%</span></span>
<span><span class="co">## ssc04620 tags=25%, list=14%, signal=22%</span></span>
<span><span class="co">## ssc03020 tags=39%, list=20%, signal=32%</span></span>
<span><span class="co">## ssc00350 tags=24%, list=11%, signal=22%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                 core_enrichment</span></span>
<span><span class="co">## ssc03450                                                                                                                                                                                                100155891/100515113/100153781/100156379</span></span>
<span><span class="co">## ssc05340                                                                                                               100627308/100152350/733649/100526050/396636/100190992/100739325/100519979/100271930/396781/106508259/100517711/100127355</span></span>
<span><span class="co">## ssc00051                                                                                                                   100626946/100523270/100513376/100153520/100520846/100523241/100515523/100154112/100513664/396816/100522855/100157582</span></span>
<span><span class="co">## ssc04620 396668/397384/100144476/396607/751869/100152100/100310806/397441/100157735/100156630/445459/100135663/396913/100037289/100153325/100135681/396629/396609/397076/100037296/595129/100625946/396621/396613/100127355/100153785/100125828</span></span>
<span><span class="co">## ssc03020                                                                                                         100154035/100156445/100517396/100152157/100627477/100156229/100737214/100516367/102159905/397001/100152094/100157033/100737040</span></span>
<span><span class="co">## ssc00350                                                                                                                                                          396967/100155530/397443/733609/110256000/100520329/100512615/110256733/574066</span></span></code></pre>
<ol start="3" style="list-style-type: decimal">
<li>MSigDB</li>
</ol>
<p>Use <code><a href="https://igordot.github.io/msigdbr/reference/msigdbr_species.html" class="external-link">msigdbr::msigdbr_species()</a></code> to see which organisms are supported.</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"pig"</span>, category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="va">map</span> <span class="op">=</span> <span class="va">gene_sets</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"entrez_gene"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">map</span><span class="op">$</span><span class="va">entrez_gene</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">map</span><span class="op">$</span><span class="va">entrez_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">GSEA</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">scores</span>, TERM2GENE <span class="op">=</span> <span class="va">map</span>, pvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                ID</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_UP                   HALLMARK_UV_RESPONSE_UP</span></span>
<span><span class="co">## HALLMARK_PI3K_AKT_MTOR_SIGNALING HALLMARK_PI3K_AKT_MTOR_SIGNALING</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE     HALLMARK_INFLAMMATORY_RESPONSE</span></span>
<span><span class="co">## HALLMARK_NOTCH_SIGNALING                 HALLMARK_NOTCH_SIGNALING</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS                         HALLMARK_E2F_TARGETS</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                   HALLMARK_MYC_TARGETS_V2</span></span>
<span><span class="co">##                                                       Description setSize</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_UP                   HALLMARK_UV_RESPONSE_UP     155</span></span>
<span><span class="co">## HALLMARK_PI3K_AKT_MTOR_SIGNALING HALLMARK_PI3K_AKT_MTOR_SIGNALING     103</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE     HALLMARK_INFLAMMATORY_RESPONSE     194</span></span>
<span><span class="co">## HALLMARK_NOTCH_SIGNALING                 HALLMARK_NOTCH_SIGNALING      32</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS                         HALLMARK_E2F_TARGETS     198</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                   HALLMARK_MYC_TARGETS_V2      57</span></span>
<span><span class="co">##                                  enrichmentScore       NES     pvalue  p.adjust</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_UP                0.2656682  1.287022 0.05091650 0.8316566</span></span>
<span><span class="co">## HALLMARK_PI3K_AKT_MTOR_SIGNALING      -0.2799971 -1.273421 0.08284024 0.8316566</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE        -0.2553856 -1.269014 0.05468750 0.8316566</span></span>
<span><span class="co">## HALLMARK_NOTCH_SIGNALING              -0.3527942 -1.245633 0.16532258 0.8316566</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS                  -0.2465108 -1.231646 0.07572816 0.8316566</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                0.3006973  1.218717 0.14168378 0.8316566</span></span>
<span><span class="co">##                                     qvalue  rank                   leading_edge</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_UP          0.8316566  7281 tags=23%, list=16%, signal=19%</span></span>
<span><span class="co">## HALLMARK_PI3K_AKT_MTOR_SIGNALING 0.8316566 10897 tags=36%, list=24%, signal=27%</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE   0.8316566 12369 tags=32%, list=27%, signal=23%</span></span>
<span><span class="co">## HALLMARK_NOTCH_SIGNALING         0.8316566 11311 tags=34%, list=25%, signal=26%</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS             0.8316566  7968 tags=24%, list=18%, signal=20%</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2          0.8316566 13898 tags=44%, list=31%, signal=30%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                core_enrichment</span></span>
<span><span class="co">## HALLMARK_UV_RESPONSE_UP                                                                                                                                                                                                                                                       397118/100511107/100153111/100151951/110259225/397227/594852/100511951/100521449/397147/397064/100627072/100628213/397323/100511587/100154715/100522503/396907/100152073/100126235/397468/397543/100519301/100144486/100512716/100517196/780412/102159943/397639/100155053/110255321/692188/397465/397471/397244</span></span>
<span><span class="co">## HALLMARK_PI3K_AKT_MTOR_SIGNALING                                                                                                                                                                                                  100519842/100627149/100505408/100518810/100192438/100523972/100156099/100381266/100152116/100233194/100521736/100155762/110256149/100512827/100524587/100519412/397493/100038030/100622774/100515364/100519790/397070/100126852/445532/100156264/100155905/100135663/396726/397470/396609/100286871/100240743/100513842/100625946/414426/100125828/100512983</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE   100526076/110258749/100626904/396750/397106/100152890/397195/100514123/733577/100625126/100155079/100625349/397094/100500718/397122/396945/396658/100152462/100157931/780407/397282/100736963/100037941/396875/733692/100522231/733595/100127164/414370/100155428/397564/100516594/100622336/399503/100271930/100511843/396607/751869/397588/100156614/100625072/396915/397614/100037289/100157723/102168145/100624969/397108/100126285/100524634/780421/100144421/100135681/397076/397179/100156931/100516171/396613/100520035/100037306/100512581/100513084</span></span>
<span><span class="co">## HALLMARK_NOTCH_SIGNALING                                                                                                                                                                                                                                                                                                                                                                                                                                                            100524958/100620430/102158881/100620481/100739492/397297/100523218/110260638/100738589/100524215/100156773</span></span>
<span><span class="co">## HALLMARK_E2F_TARGETS                                                                                                 100192434/100512157/100525428/100514809/100515361/106504085/102164100/100157339/100737806/100519979/100519426/100624116/100525658/100152604/100142626/100513605/397266/100625497/100518577/100624907/100156398/100626508/110255643/100523151/100620553/110261209/100627239/397218/100152381/396553/100153874/100523170/110261061/100737986/100625681/100623654/100329127/100522950/100153939/100523938/100525752/397589/100515210/100523071/100516672/100156379/100515360</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                                                                                                                                                                                                                                                                                                                             396953/100154050/100516425/100517974/595107/100144492/100154484/100517044/100518596/397575/100156024/100516080/100155960/397641/110256064/100579176/100157224/100737962/414385/100525313/100512446/100157673/100622432/100511136/100621511</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="examples-for-less-well-studied-organisms">Examples for less well-studied organisms<a class="anchor" aria-label="anchor" href="#examples-for-less-well-studied-organisms"></a>
</h2>
<div class="section level3">
<h3 id="when-an-orgdb-object-is-avaiable">When an <code>OrgDb</code> object is avaiable<a class="anchor" aria-label="anchor" href="#when-an-orgdb-object-is-avaiable"></a>
</h3>
<p>Taking dolphin as an example. Since <code>org_db</code> is an <code>OrgDb</code> object, we can directly use <code><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO()</a></code>:</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: BiocFileCache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: dbplyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'AnnotationHub'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     cache</span></span></code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ah</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">org_db</span> <span class="op">=</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH112418"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keys</a></span><span class="op">(</span><span class="va">org_db</span>, keytype <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="va">scores</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">scores</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org_db</span>, pvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                                              Description</span></span>
<span><span class="co">## GO:0034614 GO:0034614             cellular response to reactive oxygen species</span></span>
<span><span class="co">## GO:0062197 GO:0062197                     cellular response to chemical stress</span></span>
<span><span class="co">## GO:0002753 GO:0002753 cytosolic pattern recognition receptor signaling pathway</span></span>
<span><span class="co">## GO:0034599 GO:0034599                    cellular response to oxidative stress</span></span>
<span><span class="co">## GO:0000302 GO:0000302                      response to reactive oxygen species</span></span>
<span><span class="co">## GO:0046777 GO:0046777                              protein autophosphorylation</span></span>
<span><span class="co">##            setSize enrichmentScore       NES      pvalue  p.adjust    qvalue</span></span>
<span><span class="co">## GO:0034614      12      -0.6875666 -1.863103 0.002624399 0.8408457 0.8408457</span></span>
<span><span class="co">## GO:0062197      25      -0.5477809 -1.835197 0.003733124 0.8408457 0.8408457</span></span>
<span><span class="co">## GO:0002753      10       0.7105868  1.815385 0.002413892 0.8408457 0.8408457</span></span>
<span><span class="co">## GO:0034599      24      -0.5479354 -1.807229 0.004151789 0.8408457 0.8408457</span></span>
<span><span class="co">## GO:0000302      15      -0.6112687 -1.761619 0.009529663 0.8408457 0.8408457</span></span>
<span><span class="co">## GO:0046777      10      -0.6818647 -1.757078 0.006897975 0.8408457 0.8408457</span></span>
<span><span class="co">##            rank                   leading_edge</span></span>
<span><span class="co">## GO:0034614 1311  tags=42%, list=5%, signal=40%</span></span>
<span><span class="co">## GO:0062197 8658 tags=56%, list=30%, signal=39%</span></span>
<span><span class="co">## GO:0002753 3136 tags=40%, list=11%, signal=36%</span></span>
<span><span class="co">## GO:0034599 8658 tags=58%, list=30%, signal=41%</span></span>
<span><span class="co">## GO:0000302 3962 tags=73%, list=14%, signal=63%</span></span>
<span><span class="co">## GO:0046777 6253 tags=70%, list=22%, signal=55%</span></span>
<span><span class="co">##                                                                                                                                        core_enrichment</span></span>
<span><span class="co">## GO:0034614                                                                                           101325383/101322317/101334941/101322568/101316954</span></span>
<span><span class="co">## GO:0062197 101329517/101336363/101327438/101337383/101326215/101320655/101319146/101327377/101317019/101325383/101322317/101334941/101322568/101316954</span></span>
<span><span class="co">## GO:0002753                                                                                                     101337459/101326456/101336951/101316515</span></span>
<span><span class="co">## GO:0034599 101329517/101336363/101327438/101337383/101326215/101320655/101319146/101327377/101317019/101325383/101322317/101334941/101322568/101316954</span></span>
<span><span class="co">## GO:0000302                               101329517/101336363/101327438/101326215/101321208/101319583/101325383/101322317/101334941/101322568/101316954</span></span>
<span><span class="co">## GO:0046777                                                                       101319304/101327293/101318843/101319901/101337119/101338549/101324514</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="gsekegg-also-supports-many-other-orgainsms">gseKEGG() also supports many other orgainsms<a class="anchor" aria-label="anchor" href="#gsekegg-also-supports-many-other-orgainsms"></a>
</h3>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseKEGG.html" class="external-link">gseKEGG</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">scores</span>, organism <span class="op">=</span> <span class="va">...</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="manually-construct-the-gene-sets">Manually construct the gene sets<a class="anchor" aria-label="anchor" href="#manually-construct-the-gene-sets"></a>
</h3>
<p>We have introduced many ways to obtain gene sets for less well-studies organisms. The only thing to do here is to convert gene sets to a two-column data frame where gene sets are in the first column and genes are in the second column. Then use <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">GSEA()</a></code>.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">GSEA</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">scores</span>, TERM2GENE <span class="op">=</span> <span class="va">...</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="look-at-the-tb-object">Look at the <code>tb</code> object<a class="anchor" aria-label="anchor" href="#look-at-the-tb-object"></a>
</h2>
<p>We used the same variable name <code>tb</code> for the object returned by the various enrichment functions. They are all in the same format. It looks like a table, but be careful, it is actually not:</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "gseaResult"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "DOSE"</span></span></code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Formal class 'gseaResult' [package "DOSE"] with 13 slots</span></span>
<span><span class="co">##   ..@ result     :'data.frame':  1399 obs. of  11 variables:</span></span>
<span><span class="co">##   .. ..$ ID             : chr [1:1399] "GO:0034614" "GO:0062197" "GO:0002753" "GO:0034599" ...</span></span>
<span><span class="co">##   .. ..$ Description    : chr [1:1399] "cellular response to reactive oxygen species" "cellular response to chemical stress" "cytosolic pattern recognition receptor signaling pathway" "cellular response to oxidative stress" ...</span></span>
<span><span class="co">##   .. ..$ setSize        : int [1:1399] 12 25 10 24 15 10 26 26 35 36 ...</span></span>
<span><span class="co">##   .. ..$ enrichmentScore: num [1:1399] -0.688 -0.548 0.711 -0.548 -0.611 ...</span></span>
<span><span class="co">##   .. ..$ NES            : num [1:1399] -1.86 -1.84 1.82 -1.81 -1.76 ...</span></span>
<span><span class="co">##   .. ..$ pvalue         : num [1:1399] 0.00262 0.00373 0.00241 0.00415 0.00953 ...</span></span>
<span><span class="co">##   .. ..$ p.adjust       : num [1:1399] 0.841 0.841 0.841 0.841 0.841 ...</span></span>
<span><span class="co">##   .. ..$ qvalue         : num [1:1399] 0.841 0.841 0.841 0.841 0.841 ...</span></span>
<span><span class="co">##   .. ..$ rank           : num [1:1399] 1311 8658 3136 8658 3962 ...</span></span>
<span><span class="co">##   .. ..$ leading_edge   : chr [1:1399] "tags=42%, list=5%, signal=40%" "tags=56%, list=30%, signal=39%" "tags=40%, list=11%, signal=36%" "tags=58%, list=30%, signal=41%" ...</span></span>
<span><span class="co">##   .. ..$ core_enrichment: chr [1:1399] "101325383/101322317/101334941/101322568/101316954" "101329517/101336363/101327438/101337383/101326215/101320655/101319146/101327377/101317019/101325383/101322317/1"| __truncated__ "101337459/101326456/101336951/101316515" "101329517/101336363/101327438/101337383/101326215/101320655/101319146/101327377/101317019/101325383/101322317/1"| __truncated__ ...</span></span>
<span><span class="co">##   ..@ organism   : chr "Tursiops truncatus"</span></span>
<span><span class="co">##   ..@ setType    : chr "BP"</span></span>
<span><span class="co">##   ..@ geneSets   :List of 6859</span></span>
<span><span class="co">##   .. ..$ GO:0000002: chr [1:5] "101319146" "101322275" "101323809" "101330306" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000003: chr [1:87] "101335701" "101315473" "101315779" "101315844" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000018: chr [1:8] "101315900" "101319565" "101320166" "101320856" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000027: chr "101323418"</span></span>
<span><span class="co">##   .. ..$ GO:0000028: chr "101318467"</span></span>
<span><span class="co">##   .. ..$ GO:0000038: chr [1:4] "101321997" "101330026" "101334078" "101336885"</span></span>
<span><span class="co">##   .. ..$ GO:0000041: chr [1:14] "101321017" "101336438" "101318005" "101320570" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000045: chr [1:9] "101316053" "101320545" "101322490" "101331278" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000050: chr [1:3] "101316168" "101319666" "101339624"</span></span>
<span><span class="co">##   .. ..$ GO:0000054: chr "101332258"</span></span>
<span><span class="co">##   .. ..$ GO:0000055: chr "101332258"</span></span>
<span><span class="co">##   .. ..$ GO:0000070: chr [1:26] "101315815" "101316537" "101316695" "101318454" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000075: chr [1:23] "101316695" "101316733" "101318513" "101318624" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000077: chr [1:14] "101316733" "101318513" "101319146" "101320252" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000079: chr [1:11] "101324115" "101324169" "101326949" "101327555" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000082: chr [1:15] "101317019" "101319146" "101321188" "101322275" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000086: chr [1:15] "101316752" "101318513" "101319016" "101319117" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000096: chr [1:7] "101318072" "101320414" "101320471" "101320949" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000097: chr [1:6] "101320414" "101320471" "101320949" "101331056" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000103: chr "101333364"</span></span>
<span><span class="co">##   .. ..$ GO:0000122: chr [1:35] "101316020" "101319446" "101320166" "101321309" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000132: chr [1:4] "101321450" "101328452" "101330245" "101338521"</span></span>
<span><span class="co">##   .. ..$ GO:0000154: chr [1:3] "101318650" "101324394" "101335796"</span></span>
<span><span class="co">##   .. ..$ GO:0000165: chr [1:48] "101315703" "101315844" "101316074" "101316105" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000183: chr [1:2] "101320305" "101323967"</span></span>
<span><span class="co">##   .. ..$ GO:0000184: chr [1:12] "101317208" "101320986" "101322520" "101324720" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000209: chr [1:33] "101315684" "101316554" "101319013" "101319502" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000212: chr "101326437"</span></span>
<span><span class="co">##   .. ..$ GO:0000226: chr [1:61] "101315537" "101316394" "101316518" "101317000" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000244: chr "101334043"</span></span>
<span><span class="co">##   .. ..$ GO:0000245: chr [1:9] "101321273" "101322182" "101327231" "101330237" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000255: chr "101338869"</span></span>
<span><span class="co">##   .. ..$ GO:0000266: chr [1:9] "101319797" "101320522" "101322601" "101323172" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000270: chr "101327680"</span></span>
<span><span class="co">##   .. ..$ GO:0000271: chr [1:9] "101323622" "101324350" "101329904" "101331306" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000272: chr "101330457"</span></span>
<span><span class="co">##   .. ..$ GO:0000278: chr [1:95] "101315815" "101316323" "101316500" "101316537" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000280: chr [1:45] "101315815" "101316537" "101316695" "101316904" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000281: chr [1:8] "101318454" "101318926" "101329144" "101331655" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000288: chr [1:15] "101316306" "101322274" "101324369" "101325801" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000289: chr [1:4] "101322274" "101324369" "101325801" "101337967"</span></span>
<span><span class="co">##   .. ..$ GO:0000290: chr [1:5] "101322274" "101329036" "101331769" "101332555" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000291: chr [1:4] "101322274" "101325801" "101326312" "101332993"</span></span>
<span><span class="co">##   .. ..$ GO:0000301: chr "101335576"</span></span>
<span><span class="co">##   .. ..$ GO:0000302: chr [1:15] "101316954" "101319583" "101321208" "101322317" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000303: chr [1:2] "101325383" "101327438"</span></span>
<span><span class="co">##   .. ..$ GO:0000305: chr [1:2] "101325383" "101327438"</span></span>
<span><span class="co">##   .. ..$ GO:0000320: chr "101317123"</span></span>
<span><span class="co">##   .. ..$ GO:0000338: chr [1:3] "101320449" "101328620" "101337481"</span></span>
<span><span class="co">##   .. ..$ GO:0000375: chr [1:74] "101315667" "101316328" "101316554" "101317380" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000377: chr [1:74] "101315667" "101316328" "101316554" "101317380" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000380: chr [1:6] "101317431" "101324720" "101325768" "101328568" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000381: chr [1:5] "101324720" "101325768" "101328568" "101333925" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000387: chr [1:11] "101318020" "101318583" "101318698" "101319671" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000394: chr [1:5] "101319579" "101327326" "101333559" "101338459" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000395: chr "101327231"</span></span>
<span><span class="co">##   .. ..$ GO:0000398: chr [1:74] "101315667" "101316328" "101316554" "101317380" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000413: chr [1:18] "101316348" "101322428" "101325477" "101325900" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000422: chr [1:11] "101316053" "101316255" "101317608" "101319146" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000423: chr [1:6] "101316053" "101317608" "101319146" "101325333" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000425: chr [1:2] "101322317" "101334078"</span></span>
<span><span class="co">##   .. ..$ GO:0000429: chr "101329457"</span></span>
<span><span class="co">##   .. ..$ GO:0000430: chr "101329457"</span></span>
<span><span class="co">##   .. ..$ GO:0000432: chr "101329457"</span></span>
<span><span class="co">##   .. ..$ GO:0000436: chr "101329457"</span></span>
<span><span class="co">##   .. ..$ GO:0000459: chr "101339801"</span></span>
<span><span class="co">##   .. ..$ GO:0000460: chr [1:2] "101318462" "101339801"</span></span>
<span><span class="co">##   .. ..$ GO:0000463: chr [1:3] "101315781" "101320270" "101327032"</span></span>
<span><span class="co">##   .. ..$ GO:0000466: chr "101339801"</span></span>
<span><span class="co">##   .. ..$ GO:0000467: chr "101339801"</span></span>
<span><span class="co">##   .. ..$ GO:0000469: chr "101339801"</span></span>
<span><span class="co">##   .. ..$ GO:0000470: chr [1:3] "101315781" "101320270" "101327032"</span></span>
<span><span class="co">##   .. ..$ GO:0000491: chr "101320200"</span></span>
<span><span class="co">##   .. ..$ GO:0000493: chr "101320200"</span></span>
<span><span class="co">##   .. ..$ GO:0000578: chr "101320735"</span></span>
<span><span class="co">##   .. ..$ GO:0000711: chr "101330719"</span></span>
<span><span class="co">##   .. ..$ GO:0000722: chr "101328782"</span></span>
<span><span class="co">##   .. ..$ GO:0000723: chr [1:14] "101315665" "101319304" "101320264" "101324392" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000724: chr [1:20] "101315841" "101315900" "101317036" "101319304" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000725: chr [1:21] "101315841" "101315900" "101317036" "101319304" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000731: chr [1:7] "101319167" "101319446" "101323869" "101324001" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000768: chr "109552510"</span></span>
<span><span class="co">##   .. ..$ GO:0000819: chr [1:28] "101315815" "101316537" "101316695" "101318454" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000902: chr [1:41] "101316391" "101316954" "101317123" "101317226" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000904: chr [1:17] "101317123" "101317632" "101318355" "101319110" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000910: chr [1:18] "101316518" "101316752" "101318454" "101318926" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000912: chr "101324212"</span></span>
<span><span class="co">##   .. ..$ GO:0000915: chr "101324212"</span></span>
<span><span class="co">##   .. ..$ GO:0000956: chr [1:30] "101316306" "101317208" "101320986" "101322274" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000957: chr "101333806"</span></span>
<span><span class="co">##   .. ..$ GO:0000958: chr "101333806"</span></span>
<span><span class="co">##   .. ..$ GO:0000959: chr [1:5] "101317667" "101319586" "101320594" "101333806" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000963: chr [1:2] "101317667" "101319586"</span></span>
<span><span class="co">##   .. ..$ GO:0000964: chr "101319586"</span></span>
<span><span class="co">##   .. ..$ GO:0000966: chr [1:6] "101319586" "101322154" "101326512" "101327203" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001101: chr [1:5] "101316666" "101320291" "101326859" "101328169" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001109: chr "101326268"</span></span>
<span><span class="co">##   .. ..$ GO:0001111: chr "101326268"</span></span>
<span><span class="co">##   .. ..$ GO:0001173: chr "101324154"</span></span>
<span><span class="co">##   .. .. [list output truncated]</span></span>
<span><span class="co">##   ..@ geneList   : Named num [1:28931] 4.48 4.25 3.9 3.74 3.53 ...</span></span>
<span><span class="co">##   .. ..- attr(*, "names")= chr [1:28931] "117308403" "101331945" "101338926" "109548744" ...</span></span>
<span><span class="co">##   ..@ keytype    : chr "ENTREZID"</span></span>
<span><span class="co">##   ..@ permScores : num[0 , 0 ] </span></span>
<span><span class="co">##   ..@ params     :List of 6</span></span>
<span><span class="co">##   .. ..$ pvalueCutoff : num 1</span></span>
<span><span class="co">##   .. ..$ eps          : num 1e-10</span></span>
<span><span class="co">##   .. ..$ pAdjustMethod: chr "BH"</span></span>
<span><span class="co">##   .. ..$ exponent     : num 1</span></span>
<span><span class="co">##   .. ..$ minGSSize    : num 10</span></span>
<span><span class="co">##   .. ..$ maxGSSize    : num 500</span></span>
<span><span class="co">##   ..@ gene2Symbol: chr(0) </span></span>
<span><span class="co">##   ..@ readable   : logi FALSE</span></span>
<span><span class="co">##   ..@ termsim    : num[0 , 0 ] </span></span>
<span><span class="co">##   ..@ method     : chr(0) </span></span>
<span><span class="co">##   ..@ dr         : list()</span></span></code></pre>
<style>
blockquote {
    padding: 10px 20px;
    margin: 0 0 20px;
    font-size: 14px;
    background: #eee;
    border-left: 6px #CCCCCC solid;
}
</style>
<blockquote>
<p>In the ORA analysis by <strong>clusterProfiler</strong>, we mentioned using <code>tb@result</code> returns the full enrichment table. <strong>But this is not the case for the GSEA anlaysis!</strong>. In GSEA anlaysis by <strong>clusterProfiler</strong>, <code>tb@result</code> still returns the significant terms. Add <code>pvalueCutoff = 1</code> argument if you need the full table.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>The input for e.g. <code><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO()</a></code> is a vector of gene-level scores, thus what metric is used for measuring gene-level differentiation has impact on the GSEA analysis. Here you use the following four gene-level metrics, apply them with <code><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO()</a></code> (GO BP + human) and compare the GSEA results.</p>
<p>Here we use the <code>p53_expr.rds</code> dataset.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"p53_expr.rds"</span>, package <span class="op">=</span> <span class="st">"GSEAtraining"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">expr</span> <span class="op">=</span> <span class="va">lt</span><span class="op">$</span><span class="va">expr</span></span>
<span><span class="va">condition</span> <span class="op">=</span> <span class="va">lt</span><span class="op">$</span><span class="va">condition</span></span></code></pre></div>
<p>First convert gene IDs to EntreZ IDs for <code>expr</code>:</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mat</span> <span class="op">=</span> <span class="fu"><a href="../reference/convert_to_entrez_id.html">convert_to_entrez_id</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   gene id might be SYMBOL (p =  0.710 )</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<p>Then calculate the column indicies of WT and MUT</p>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ind_MUT</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">condition</span> <span class="op">==</span> <span class="st">"MUT"</span><span class="op">)</span></span>
<span><span class="va">ind_WT</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">condition</span> <span class="op">==</span> <span class="st">"WT"</span><span class="op">)</span></span></code></pre></div>
<p>The four gene-level metrics:</p>
<ol style="list-style-type: decimal">
<li>signal-to-noise ratio</li>
<li>t-value</li>
<li>log2 fold change</li>
<li>signed log10 p-values</li>
</ol>
<p>They can be calculated as:</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s2n</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">mat</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x1</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_MUT</span><span class="op">]</span></span>
<span>    <span class="va">x2</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_WT</span><span class="op">]</span></span>
<span></span>
<span>    <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu">sd</span><span class="op">(</span><span class="va">x1</span><span class="op">)</span> <span class="op">+</span> <span class="fu">sd</span><span class="op">(</span><span class="va">x2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tvalue</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">mat</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x1</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_MUT</span><span class="op">]</span></span>
<span>    <span class="va">x2</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_WT</span><span class="op">]</span></span>
<span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="va">x1</span>, <span class="va">x2</span><span class="op">)</span><span class="op">$</span><span class="va">statistic</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">log2fc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">mat</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x1</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_MUT</span><span class="op">]</span></span>
<span>    <span class="va">x2</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_WT</span><span class="op">]</span></span>
<span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">logp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">mat</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x1</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_MUT</span><span class="op">]</span></span>
<span>    <span class="va">x2</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">ind_WT</span><span class="op">]</span></span>
<span></span>
<span>    <span class="va">t</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="va">x1</span>, <span class="va">x2</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sign.html" class="external-link">sign</a></span><span class="op">(</span><span class="va">t</span><span class="op">$</span><span class="va">statistic</span><span class="op">)</span><span class="op">*</span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">t</span><span class="op">$</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>Now use these four numeric vectors to perform GSEA analysis (on GO BP gene sets).</p>
</div>
<div class="section level4">
<h4 id="solution">Solution<a class="anchor" aria-label="anchor" href="#solution"></a>
</h4>
<p>First to sort all the numeric vectors:</p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s2n</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">s2n</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">tvalue</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">tvalue</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">log2fc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">log2fc</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">logp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">logp</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">s2n</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : For some pathways, in reality P-values are less than 1e-10. You can</span></span>
<span><span class="co">## set the `eps` argument to zero for better estimation.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">tvalue</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb3</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">log2fc</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb4</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html" class="external-link">gseGO</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">logp</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<p>Get the “real data frames” for all terms.</p>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb1</span> <span class="op">=</span> <span class="va">tb1</span><span class="op">@</span><span class="va">result</span></span>
<span><span class="va">tb2</span> <span class="op">=</span> <span class="va">tb2</span><span class="op">@</span><span class="va">result</span></span>
<span><span class="va">tb3</span> <span class="op">=</span> <span class="va">tb3</span><span class="op">@</span><span class="va">result</span></span>
<span><span class="va">tb4</span> <span class="op">=</span> <span class="va">tb4</span><span class="op">@</span><span class="va">result</span></span></code></pre></div>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jolars/eulerr" class="external-link">eulerr</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://jolars.github.io/eulerr/reference/euler.html" class="external-link">euler</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"s2n"</span> <span class="op">=</span> <span class="va">tb1</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">tb1</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span>,</span>
<span>                <span class="st">"tvalue"</span> <span class="op">=</span> <span class="va">tb2</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">tb2</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span>,</span>
<span>                <span class="st">"log2fc"</span> <span class="op">=</span> <span class="va">tb3</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">tb3</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span>,</span>
<span>                <span class="st">"logp"</span> <span class="op">=</span> <span class="va">tb4</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">tb4</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    quantities <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic2_05_clusterProfiler_gsea_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s2n</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tvalue</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">log2fc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">logp</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic2_05_clusterProfiler_gsea_files/figure-html/unnamed-chunk-25-1.png" width="768"></p>
<p>Conclusion: do not use log2 fold change for GSEA analysis, and using signal-to-noise or t-value (or other similar statistics) is always a good choice.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
