<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 1-01: Get GO gene sets • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 1-01: Get GO gene sets">
<meta property="og:description" content="GSEAtraining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 1-01: Get GO gene sets</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2024-01-21</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic1_01_GO.Rmd" class="external-link"><code>vignettes/topic1_01_GO.Rmd</code></a></small>
      <div class="hidden name"><code>topic1_01_GO.Rmd</code></div>

    </div>

    
    
<p>To obtain GO gene sets, we need to use two packages: <strong>GO.db</strong> and <strong>org.*.db</strong> packages.</p>
<div class="section level2">
<h2 id="the-go-db-package">The GO.db package<a class="anchor" aria-label="anchor" href="#the-go-db-package"></a>
</h2>
<p>We first load the package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GO.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: package 'S4Vectors' was built under R version 4.3.2</span></span></code></pre>
<div class="section level3">
<h3 id="use-go-db-as-a-database-object">use GO.db as a database object<a class="anchor" aria-label="anchor" href="#use-go-db-as-a-database-object"></a>
</h3>
<p>Before we introduce the <strong>GO.db</strong> package, we need to know that the <strong>GO.db</strong> package is built on top of the low-level infrastructure package <strong>AnnotationDbi</strong>. Thus, almost all Bioconductor “official data packages” inhert the same interface for querying data. Two important points:</p>
<ol style="list-style-type: decimal">
<li>There is a database object which has the same name as the package name.</li>
<li>They all implement the <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select()</a></code> interface.</li>
</ol>
<p>E.g. for <strong>GO.db</strong>, the object <code>GO.db</code> contains a connection to the internal SQLite database object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GO.db</span></span></code></pre></div>
<pre><code><span><span class="co">## GODb object:</span></span>
<span><span class="co">## | GOSOURCENAME: Gene Ontology</span></span>
<span><span class="co">## | GOSOURCEURL: http://current.geneontology.org/ontology/go-basic.obo</span></span>
<span><span class="co">## | GOSOURCEDATE: 2023-01-01</span></span>
<span><span class="co">## | Db type: GODb</span></span>
<span><span class="co">## | package: AnnotationDbi</span></span>
<span><span class="co">## | DBSCHEMA: GO_DB</span></span>
<span><span class="co">## | GOEGSOURCEDATE: 2023-Mar05</span></span>
<span><span class="co">## | GOEGSOURCENAME: Entrez Gene</span></span>
<span><span class="co">## | GOEGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span></span>
<span><span class="co">## | DBSCHEMAVERSION: 2.1</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Please see: help('select') for usage information</span></span></code></pre>
<p><code>GO.db</code> can be thought as a huge table, where <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select()</a></code> can be used to extract records from certain columns under certain constraits.</p>
<p><img src="go_select.png"></p>
<p><code>GO.db</code> is a relatively simple database. The typical use is as:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">GO.db</span>, keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:0000001"</span>, <span class="st">"GO:0000002"</span><span class="op">)</span>, columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ONTOLOGY"</span>, <span class="st">"TERM"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">##         GOID ONTOLOGY                             TERM</span></span>
<span><span class="co">## 1 GO:0000001       BP        mitochondrion inheritance</span></span>
<span><span class="co">## 2 GO:0000002       BP mitochondrial genome maintenance</span></span></code></pre>
<p>It can be read as “select records in the ONTOLOGY and TERM columns of the GO.db table where GOID is <a href="GO:0000001" class="uri">GO:0000001</a> or <a href="GO:0000002" class="uri">GO:0000002</a>”.</p>
<p>The valid columns names can be obtained by <code>columns(GO.db)</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">columns</a></span><span class="op">(</span><span class="va">GO.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "DEFINITION" "GOID"       "ONTOLOGY"   "TERM"</span></span></code></pre>
<p>But note for <strong>GO.db</strong>, there are already objects generated for retrieving GO data more easily. It is unlikely in practice to directly use <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="the-hierarchical-relations-of-go-terms">The hierarchical relations of GO terms<a class="anchor" aria-label="anchor" href="#the-hierarchical-relations-of-go-terms"></a>
</h3>
<p><strong>GO.db</strong> provides variables that contain relations between GO terms. Taking biological process (BP) namespace as an example, there are the following four variables (similar for other two namespaces, but with <code>GOCC</code> and <code>GOMF</code> prefix).</p>
<ul>
<li><code>GOBPCHILDREN</code></li>
<li><code>GOBPPARENTS</code></li>
<li><code>GOBPOFFSPRING</code></li>
<li><code>GOBPANCESTOR</code></li>
</ul>
<p><code>GOBPCHILDREN</code> and <code>GOBPPARENTS</code> contain parent-child relations. <code>GOBPOFFSPRING</code> contains all offspring terms of GO terms (i.e., all downstream terms of a term in the GO tree) and <code>GOBPANCESTOR</code> contains all ancestor terms of a GO term (i.e., all upstream terms of a term). The information in the four variables are actually redudant, e.g., all the other three objects can be constructed from <code>GOBPCHILDREN</code>. However, these pre-computated objects will save time in downstream analysis because traversing the GO tree is time-consuming.</p>
<p>The four variables are in the same format (objects of the <code>AnnDbBimap</code> class). Taking <code>GOBPCHILDREN</code> as an example, we can convert it to a simpler format by:</p>
<ul>
<li><code><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list()</a></code></li>
<li><code>asTable()</code></li>
</ul>
<p><code><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list()</a></code> is also suggested by <strong>GO.db</strong> in its documentations.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOBPCHILDREN</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000001`</span></span>
<span><span class="co">## [1] NA</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##      part of </span></span>
<span><span class="co">## "GO:0032042" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000003`</span></span>
<span><span class="co">##          isa          isa      part of          isa          isa          isa </span></span>
<span><span class="co">## "GO:0019953" "GO:0019954" "GO:0022414" "GO:0032504" "GO:0032505" "GO:0075325" </span></span>
<span><span class="co">##          isa </span></span>
<span><span class="co">## "GO:1990277" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000011`</span></span>
<span><span class="co">## [1] NA</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000012`</span></span>
<span><span class="co">##            regulates negatively regulates positively regulates </span></span>
<span><span class="co">##         "GO:1903516"         "GO:1903517"         "GO:1903518" </span></span>
<span><span class="co">##                  isa                  isa </span></span>
<span><span class="co">##         "GO:1903823"         "GO:1990396" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000017`</span></span>
<span><span class="co">## [1] NA</span></span></code></pre>
<p><code>lt</code> is a simple list of vectors where each vector are child terms of a specific GO term, e.g., <code>GO:0000002</code> has a child term <code>GO:0032042</code>. The element vectors in <code>lt</code> are also named and the names represent the relation of the child term to the parent term. When the element vector has a value <code>NA</code>, e.g. <code>GO::0000001</code>, this means the GO term is a leaf in the GO tree, and it has no child term.</p>
<p><code><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable()</a></code> converts to a data frame:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">GOBPCHILDREN</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        go_id      go_id RelationshipType</span></span>
<span><span class="co">## 1 GO:0032042 GO:0000002          part of</span></span>
<span><span class="co">## 2 GO:0019953 GO:0000003              isa</span></span>
<span><span class="co">## 3 GO:0019954 GO:0000003              isa</span></span>
<span><span class="co">## 4 GO:0022414 GO:0000003          part of</span></span>
<span><span class="co">## 5 GO:0032504 GO:0000003              isa</span></span>
<span><span class="co">## 6 GO:0032505 GO:0000003              isa</span></span></code></pre>
<p>Unfortunately, the first two columns in <code>tb</code> have the same name. A good idea is to add meaningful column names to it.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"child"</span>, <span class="st">"parent"</span><span class="op">)</span></span></code></pre></div>
<p>Please note, the previous column names <code>c("child", "parent")</code> are only valid for <code>GOBPCHILDREN</code>. If it is from one of the three <code>GOBP*</code> objects, readers please inspect the output to determine proper column names for it. E.g. you should assign <code>c("parent", "child")</code> to <code>GOBPPARENTS</code>.</p>
<p>With <code>tb</code>, we can calculate the fraction of different relations of GO terms.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">GOBPCHILDREN</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##                  isa negatively regulates              part of </span></span>
<span><span class="co">##                50938                 2729                 5018 </span></span>
<span><span class="co">## positively regulates            regulates </span></span>
<span><span class="co">##                 2718                 3157</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">GOMFCHILDREN</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##     isa part of </span></span>
<span><span class="co">##   13736      11</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"MF"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-8-2.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">GOCCCHILDREN</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##     isa part of </span></span>
<span><span class="co">##    4676    1822</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"CC"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-8-3.png" width="700"></p>
<p>We can look at the distribution of the numbers of GO terms in each object.</p>
<ol style="list-style-type: decimal">
<li>Number of child terms. The following plot shows it follows a power-law distribution (majority of GO terms have very few child terms).</li>
</ol>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOBPCHILDREN</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lt</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="st">'log10'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="st">'log10'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Number of child terms"</span>, y <span class="op">=</span> <span class="st">"Number of GO terms"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"GOBPCHILDREN"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<ol start="2" style="list-style-type: decimal">
<li>Number of parent terms. The term “<a href="GO:0008150" class="uri">GO:0008150</a>” (biological process) is removed from the analysis because it is the top node in BP namespace and it has no parent term.</li>
</ol>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOBPPARENTS</span><span class="op">)</span></span>
<span><span class="va">lt</span> <span class="op">=</span> <span class="va">lt</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"GO:0008150"</span><span class="op">]</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lt</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="st">'log10'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Number of parent terms"</span>, y <span class="op">=</span> <span class="st">"Number of GO terms"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"GOBOPARENTS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p><code>GOBPCHILDREN</code> and <code>GOBPPARENTS</code> contain “local/direct” relations between GO terms, <code>GOBPOFFSPRING</code> and <code>GOBPANCESTOR</code> contain “local + distal” relations between GO terms.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOBPOFFSPRING</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000001`</span></span>
<span><span class="co">## [1] NA</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "GO:0006264" "GO:0032042" "GO:0032043" "GO:0043504" "GO:0090296"</span></span>
<span><span class="co">##  [6] "GO:0090297" "GO:0090298" "GO:0110166" "GO:0140909" "GO:1901858"</span></span>
<span><span class="co">## [11] "GO:1901859" "GO:1901860" "GO:1905951"</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">GOBPOFFSPRING</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        go_id      go_id</span></span>
<span><span class="co">## 1 GO:0006264 GO:0000002</span></span>
<span><span class="co">## 2 GO:0032042 GO:0000002</span></span>
<span><span class="co">## 3 GO:0032043 GO:0000002</span></span>
<span><span class="co">## 4 GO:0043504 GO:0000002</span></span>
<span><span class="co">## 5 GO:0090296 GO:0000002</span></span>
<span><span class="co">## 6 GO:0090297 GO:0000002</span></span></code></pre>
<p>We can compare child terms and all offspring terms of “<a href="GO:0000002" class="uri">GO:0000002</a>”:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GOBPCHILDREN</span><span class="op">[[</span><span class="st">"GO:0000002"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##      part of </span></span>
<span><span class="co">## "GO:0032042"</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GOBPOFFSPRING</span><span class="op">[[</span><span class="st">"GO:0000002"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "GO:0006264" "GO:0032042" "GO:0032043" "GO:0043504" "GO:0090296"</span></span>
<span><span class="co">##  [6] "GO:0090297" "GO:0090298" "GO:0110166" "GO:0140909" "GO:1901858"</span></span>
<span><span class="co">## [11] "GO:1901859" "GO:1901860" "GO:1905951"</span></span></code></pre>
<p>Constructing <code>GOBPOFFSPRING</code> from <code>GOBPCHILDREN</code> needs a recursive processing and it takes time:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt_children</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOBPCHILDREN</span><span class="op">)</span></span>
<span></span>
<span><span class="va">add_children</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">term</span>, <span class="va">env</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">children</span> <span class="op">=</span> <span class="va">lt_children</span><span class="op">[[</span><span class="va">term</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">children</span>, <span class="cn">NA</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>    <span class="va">env</span><span class="op">$</span><span class="va">offspring</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">env</span><span class="op">$</span><span class="va">offspring</span>, <span class="va">children</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">children</span>, <span class="va">add_children</span>, <span class="va">env</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">i</span> <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="va">lt_offspring</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">lt_children</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">i</span> <span class="op">&lt;&lt;-</span> <span class="va">i</span> <span class="op">+</span> <span class="fl">1</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">i</span>, <span class="st">"/"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lt_children</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">x</span>, <span class="cn">NA</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>        <span class="va">env</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/environment.html" class="external-link">new.env</a></span><span class="op">(</span><span class="op">)</span></span>
<span>        <span class="va">env</span><span class="op">$</span><span class="va">offspring</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span></span>
<span>        <span class="kw">for</span><span class="op">(</span><span class="va">y</span> <span class="kw">in</span> <span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>            <span class="fu">add_children</span><span class="op">(</span><span class="va">y</span>, <span class="va">env</span><span class="op">)</span></span>
<span>        <span class="op">}</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">env</span><span class="op">$</span><span class="va">offspring</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="information-of-go-terms">Information of GO terms<a class="anchor" aria-label="anchor" href="#information-of-go-terms"></a>
</h3>
<p>Only we only see the GO IDs, the object <code>GOTERM</code> can be used to get the names and long descriptions of GO terms.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GOTERM</span></span></code></pre></div>
<pre><code><span><span class="co">## TERM map for GO (object of class "GOTermsAnnDbBimap")</span></span></code></pre>
<p>Similarly, you can use <code><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list()</a></code> or <code><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable()</a></code> to format this object:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">)</span></span></code></pre></div>
<p>There are the following three types of useful information for each GO Term:</p>
<ul>
<li>GO name</li>
<li>GO description/definition</li>
<li>GO ontology/namespace</li>
</ul>
<p><code><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Term()</a></code>, <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Definition()</a></code> and <code><a href="https://rdrr.io/pkg/BiocGenerics/man/Ontology.html" class="external-link">Ontology()</a></code> can be directly applied to <code>GOTERM</code>:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Term</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                 GO:0000001 </span></span>
<span><span class="co">##                                "mitochondrion inheritance" </span></span>
<span><span class="co">##                                                 GO:0000002 </span></span>
<span><span class="co">##                         "mitochondrial genome maintenance" </span></span>
<span><span class="co">##                                                 GO:0000003 </span></span>
<span><span class="co">##                                             "reproduction" </span></span>
<span><span class="co">##                                                 GO:0000006 </span></span>
<span><span class="co">##    "high-affinity zinc transmembrane transporter activity" </span></span>
<span><span class="co">##                                                 GO:0000007 </span></span>
<span><span class="co">## "low-affinity zinc ion transmembrane transporter activity" </span></span>
<span><span class="co">##                                                 GO:0000009 </span></span>
<span><span class="co">##                   "alpha-1,6-mannosyltransferase activity"</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Definition</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                                                                                                                                                                                                                                      GO:0000001 </span></span>
<span><span class="co">##                                                                                                       "The distribution of mitochondria, including the mitochondrial genome, into daughter cells after mitosis or meiosis, mediated by interactions between mitochondria and the cytoskeleton." </span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                      GO:0000002 </span></span>
<span><span class="co">##                                                                                                                                             "The maintenance of the structure and integrity of the mitochondrial genome; includes replication and segregation of the mitochondrial chromosome." </span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                      GO:0000003 </span></span>
<span><span class="co">##                                                                                                                                                                  "The production of new individuals that contain some portion of genetic material inherited from one or more parent organisms." </span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                      GO:0000006 </span></span>
<span><span class="co">##                                      "Enables the transfer of zinc ions (Zn2+) from one side of a membrane to the other, probably powered by proton motive force. In high-affinity transport the transporter is able to bind the solute even if it is only present at very low concentrations." </span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                      GO:0000007 </span></span>
<span><span class="co">## "Enables the transfer of a solute or solutes from one side of a membrane to the other according to the reaction: Zn2+ = Zn2+, probably powered by proton motive force. In low-affinity transport the transporter is able to bind the solute only if it is present at very high concentrations." </span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                      GO:0000009 </span></span>
<span><span class="co">##                                                                                                                                                                                        "Catalysis of the transfer of a mannose residue to an oligosaccharide, forming an alpha-(1-&gt;6) linkage."</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/Ontology.html" class="external-link">Ontology</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GO:0000001 GO:0000002 GO:0000003 GO:0000006 GO:0000007 GO:0000009 </span></span>
<span><span class="co">##       "BP"       "BP"       "BP"       "MF"       "MF"       "MF"</span></span></code></pre>
<p>Moreover, you can get subset the <code>GOTERM</code> object:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GOTERM</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:0000001"</span>, <span class="st">"GO:0000002"</span>, <span class="st">"GO:0000003"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## TERM submap for GO (object of class "GOTermsAnnDbBimap")</span></span></code></pre>
<p>With <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Term()</a></code>, <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Definition()</a></code> and <code><a href="https://rdrr.io/pkg/BiocGenerics/man/Ontology.html" class="external-link">Ontology()</a></code>, it is convinient to add additional information of GO terms:</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Term</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:0000001"</span>, <span class="st">"GO:0000002"</span>, <span class="st">"GO:0000003"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                         GO:0000001                         GO:0000002 </span></span>
<span><span class="co">##        "mitochondrion inheritance" "mitochondrial genome maintenance" </span></span>
<span><span class="co">##                         GO:0000003 </span></span>
<span><span class="co">##                     "reproduction"</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/GOTerms-class.html" class="external-link">Definition</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:0000001"</span>, <span class="st">"GO:0000002"</span>, <span class="st">"GO:0000003"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                                                                                                                                GO:0000001 </span></span>
<span><span class="co">## "The distribution of mitochondria, including the mitochondrial genome, into daughter cells after mitosis or meiosis, mediated by interactions between mitochondria and the cytoskeleton." </span></span>
<span><span class="co">##                                                                                                                                                                                GO:0000002 </span></span>
<span><span class="co">##                                       "The maintenance of the structure and integrity of the mitochondrial genome; includes replication and segregation of the mitochondrial chromosome." </span></span>
<span><span class="co">##                                                                                                                                                                                GO:0000003 </span></span>
<span><span class="co">##                                                            "The production of new individuals that contain some portion of genetic material inherited from one or more parent organisms."</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/Ontology.html" class="external-link">Ontology</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:0000001"</span>, <span class="st">"GO:0000002"</span>, <span class="st">"GO:0000003"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GO:0000001 GO:0000002 GO:0000003 </span></span>
<span><span class="co">##       "BP"       "BP"       "BP"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="link-go-terms-to-genes">Link GO terms to genes<a class="anchor" aria-label="anchor" href="#link-go-terms-to-genes"></a>
</h2>
<p><strong>GO.db</strong> only contains information for GO terms. GO also provides gene annotated to GO terms, by manual curation or computational prediction. Such annotations are represented as mappings between GO IDs and gene IDs from external databases, which are usually synchronized between major public databases such NCBI.</p>
<div class="section level3">
<h3 id="org-hs-eg-db">org.Hs.eg.db<a class="anchor" aria-label="anchor" href="#org-hs-eg-db"></a>
</h3>
<p>To obtains genes in each GO term in R, Bioconductor provides a family of packages with name of <strong>org.*.db</strong>. Let’s take human for example, the corresponding package is <strong>org.Hs.eg.db</strong>. <strong>org.Hs.eg.db</strong> provides a standard way to provide mappings from Entrez gene IDs to a variaty of other databases.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<p>Similarly, <code>org.Hs.eg.db</code> is a database object.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">org.Hs.eg.db</span></span></code></pre></div>
<pre><code><span><span class="co">## OrgDb object:</span></span>
<span><span class="co">## | DBSCHEMAVERSION: 2.1</span></span>
<span><span class="co">## | Db type: OrgDb</span></span>
<span><span class="co">## | Supporting package: AnnotationDbi</span></span>
<span><span class="co">## | DBSCHEMA: HUMAN_DB</span></span>
<span><span class="co">## | ORGANISM: Homo sapiens</span></span>
<span><span class="co">## | SPECIES: Human</span></span>
<span><span class="co">## | EGSOURCEDATE: 2023-Mar05</span></span>
<span><span class="co">## | EGSOURCENAME: Entrez Gene</span></span>
<span><span class="co">## | EGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span></span>
<span><span class="co">## | CENTRALID: EG</span></span>
<span><span class="co">## | TAXID: 9606</span></span>
<span><span class="co">## | GOSOURCENAME: Gene Ontology</span></span>
<span><span class="co">## | GOSOURCEURL: http://current.geneontology.org/ontology/go-basic.obo</span></span>
<span><span class="co">## | GOSOURCEDATE: 2023-01-01</span></span>
<span><span class="co">## | GOEGSOURCEDATE: 2023-Mar05</span></span>
<span><span class="co">## | GOEGSOURCENAME: Entrez Gene</span></span>
<span><span class="co">## | GOEGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span></span>
<span><span class="co">## | KEGGSOURCENAME: KEGG GENOME</span></span>
<span><span class="co">## | KEGGSOURCEURL: ftp://ftp.genome.jp/pub/kegg/genomes</span></span>
<span><span class="co">## | KEGGSOURCEDATE: 2011-Mar15</span></span>
<span><span class="co">## | GPSOURCENAME: UCSC Genome Bioinformatics (Homo sapiens)</span></span>
<span><span class="co">## | GPSOURCEURL: </span></span>
<span><span class="co">## | GPSOURCEDATE: 2023-Feb8</span></span>
<span><span class="co">## | ENSOURCEDATE: 2023-Feb10</span></span>
<span><span class="co">## | ENSOURCENAME: Ensembl</span></span>
<span><span class="co">## | ENSOURCEURL: ftp://ftp.ensembl.org/pub/current_fasta</span></span>
<span><span class="co">## | UPSOURCENAME: Uniprot</span></span>
<span><span class="co">## | UPSOURCEURL: http://www.UniProt.org/</span></span>
<span><span class="co">## | UPSOURCEDATE: Wed Mar 15 15:58:07 2023</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Please see: help('select') for usage information</span></span></code></pre>
<p>You can use <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select()</a></code> to obtain gene-GO term relations, but there are already objects calculated and easier to use. In this package, there are two objects for mapping between GO IDs and genes:</p>
<ul>
<li><code>org.Hs.egGO2EG</code></li>
<li><code>org.Hs.egGO2ALLEGS</code></li>
</ul>
<p>The difference between the two objects is <code>org.Hs.egGO2EG</code> contains genes that are <em>directly annotated</em> to every GO term, while <code>org.Hs.egGO2ALLEGS</code> contains genes that directly assigned to the GO term, <em>as well as</em> genes assigned to all its ancestor terms. For example if term A is a parent of term B where A is more general, genes with function B should also have function A. Thus <code>org.Hs.egGO2ALLEGS</code> is the proper object for GO gene sets.</p>
<p><img src="go_anno_true_path.png"></p>
<p>Again, <code>org.Hs.egGO2ALLEGS</code> is a database object. There are two ways to obtain gene annotations to GO terms.</p>
<ul>
<li><code><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list()</a></code></li>
<li><code><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable()</a></code></li>
</ul>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egGO2ALLEGS</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000012`</span></span>
<span><span class="co">##         IGI         IDA         IDA         IDA         NAS         IDA </span></span>
<span><span class="co">##       "142"      "1161"      "2074"      "3981"      "7014"      "7141" </span></span>
<span><span class="co">##         IEA         IGI         IMP         IMP         IBA         IDA </span></span>
<span><span class="co">##      "7515"      "7515"      "7515"     "23411"     "54840"     "54840" </span></span>
<span><span class="co">##         IMP         IBA         IDA         IMP         IMP         IEA </span></span>
<span><span class="co">##     "54840"     "55775"     "55775"     "55775"    "200558" "100133315" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000017`</span></span>
<span><span class="co">##    IDA    IMP    ISS    IDA </span></span>
<span><span class="co">## "6523" "6523" "6523" "6524"</span></span></code></pre>
<p>If we compare to <code>org.Hs.egGO2EG</code> which contains incomplete genes:</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egGO2EG</span><span class="op">)</span></span>
<span><span class="va">lt2</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000012`</span></span>
<span><span class="co">##         IDA         IDA         IDA         IDA         IEA         IMP </span></span>
<span><span class="co">##      "1161"      "2074"      "3981"      "7141"      "7515"     "23411" </span></span>
<span><span class="co">##         IBA         IDA         IMP         IBA         IDA         IMP </span></span>
<span><span class="co">##     "54840"     "54840"     "54840"     "55775"     "55775"     "55775" </span></span>
<span><span class="co">##         IMP         IEA </span></span>
<span><span class="co">##    "200558" "100133315" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0000017`</span></span>
<span><span class="co">##    IDA    IMP    ISS    IDA </span></span>
<span><span class="co">## "6523" "6523" "6523" "6524"</span></span></code></pre>
<p>The gene IDs have names. They are evidence of how genes are annotated to GO terms.</p>
<p>Let’s try <code><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable()</a></code>:</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">org.Hs.egGO2ALLEGS</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   gene_id      go_id Evidence Ontology</span></span>
<span><span class="co">## 1       1 GO:0008150       ND       BP</span></span>
<span><span class="co">## 2       2 GO:0000003      IEA       BP</span></span>
<span><span class="co">## 3       2 GO:0001553      IEA       BP</span></span>
<span><span class="co">## 4       2 GO:0001867      IDA       BP</span></span>
<span><span class="co">## 5       2 GO:0001868      IDA       BP</span></span>
<span><span class="co">## 6       2 GO:0001869      IDA       BP</span></span></code></pre>
<p>Now there is an additional column <code>"Ontology"</code>. This is convinient because <code>org.Hs.egGO2ALLEGS</code> contains GO terms from the three namespaces and the <code><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list()</a></code> cannot distinguish the different namespaces.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span><span class="op">[</span><span class="va">tb</span><span class="op">$</span><span class="va">Ontology</span> <span class="op">==</span> <span class="st">"BP"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">onto</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/Ontology.html" class="external-link">Ontology</a></span><span class="op">(</span><span class="va">GOTREM</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span> <span class="fu">onto</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="st">"BP"</span> <span class="op">]</span></span></code></pre></div>
<p>Let’s calculate the frequency of evidence code:</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">$</span><span class="va">Evidence</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
<p>Now it seems we have obtained the complete gene-GO relations. Can we directly use <code>org.Hs.egGO2ALLEGS</code> as the GO gene sets? The answer is no, check the gene annotation for “<a href="GO:0000002" class="uri">GO:0000002</a>”:</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">org.Hs.egGO2ALLEGS</span><span class="op">[[</span><span class="st">"GO:0000002"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##      IMP      TAS      IDA      IMP      IMP      IMP      ISS      IMP </span></span>
<span><span class="co">##    "142"    "291"   "1763"   "1890"   "2021"   "3980"   "4205"   "4358" </span></span>
<span><span class="co">##      IMP      IBA      IDA      IMP      IMP      IBA      IDA      IMP </span></span>
<span><span class="co">##   "4976"   "5428"   "5428"   "5428"   "6240"   "6742"   "6742"   "7156" </span></span>
<span><span class="co">##      IEA      IEA      NAS      IMP      IEA      IBA       IC      IDA </span></span>
<span><span class="co">##   "7157"   "9093"   "9361"  "10000"  "10891"  "11232"  "11232"  "11232" </span></span>
<span><span class="co">##      IMP      IBA      IDA      IBA      IMP      NAS      IMP      IEA </span></span>
<span><span class="co">##  "50484"  "55186"  "55186"  "56652"  "56652"  "56652"  "64863"  "80119" </span></span>
<span><span class="co">##      IEA      IBA      IDA      IBA      IMP      TAS      IEA      IBA </span></span>
<span><span class="co">##  "83667"  "84275"  "84275"  "92667"  "92667"  "92667" "201163" "201973" </span></span>
<span><span class="co">##      IMP      ISS      IMP </span></span>
<span><span class="co">## "201973" "201973" "219736"</span></span></code></pre>
<p>A gene can be duplicatedly annotated to a GO term with differnet evidence code. Thus, to obtain the GO gene sets, we need to take the unique genes.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lt</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">lt</span>, <span class="va">unique</span><span class="op">)</span></span>
<span><span class="va">n2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lt</span>, <span class="va">length</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">n1</span>, <span class="va">n2</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">n1</span><span class="op">)</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">n1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"with duplicated"</span>, ylab <span class="op">=</span> <span class="st">"without duplicated"</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"number of genes in GO BP gene sets"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="va">tb</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<p>With <code>tb</code>, we can look at the distribution of numbers of genes in GO gene sets. It approximately follows a power-law distribution. This means majority of GO gene sets only contain intermediate numbers of genes.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/toTable.html" class="external-link">toTable</a></span><span class="op">(</span><span class="va">org.Hs.egGO2ALLEGS</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span><span class="va">tb</span><span class="op">$</span><span class="va">Ontology</span> <span class="op">==</span> <span class="st">"BP"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">t1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">$</span><span class="va">go_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">t1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">t1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x1</span>, y <span class="op">=</span> <span class="va">y1</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="st">'log10'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="st">'log10'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Number of annotated genes"</span>, y <span class="op">=</span> <span class="st">"Number of GO terms"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"GOBP"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-26-1.png" width="700"></p>
<p>And the distribution of numbers of GO gene sets that a gene is annotated.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">t2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">t2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">t2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x2</span>, y <span class="op">=</span> <span class="va">y2</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="st">'log10'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="st">'log10'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Number of gene sets"</span>, y <span class="op">=</span> <span class="st">"Number of genes"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"GOBP"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-27-1.png" width="700"></p>
<p>Biocondutor core team maintaines <strong>org.*.db</strong> for 18 organisms</p>
<table class="table">
<thead><tr class="header">
<th align="left">Package</th>
<th align="left">Organism</th>
<th align="left">Package</th>
<th align="left">Organism</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>org.Hs.eg.db</code></td>
<td align="left">Human</td>
<td align="left"><code>org.Ss.eg.db</code></td>
<td align="left">Pig</td>
</tr>
<tr class="even">
<td align="left"><code>org.Mm.eg.db</code></td>
<td align="left">Mouse</td>
<td align="left"><code>org.Gg.eg.db</code></td>
<td align="left">Chicken</td>
</tr>
<tr class="odd">
<td align="left"><code>org.Rn.eg.db</code></td>
<td align="left">Rat</td>
<td align="left"><code>org.Mmu.eg.db</code></td>
<td align="left">Rhesus monkey</td>
</tr>
<tr class="even">
<td align="left"><code>org.Dm.eg.db</code></td>
<td align="left">Fruit fly</td>
<td align="left"><code>org.Cf.eg.db</code></td>
<td align="left">Canine</td>
</tr>
<tr class="odd">
<td align="left"><code>org.At.tair.db</code></td>
<td align="left">Arabidopsis</td>
<td align="left"><code>org.EcK12.eg.db</code></td>
<td align="left">E coli strain K12</td>
</tr>
<tr class="even">
<td align="left"><code>org.Sc.sgd.db</code></td>
<td align="left">Yeast</td>
<td align="left"><code>org.Xl.eg.db</code></td>
<td align="left">African clawed frog</td>
</tr>
<tr class="odd">
<td align="left"><code>org.Dr.eg.db</code></td>
<td align="left">Zebrafish</td>
<td align="left"><code>org.Ag.eg.db</code></td>
<td align="left">Malaria mosquito</td>
</tr>
<tr class="even">
<td align="left"><code>org.Ce.eg.db</code></td>
<td align="left">Nematode</td>
<td align="left"><code>org.Pt.eg.db</code></td>
<td align="left">Chimpanzee</td>
</tr>
<tr class="odd">
<td align="left"><code>org.Bt.eg.db</code></td>
<td align="left">Bovine</td>
<td align="left"><code>org.EcSakai.eg.db</code></td>
<td align="left">E coli strain Sakai</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="use-the-select-interface">Use the <code>select()</code> interface<a class="anchor" aria-label="anchor" href="#use-the-select-interface"></a>
</h3>
<p>This is normally a two-step process:</p>
<ol style="list-style-type: decimal">
<li>get all genes</li>
<li>get all GO terms</li>
</ol>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keys</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keytype <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys <span class="op">=</span> <span class="va">all_genes</span>, keytype <span class="op">=</span> <span class="st">"ENTREZID"</span>, </span>
<span>    columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GOALL"</span>, <span class="st">"ONTOLOGYALL"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   ENTREZID      GOALL EVIDENCEALL ONTOLOGYALL</span></span>
<span><span class="co">## 1        1 GO:0003674          ND          MF</span></span>
<span><span class="co">## 2        1 GO:0005575         HDA          CC</span></span>
<span><span class="co">## 3        1 GO:0005575         IBA          CC</span></span>
<span><span class="co">## 4        1 GO:0005575         IDA          CC</span></span>
<span><span class="co">## 5        1 GO:0005575         TAS          CC</span></span>
<span><span class="co">## 6        1 GO:0005576         HDA          CC</span></span></code></pre>
<p>Then you might need to take the unique genes:</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="va">tb</span><span class="op">[</span><span class="va">tb</span><span class="op">$</span><span class="va">ONTOLOGYALL</span> <span class="op">==</span> <span class="st">"BP"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    ENTREZID      GOALL</span></span>
<span><span class="co">## 13        1 GO:0008150</span></span>
<span><span class="co">## 52        2 GO:0000003</span></span>
<span><span class="co">## 53        2 GO:0001553</span></span>
<span><span class="co">## 54        2 GO:0001867</span></span>
<span><span class="co">## 55        2 GO:0001868</span></span>
<span><span class="co">## 56        2 GO:0001869</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>We have demonstrated a gene can be duplicatedly annotated to a same GO term. Taking <code>org.Hs.egGO2ALLEGS</code>, can you calculate the percent of GO terms that have duplicated genes?</p>
</div>
<div class="section level4">
<h4 id="solution">Solution<a class="anchor" aria-label="anchor" href="#solution"></a>
</h4>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egGO2ALLEGS</span><span class="op">)</span></span>
<span><span class="va">n</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span></span>
<span><span class="va">nd</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lt</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">nd</span><span class="op">/</span><span class="va">n</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.5333566</span></span></code></pre>
</div>
</div>
<div class="section level3 tabset">
<h3 class="tabset" id="section-1"></h3>
<div class="section level4">
<h4 id="practice-2">Practice 2<a class="anchor" aria-label="anchor" href="#practice-2"></a>
</h4>
<p><code>org.Hs.egGO2ALLEGS</code> has already merged genes annotated to all offspring terms. Try to manually construct GO gene sets with <code>GOBPOFFSPRING</code> and <code>org.Hs.egGO2EGS</code>, then compare to gene sets in <code>org.Hs.egGO2ALLEGS</code>.</p>
</div>
<div class="section level4">
<h4 id="solution-1">Solution<a class="anchor" aria-label="anchor" href="#solution-1"></a>
</h4>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egGO2EG</span><span class="op">)</span></span>
<span><span class="va">lt_terms</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOBPOFFSPRING</span><span class="op">)</span></span></code></pre></div>
<p>For a GO term, we need to consider its offspring terms + the term itself:</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span><span class="op">(</span><span class="va">nm</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lt_terms</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">lt_terms</span><span class="op">[[</span><span class="va">nm</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">lt_terms</span><span class="op">[[</span><span class="va">nm</span><span class="op">]</span><span class="op">]</span>, <span class="va">nm</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Merge annotated genes for all offspring terms:</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt_genes_manual</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">lt_terms</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">lt_genes</span><span class="op">[</span><span class="va">x</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>Compare to <code>org.Hs.egGO2ALLEGS</code>:</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt_genes_all</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egGO2ALLEGS</span><span class="op">)</span></span>
<span><span class="va">lt_genes_all</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">lt_genes_all</span>, <span class="va">unique</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cn</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lt_genes_manual</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lt_genes_all</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lt_genes_manual</span><span class="op">[</span><span class="va">cn</span><span class="op">]</span>, <span class="va">length</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lt_genes_all</span><span class="op">[</span><span class="va">cn</span><span class="op">]</span>, <span class="va">length</span><span class="op">)</span>, log <span class="op">=</span> <span class="st">"xy"</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"manual gene sets"</span>, ylab <span class="op">=</span> <span class="st">"org.Hs.egGO2ALLEGS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_01_GO_files/figure-html/unnamed-chunk-36-1.png" width="700"></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
