<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 3-01: Online GREAT analysis • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 3-01: Online GREAT analysis">
<meta property="og:description" content="GSEAtraining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 3-01: Online GREAT analysis</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2024-01-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic3_01_online_GREAT.Rmd" class="external-link"><code>vignettes/topic3_01_online_GREAT.Rmd</code></a></small>
      <div class="hidden name"><code>topic3_01_online_GREAT.Rmd</code></div>

    </div>

    
    
<p>We first read the peaks (<a href="https://github.com/jokergoo/GSEAtraining/blob/master/inst/extdata/mm9_chipseq_peaks.bed" class="external-link uri">https://github.com/jokergoo/GSEAtraining/blob/master/inst/extdata/mm9_chipseq_peaks.bed</a>) as a data frame.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bed</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mm9_chipseq_peaks.bed"</span>, package <span class="op">=</span> <span class="st">"GSEAtraining"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     V1       V2       V3</span></span>
<span><span class="co">## 1 chr1 11017025 11017401</span></span>
<span><span class="co">## 2 chr1 11734825 11735326</span></span>
<span><span class="co">## 3 chr1 12605475 12606176</span></span>
<span><span class="co">## 4 chr1 12680375 12681376</span></span>
<span><span class="co">## 5 chr1 13416275 13416776</span></span>
<span><span class="co">## 6 chr1 13850975 13851726</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 2105    3</span></span></code></pre>
<div class="section level2">
<h2 id="directly-on-the-great-website">Directly on the GREAT website<a class="anchor" aria-label="anchor" href="#directly-on-the-great-website"></a>
</h2>
<p>URL: <a href="https://great.stanford.edu/great/public/html/index.php" class="external-link uri">https://great.stanford.edu/great/public/html/index.php</a></p>
</div>
<div class="section level2">
<h2 id="online-great-analysis">Online GREAT analysis<a class="anchor" aria-label="anchor" href="#online-great-analysis"></a>
</h2>
<p>Use <code><a href="https://rdrr.io/pkg/rGREAT/man/submitGreatJob.html" class="external-link">submitGreatJob()</a></code> to submit the regions to the GREAT web server. By default the species is human (hg19), so here we need to manually set the <code>species</code> argument.</p>
<p>Entering the object <code>job</code> prints the summary of the job.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/rGREAT" class="external-link">rGREAT</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: package 'S4Vectors' was built under R version 4.3.2</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">job</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/submitGreatJob.html" class="external-link">submitGreatJob</a></span><span class="op">(</span><span class="va">bed</span>, species <span class="op">=</span> <span class="st">"mm9"</span><span class="op">)</span></span>
<span><span class="va">job</span></span></code></pre></div>
<pre><code><span><span class="co">## Submit time: 2024-01-26 09:45:07 </span></span>
<span><span class="co">##   Note the results may only be avaiable on GREAT server for 24 hours.</span></span>
<span><span class="co">## Version: 4.0.4 </span></span>
<span><span class="co">## Species: mm9 </span></span>
<span><span class="co">## Inputs: 2105 regions</span></span>
<span><span class="co">## Mode: Basal plus extension </span></span>
<span><span class="co">##   Proximal: 5 kb upstream, 1 kb downstream,</span></span>
<span><span class="co">##   plus Distal: up to 1000 kb</span></span>
<span><span class="co">## Include curated regulatory domains</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Enrichment tables for following ontologies have been downloaded:</span></span>
<span><span class="co">##   None</span></span></code></pre>
<p>There are several ontologies integrated in GREAT. The total ontologies can be get by:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/availableOntologies-GreatJob-method.html" class="external-link">availableOntologies</a></span><span class="op">(</span><span class="va">job</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "GO Molecular Function"     "GO Biological Process"    </span></span>
<span><span class="co">## [3] "GO Cellular Component"     "Mouse Phenotype"          </span></span>
<span><span class="co">## [5] "Mouse Phenotype Single KO" "Human Phenotype"          </span></span>
<span><span class="co">## [7] "Ensembl Genes"</span></span></code></pre>
<p>By default three GO ontologies are used. More ontologies can be set via the <code>ontology</code> argument in <code><a href="https://rdrr.io/pkg/rGREAT/man/getEnrichmentTables-dispatch.html" class="external-link">getEnrichmentTables()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getEnrichmentTables-dispatch.html" class="external-link">getEnrichmentTables</a></span><span class="op">(</span><span class="va">job</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The default enrichment table does not contain informatin of associated</span></span>
<span><span class="co">## genes for each input region. You can set `download_by = 'tsv'` to</span></span>
<span><span class="co">## download the complete table, but note only the top 500 regions can be</span></span>
<span><span class="co">## retreived. See the following link:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## https://great-help.atlassian.net/wiki/spaces/GREAT/pages/655401/Export#Export-GlobalExport</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Except the additional gene-region association column if taking 'tsv' as</span></span>
<span><span class="co">## the source of result, all other columns are the same if you choose</span></span>
<span><span class="co">## 'json' (the default) as the source. Or you can try the local GREAT</span></span>
<span><span class="co">## analysis with the function `great()`.</span></span></code></pre>
<p><code>tb</code> is a list of three tables and each table contains enrichment results for a GO ontology:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tbl</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "GO Molecular Function" "GO Biological Process" "GO Cellular Component"</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tbl</span><span class="op">[[</span><span class="st">"GO Biological Process"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           ID                                      name Binom_Genome_Fraction</span></span>
<span><span class="co">## 1 GO:0048522   positive regulation of cellular process             0.4215839</span></span>
<span><span class="co">## 2 GO:0048518 positive regulation of biological process             0.4511084</span></span>
<span><span class="co">## 3 GO:0007275        multicellular organism development             0.4524430</span></span>
<span><span class="co">## 4 GO:0032502                     developmental process             0.4867658</span></span>
<span><span class="co">## 5 GO:0048856          anatomical structure development             0.4680561</span></span>
<span><span class="co">## 6 GO:0044767     single-organism developmental process             0.4846442</span></span>
<span><span class="co">##   Binom_Expected Binom_Observed_Region_Hits Binom_Fold_Enrichment</span></span>
<span><span class="co">## 1       887.4340                       1245              1.402921</span></span>
<span><span class="co">## 2       949.5831                       1304              1.373234</span></span>
<span><span class="co">## 3       952.3924                       1294              1.358684</span></span>
<span><span class="co">## 4      1024.6420                       1361              1.328269</span></span>
<span><span class="co">## 5       985.2581                       1322              1.341780</span></span>
<span><span class="co">## 6      1020.1760                       1356              1.329182</span></span>
<span><span class="co">##   Binom_Region_Set_Coverage Binom_Raw_PValue Binom_Adjp_BH Hyper_Total_Genes</span></span>
<span><span class="co">## 1                 0.5914489     2.233269e-55  2.917766e-51              4726</span></span>
<span><span class="co">## 2                 0.6194774     2.410611e-54  1.574732e-50              5180</span></span>
<span><span class="co">## 3                 0.6147268     1.318609e-50  5.742542e-47              4614</span></span>
<span><span class="co">## 4                 0.6465558     2.030623e-49  6.632522e-46              5254</span></span>
<span><span class="co">## 5                 0.6280285     2.644837e-49  6.768552e-46              4923</span></span>
<span><span class="co">## 6                 0.6441805     3.108405e-49  6.768552e-46              5215</span></span>
<span><span class="co">##   Hyper_Expected Hyper_Observed_Gene_Hits Hyper_Fold_Enrichment</span></span>
<span><span class="co">## 1       531.5450                      773              1.454251</span></span>
<span><span class="co">## 2       582.6075                      831              1.426346</span></span>
<span><span class="co">## 3       518.9481                      816              1.572412</span></span>
<span><span class="co">## 4       590.9305                      891              1.507792</span></span>
<span><span class="co">## 5       553.7021                      855              1.544152</span></span>
<span><span class="co">## 6       586.5441                      888              1.513953</span></span>
<span><span class="co">##   Hyper_Gene_Set_Coverage Hyper_Term_Gene_Coverage Hyper_Raw_PValue</span></span>
<span><span class="co">## 1               0.3437083                0.1635633     1.041517e-34</span></span>
<span><span class="co">## 2               0.3694976                0.1604247     7.678359e-35</span></span>
<span><span class="co">## 3               0.3628279                0.1768531     1.627532e-51</span></span>
<span><span class="co">## 4               0.3961761                0.1695851     4.357668e-49</span></span>
<span><span class="co">## 5               0.3801690                0.1736746     4.141140e-51</span></span>
<span><span class="co">## 6               0.3948422                0.1702780     1.081873e-49</span></span>
<span><span class="co">##   Hyper_Adjp_BH</span></span>
<span><span class="co">## 1  7.559678e-32</span></span>
<span><span class="co">## 2  5.901045e-32</span></span>
<span><span class="co">## 3  9.089050e-48</span></span>
<span><span class="co">## 4  9.488822e-46</span></span>
<span><span class="co">## 5  1.352600e-47</span></span>
<span><span class="co">## 6  2.826934e-46</span></span></code></pre>
<p>The volcano plot is -log10 p-values vs log2 fold enrichment:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotVolcano-dispatch.html" class="external-link">plotVolcano</a></span><span class="op">(</span><span class="va">job</span>, ontology <span class="op">=</span> <span class="st">"GO Biological Process"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic3_01_online_GREAT_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>The global region-gene association plots:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotRegionGeneAssociationGraphs-GreatJob-method.html" class="external-link">plotRegionGeneAssociationGraphs</a></span><span class="op">(</span><span class="va">job</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic3_01_online_GREAT_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotRegionGeneAssociations-dispatch.html" class="external-link">plotRegionGeneAssociations</a></span><span class="op">(</span><span class="va">job</span>, ontology <span class="op">=</span> <span class="st">"GO Molecular Function"</span>,</span>
<span>    term_id <span class="op">=</span> <span class="st">"GO:0004984"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The webpage for 'GO Molecular Function:GO:0004984' is available at:</span></span>
<span><span class="co">##   http://great.stanford.edu/public-4.0.4/cgi-bin/showTermDetails.php?termId=GO:0004984&amp;ontoName=GOMolecularFunction&amp;ontoUiName=GO Molecular Function&amp;sessionName=20240126-public-4.0.4-1ZzY2c&amp;species=mm9&amp;foreName=file1126039975169.gz&amp;backName=&amp;table=region</span></span>
<span><span class="co">##  Note the web page might be deleted from GREAT web server because it is only for temporary use.</span></span></code></pre>
<p><img src="topic3_01_online_GREAT_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
<p>The table of the associations between genes and regions.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getRegionGeneAssociations-dispatch.html" class="external-link">getRegionGeneAssociations</a></span><span class="op">(</span><span class="va">job</span>, ontology <span class="op">=</span> <span class="st">"GO Molecular Function"</span>,</span>
<span>    term_id <span class="op">=</span> <span class="st">"GO:0004984"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The webpage for 'GO Molecular Function:GO:0004984' is available at:</span></span>
<span><span class="co">##   http://great.stanford.edu/public-4.0.4/cgi-bin/showTermDetails.php?termId=GO:0004984&amp;ontoName=GOMolecularFunction&amp;ontoUiName=GO Molecular Function&amp;sessionName=20240126-public-4.0.4-1ZzY2c&amp;species=mm9&amp;foreName=file1126039975169.gz&amp;backName=&amp;table=region</span></span>
<span><span class="co">##  Note the web page might be deleted from GREAT web server because it is only for temporary use.</span></span></code></pre>
<pre><code><span><span class="co">## GRanges object with 6 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames              ranges strand | annotated_genes   dist_to_TSS</span></span>
<span><span class="co">##          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;CharacterList&gt; &lt;IntegerList&gt;</span></span>
<span><span class="co">##   [1]     chr4   43851550-43851876      * | Olfr159,Olfr155 -67832,-12724</span></span>
<span><span class="co">##   [2]     chr4   58682625-58684451      * |         Olfr267        115056</span></span>
<span><span class="co">##   [3]     chr4   58690550-58691401      * |         Olfr267        107619</span></span>
<span><span class="co">##   [4]     chr4   58735575-58736251      * |         Olfr267         62681</span></span>
<span><span class="co">##   [5]     chr4 118315550-118316051      * |          Olfr62        -22324</span></span>
<span><span class="co">##   [6]     chr7 115940725-115941576      * | Olfr514,Olfr513   28361,42778</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Try the shiny app:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/shinyReport-dispatch.html" class="external-link">shinyReport</a></span><span class="op">(</span><span class="va">job</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>Perform GREAT analysis on the GREAT website with a list of TFBS from UCSC table browser:</p>
<pre><code>clade: Mammal
genome: Human
assembly: GRCh37/hg19
group: Regulation
track: ENCODE 3 TFBS
table: 22Rv1 ZFX (encTfChipRkENCFF445OFX)</code></pre>
<p>In the “output format” field, select “BED - browser extensible data”, then click the button “get output”. In the next page, click the button “get BED”.</p>
</div>
</div>
<div class="section level3 tabset">
<h3 class="tabset" id="section-1"></h3>
<div class="section level4">
<h4 id="practice-2">Practice 2<a class="anchor" aria-label="anchor" href="#practice-2"></a>
</h4>
<p>Download the regions in practice 1 and perform online GREAT analysis with the <strong>rGREAT</strong> package.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
