<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 1-07: Get GO/KEGG gene sets for other organisms • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 1-07: Get GO/KEGG gene sets for other organisms">
<meta property="og:description" content="GSEAtraining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 1-07: Get GO/KEGG gene sets for other organisms</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2024-01-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic1_07_more_gene_sets.Rmd" class="external-link"><code>vignettes/topic1_07_more_gene_sets.Rmd</code></a></small>
      <div class="hidden name"><code>topic1_07_more_gene_sets.Rmd</code></div>

    </div>

    
    
<p>Besides those well-studied organisms, there are also resources for less well-studies organisms, mainly for KEGG and GO gene sets.</p>
<div class="section level2">
<h2 id="kegg-pathways">KEGG pathways<a class="anchor" aria-label="anchor" href="#kegg-pathways"></a>
</h2>
<p>KEGG pathway supports a huge number of organisms, just try different organism code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/ptr"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/pps"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/ggo"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/pon"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">...</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="biomartgogenesets">BioMartGOGeneSets<a class="anchor" aria-label="anchor" href="#biomartgogenesets"></a>
</h2>
<p>The <strong>BioMartGOGeneSets</strong> package has compiled GO gene sets for &gt; 700 organisms using <a href="https://www.ensembl.org/info/data/biomart/index.html" class="external-link">the biomart web service</a> (with the <strong>biomaRt</strong> package.</p>
<p>To obtain the gene sets, use the function <code><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets()</a></code>. Also you need to provide the “dataset”. The name of the dataset can be obtained by <code><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/supportedOrganisms.html" class="external-link">supportedOrganisms()</a></code> or from <a href="https://bioconductor.org/packages/devel/data/annotation/vignettes/BioMartGOGeneSets/inst/doc/supported_organisms.html" class="external-link uri">https://bioconductor.org/packages/devel/data/annotation/vignettes/BioMartGOGeneSets/inst/doc/supported_organisms.html</a>. Let’s get GO gene sets for mouse.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/BioMartGOGeneSets" class="external-link">BioMartGOGeneSets</a></span><span class="op">)</span></span>
<span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 15791</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "ENSMUSG00000019699" "ENSMUSG00000028982" "ENSMUSG00000022615"</span></span>
<span><span class="co">##  [4] "ENSMUSG00000039264" "ENSMUSG00000030557" "ENSMUSG00000030879"</span></span>
<span><span class="co">##  [7] "ENSMUSG00000032449" "ENSMUSG00000038084" "ENSMUSG00000022889"</span></span>
<span><span class="co">## [10] "ENSMUSG00000041064" "ENSMUSG00000033845" "ENSMUSG00000107283"</span></span>
<span><span class="co">## [13] "ENSMUSG00000027424" "ENSMUSG00000036875" "ENSMUSG00000038225"</span></span>
<span><span class="co">## [16] "ENSMUSG00000022292" "ENSMUSG00000004069" "ENSMUSG00000039176"</span></span>
<span><span class="co">## [19] "ENSMUSG00000025209" "ENSMUSG00000029911" "ENSMUSG00000020718"</span></span>
<span><span class="co">## [22] "ENSMUSG00000030978" "ENSMUSG00000026496" "ENSMUSG00000028893"</span></span>
<span><span class="co">## [25] "ENSMUSG00000002814" "ENSMUSG00000026365" "ENSMUSG00000015337"</span></span>
<span><span class="co">## [28] "ENSMUSG00000034203" "ENSMUSG00000059552" "ENSMUSG00000055660"</span></span>
<span><span class="co">## [31] "ENSMUSG00000028455" "ENSMUSG00000030314" "ENSMUSG00000020697"</span></span>
<span><span class="co">## [34] "ENSMUSG00000036923" "ENSMUSG00000032633" "ENSMUSG00000029167"</span></span></code></pre>
<p>The variable <code>lt</code> is a list of vectors where each vector corresponds to a GO gene set with Ensembl IDs as gene identifiers.</p>
<p>You can try the following command and see what will be printed:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mouse"</span><span class="op">)</span></span></code></pre></div>
<p>In <code><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets()</a></code>, argument <code>as_table</code> can be set to <code>TRUE</code>, then the function returns a data frame.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, as_table <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   go_geneset       ensembl_gene</span></span>
<span><span class="co">## 1 GO:0000002 ENSMUSG00000019699</span></span>
<span><span class="co">## 2 GO:0000002 ENSMUSG00000028982</span></span>
<span><span class="co">## 3 GO:0000002 ENSMUSG00000022615</span></span>
<span><span class="co">## 4 GO:0000002 ENSMUSG00000039264</span></span>
<span><span class="co">## 5 GO:0000002 ENSMUSG00000030557</span></span>
<span><span class="co">## 6 GO:0000002 ENSMUSG00000030879</span></span></code></pre>
<p>Argument <code>ontology</code> controls which category of GO gene sets. Possible values should be <code>"BP"</code>, <code>"CC"</code> and <code>"MF"</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span> <span class="co"># the default one</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"CC"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, ontology <span class="op">=</span> <span class="st">"MF"</span><span class="op">)</span></span></code></pre></div>
<p>Last, argument <code>gene_id_type</code> can be set to <code>"entrez_gene"</code> or <code>"gene_symbol"</code>, then genes in the gene sets are in Entrez IDs or gene symbols. Note this depends on specific organisms, that not every organism supports Entrez IDs or gene symbols.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, gene_id_type <span class="op">=</span> <span class="st">"entrez_gene"</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "23797"  "70556"  "72962"  "83408"  "17258"  "27397"  "192287" "74143" </span></span>
<span><span class="co">##  [9] "27393"  "208084" "27395"  "17527"  "74528"  "327762" "408022" "382985"</span></span>
<span><span class="co">## [17] "83945"  "18975"  "226153" "381760" "50776"  "20133"  "11545"  "230784"</span></span>
<span><span class="co">## [25] "21975"  "12628"  "13804"  "72170"  "22059"  "76781"  "66592"  "74244" </span></span>
<span><span class="co">## [33] "16882"  "216021" "216805" "19017"</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"mmusculus_gene_ensembl"</span>, gene_id_type <span class="op">=</span> <span class="st">"gene_symbol"</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">##  [1] "Akt3"     "Slc25a33" "Tymp"     "Gimap3"   "Mef2a"    "Mrpl17"  </span></span>
<span><span class="co">##  [7] "Slc25a36" "Opa1"     "Mrpl39"   "Pif1"     "Mrpl15"   "Mpv17"   </span></span>
<span><span class="co">## [13] "Mgme1"    "Dna2"     "Primpol"  "Rrm2b"    "Dnaja3"   "Polg"    </span></span>
<span><span class="co">## [19] "Twnk"     "Ssbp1"    "Polg2"    "Rrm1"     "Parp1"    "Sesn2"   </span></span>
<span><span class="co">## [25] "Top3a"    "Cfh"      "Endog"    "Chchd4"   "Trp53"    "Mettl4"  </span></span>
<span><span class="co">## [31] "Stoml2"   "Atg7"     "Lig3"     "Stox1"    "Flcn"     "Ppargc1a"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="orgdb-objects">OrgDb objects<a class="anchor" aria-label="anchor" href="#orgdb-objects"></a>
</h2>
<p>Biocondutor core team maintaines <strong>org.*.db</strong> packages for 18 organisms</p>
<table class="table">
<thead><tr class="header">
<th align="left">Package</th>
<th align="left">Organism</th>
<th align="left">Package</th>
<th align="left">Organism</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>org.Hs.eg.db</code></td>
<td align="left">Human</td>
<td align="left"><code>org.Ss.eg.db</code></td>
<td align="left">Pig</td>
</tr>
<tr class="even">
<td align="left"><code>org.Mm.eg.db</code></td>
<td align="left">Mouse</td>
<td align="left"><code>org.Gg.eg.db</code></td>
<td align="left">Chicken</td>
</tr>
<tr class="odd">
<td align="left"><code>org.Rn.eg.db</code></td>
<td align="left">Rat</td>
<td align="left"><code>org.Mmu.eg.db</code></td>
<td align="left">Rhesus monkey</td>
</tr>
<tr class="even">
<td align="left"><code>org.Dm.eg.db</code></td>
<td align="left">Fruit fly</td>
<td align="left"><code>org.Cf.eg.db</code></td>
<td align="left">Canine</td>
</tr>
<tr class="odd">
<td align="left"><code>org.At.tair.db</code></td>
<td align="left">Arabidopsis</td>
<td align="left"><code>org.EcK12.eg.db</code></td>
<td align="left">E coli strain K12</td>
</tr>
<tr class="even">
<td align="left"><code>org.Sc.sgd.db</code></td>
<td align="left">Yeast</td>
<td align="left"><code>org.Xl.eg.db</code></td>
<td align="left">African clawed frog</td>
</tr>
<tr class="odd">
<td align="left"><code>org.Dr.eg.db</code></td>
<td align="left">Zebrafish</td>
<td align="left"><code>org.Ag.eg.db</code></td>
<td align="left">Malaria mosquito</td>
</tr>
<tr class="even">
<td align="left"><code>org.Ce.eg.db</code></td>
<td align="left">Nematode</td>
<td align="left"><code>org.Pt.eg.db</code></td>
<td align="left">Chimpanzee</td>
</tr>
<tr class="odd">
<td align="left"><code>org.Bt.eg.db</code></td>
<td align="left">Bovine</td>
<td align="left"><code>org.EcSakai.eg.db</code></td>
<td align="left">E coli strain Sakai</td>
</tr>
</tbody>
</table>
<p>These <strong>org.*.db</strong> packages can be used in the same way as what we have seen with the <strong>org.Hs.eg.db</strong> package.</p>
<p>On <strong>AnnotationHub</strong>, there are <code>OrgDb</code> objects for a huge number of other organisms (~2000) which can be used for getting GO genes and mappings between gene IDs.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span></span>
<span><span class="co">##     as.data.frame, basename, cbind, colnames, dirname, do.call,</span></span>
<span><span class="co">##     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span></span>
<span><span class="co">##     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span></span>
<span><span class="co">##     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span></span>
<span><span class="co">##     tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocFileCache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: dbplyr</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ah</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>To search for an organism, using its latin name is more suggested. Also add the “OrgDb” (data class) keyword:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cat"</span>, <span class="st">"OrgDb"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AnnotationHub with 29 records</span></span>
<span><span class="co">## # snapshotDate(): 2023-04-25</span></span>
<span><span class="co">## # $dataprovider: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/</span></span>
<span><span class="co">## # $species: Viverra suricatta, Vermicularia truncata, Tursiops truncatus, Tr...</span></span>
<span><span class="co">## # $rdataclass: OrgDb</span></span>
<span><span class="co">## # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">## #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">## # retrieve records with, e.g., 'object[["AH111712"]]' </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##              title                               </span></span>
<span><span class="co">##   AH111712 | org.Catostomus_texanus.eg.sqlite    </span></span>
<span><span class="co">##   AH111807 | org.Medicago_truncatula.eg.sqlite   </span></span>
<span><span class="co">##   AH111810 | org.Felis_catus.eg.sqlite           </span></span>
<span><span class="co">##   AH111812 | org.Felis_silvestris_catus.eg.sqlite</span></span>
<span><span class="co">##   AH111921 | org.Suricata_suricatta.eg.sqlite    </span></span>
<span><span class="co">##   ...        ...                                 </span></span>
<span><span class="co">##   AH113351 | org.Indicator_indicator.eg.sqlite   </span></span>
<span><span class="co">##   AH113366 | org.Truncatella_angustata.eg.sqlite </span></span>
<span><span class="co">##   AH113367 | org.Truncatella_truncata.eg.sqlite  </span></span>
<span><span class="co">##   AH113440 | org.Catarrhactes_adeliae.eg.sqlite  </span></span>
<span><span class="co">##   AH113528 | org.Polistes_fuscatus.eg.sqlite</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis catus"</span>, <span class="st">"OrgDb"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AnnotationHub with 1 record</span></span>
<span><span class="co">## # snapshotDate(): 2023-04-25</span></span>
<span><span class="co">## # names(): AH111810</span></span>
<span><span class="co">## # $dataprovider: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/</span></span>
<span><span class="co">## # $species: Felis catus</span></span>
<span><span class="co">## # $rdataclass: OrgDb</span></span>
<span><span class="co">## # $rdatadateadded: 2023-04-24</span></span>
<span><span class="co">## # $title: org.Felis_catus.eg.sqlite</span></span>
<span><span class="co">## # $description: NCBI gene ID based annotations about Felis catus</span></span>
<span><span class="co">## # $taxonomyid: 9685</span></span>
<span><span class="co">## # $genome: NCBI genomes</span></span>
<span><span class="co">## # $sourcetype: NCBI/UniProt</span></span>
<span><span class="co">## # $sourceurl: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/, ftp://ftp.uniprot.org/p...</span></span>
<span><span class="co">## # $sourcesize: NA</span></span>
<span><span class="co">## # $tags: c("NCBI", "Gene", "Annotation") </span></span>
<span><span class="co">## # retrieve record with 'object[["AH111810"]]'</span></span></code></pre>
<p>Besides using <code><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query()</a></code> to search for the AnnotationHub dataset, you can also use the <strong>BiocHubsShiny</strong> package to interactively searching for datasets:</p>
<p>Now we download the data.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">org_db</span> <span class="op">=</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH111810"</span><span class="op">]</span><span class="op">]</span>  <span class="co"># using `[[` downloads the dataset</span></span></code></pre></div>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:AnnotationHub':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     cache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     I, expand.grid, unname</span></span></code></pre>
<p><code>org_db</code> is an <code>OrgDb</code> object but contains less information than the <strong>org.*.db</strong> packages:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">org_db</span></span></code></pre></div>
<pre><code><span><span class="co">## OrgDb object:</span></span>
<span><span class="co">## | DBSCHEMAVERSION: 2.1</span></span>
<span><span class="co">## | DBSCHEMA: NOSCHEMA_DB</span></span>
<span><span class="co">## | ORGANISM: Felis catus</span></span>
<span><span class="co">## | SPECIES: Felis catus</span></span>
<span><span class="co">## | CENTRALID: GID</span></span>
<span><span class="co">## | Taxonomy ID: 9685</span></span>
<span><span class="co">## | Db type: OrgDb</span></span>
<span><span class="co">## | Supporting package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Please see: help('select') for usage information</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">columns</a></span><span class="op">(</span><span class="va">org_db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "ACCNUM"      "ALIAS"       "CHR"         "ENSEMBL"     "ENTREZID"   </span></span>
<span><span class="co">##  [6] "EVIDENCE"    "EVIDENCEALL" "GENENAME"    "GID"         "GO"         </span></span>
<span><span class="co">## [11] "GOALL"       "ONTOLOGY"    "ONTOLOGYALL" "PMID"        "REFSEQ"     </span></span>
<span><span class="co">## [16] "SYMBOL"</span></span></code></pre>
<p>You can obtain the GO gene sets manually by <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select()</a></code>, taking the <code>"ENTREZID"</code> and <code>"GOALL"</code> columns:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keys</a></span><span class="op">(</span><span class="va">org_db</span>, keytype <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org_db</span>, keys <span class="op">=</span> <span class="va">all_genes</span>, keytype <span class="op">=</span> <span class="st">"ENTREZID"</span>, columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GOALL"</span>, <span class="st">"ONTOLOGYALL"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    ENTREZID      GOALL ONTOLOGYALL</span></span>
<span><span class="co">## 1 100037403       &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">## 2 100037404 GO:0016339          BP</span></span>
<span><span class="co">## 3 100037404 GO:0050900          BP</span></span>
<span><span class="co">## 4 100037404 GO:0007155          BP</span></span>
<span><span class="co">## 5 100037404 GO:0008150          BP</span></span>
<span><span class="co">## 6 100037404 GO:0009987          BP</span></span></code></pre>
<p>You may need to clean the mapping table:</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="va">tb</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">tb</span><span class="op">$</span><span class="va">GOALL</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<p>With an <code>OrgDb</code> object, you can also use the helper function <code><a href="../reference/get_GO_gene_sets_from_orgdb.html">get_GO_gene_sets_from_orgdb()</a></code> in <strong>GSEAtraining</strong> package:</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/GSEAtraining" class="external-link">GSEAtraining</a></span><span class="op">)</span></span>
<span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_GO_gene_sets_from_orgdb.html">get_GO_gene_sets_from_orgdb</a></span><span class="op">(</span><span class="va">org_db</span>, <span class="st">"BP"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`GO:0000002`</span></span>
<span><span class="co">## [1] "101081509" "101083159" "101083771" "101086314" "101086515" "101086671"</span></span>
<span><span class="co">## [7] "101088958" "101093964" "101097343"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>Try to obtain GO (BP) gene sets for dolphin (latin name: Tursiops truncatus) from <strong>BioMartGOGeneSets</strong> and <strong>AnnotationHub</strong>. Compare the difference of the GO gene sets from these two sources, e.g. compare the number of annotated genes from the two sources.</p>
</div>
<div class="section level4">
<h4 id="solution">Solution<a class="anchor" aria-label="anchor" href="#solution"></a>
</h4>
<p>Get GO gene sets from biomart. First search the “dataset name” with <code><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/supportedOrganisms.html" class="external-link">supportedOrganisms()</a></code>.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets</a></span><span class="op">(</span><span class="st">"ttruncatus_gene_ensembl"</span>, <span class="st">"BP"</span>, gene_id_type <span class="op">=</span> <span class="st">"entrez_id"</span><span class="op">)</span></span></code></pre></div>
<p>Get GO gene sets from NCBI/AnnotationHub.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Tursiops truncatus"</span>, <span class="st">"OrgDb"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AnnotationHub with 1 record</span></span>
<span><span class="co">## # snapshotDate(): 2023-04-25</span></span>
<span><span class="co">## # names(): AH112418</span></span>
<span><span class="co">## # $dataprovider: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/</span></span>
<span><span class="co">## # $species: Tursiops truncatus</span></span>
<span><span class="co">## # $rdataclass: OrgDb</span></span>
<span><span class="co">## # $rdatadateadded: 2023-04-24</span></span>
<span><span class="co">## # $title: org.Tursiops_truncatus.eg.sqlite</span></span>
<span><span class="co">## # $description: NCBI gene ID based annotations about Tursiops truncatus</span></span>
<span><span class="co">## # $taxonomyid: 9739</span></span>
<span><span class="co">## # $genome: NCBI genomes</span></span>
<span><span class="co">## # $sourcetype: NCBI/UniProt</span></span>
<span><span class="co">## # $sourceurl: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/, ftp://ftp.uniprot.org/p...</span></span>
<span><span class="co">## # $sourcesize: NA</span></span>
<span><span class="co">## # $tags: c("NCBI", "Gene", "Annotation") </span></span>
<span><span class="co">## # retrieve record with 'object[["AH112418"]]'</span></span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">orgdb</span> <span class="op">=</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH112418"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs2</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_GO_gene_sets_from_orgdb.html">get_GO_gene_sets_from_orgdb</a></span><span class="op">(</span><span class="va">orgdb</span>, <span class="st">"BP"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cn</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gs1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gs2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gs1</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="va">n2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gs2</span>, <span class="va">length</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">n1</span><span class="op">[</span><span class="va">cn</span><span class="op">]</span>, <span class="va">n2</span><span class="op">[</span><span class="va">cn</span><span class="op">]</span>, xlab <span class="op">=</span> <span class="st">"getBioMartGOGeneSets"</span>, ylab <span class="op">=</span> <span class="st">"AnnotationHub"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_07_more_gene_sets_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>Actually, the GO gene sets from <code><a href="https://rdrr.io/pkg/BioMartGOGeneSets/man/getBioMartGOGeneSets.html" class="external-link">getBioMartGOGeneSets()</a></code>/Ensembl is from an older genome build for dolphin. So it is more suggested to use the data from AnnotationHub/NCBI.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
