<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 5-01: Similarities of terms â€¢ GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 5-01: Similarities of terms">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 5-01: Similarities of terms</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2025-03-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic5_01_similarity.Rmd" class="external-link"><code>vignettes/topic5_01_similarity.Rmd</code></a></small>
      <div class="hidden name"><code>topic5_01_similarity.Rmd</code></div>

    </div>

    
    
<p>There are two types of methods to calculate similarities between functional terms.</p>
<ul>
<li>purely based on the number of member genes.</li>
<li>based on the topology of the ontology.</li>
</ul>
<p>We demonstrate it with significant terms from an ORA analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/GSEAtraining" class="external-link">GSEAtraining</a></span><span class="op">)</span></span>
<span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ora.rds"</span>, package <span class="op">=</span> <span class="st">"GSEAtraining"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">diff_gene</span> <span class="op">=</span> <span class="va">lt</span><span class="op">$</span><span class="va">diff_gene</span></span>
<span><span class="va">diff_gene</span> <span class="op">=</span> <span class="fu"><a href="../reference/convert_to_entrez_id.html">convert_to_entrez_id</a></span><span class="op">(</span><span class="va">diff_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">##   gene id might be SYMBOL (p =  0.980 )</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="section level2">
<h2 id="similarity-based-on-gene-counts">Similarity based on gene counts<a class="anchor" aria-label="anchor" href="#similarity-based-on-gene-counts"></a>
</h2>
<p>First we use a small collection of gene sets, the KEGG pathways.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/link/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"hsa:"</span>, <span class="st">""</span>, <span class="va">df</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"path:"</span>,  <span class="st">""</span>, <span class="va">df</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      V1       V2</span></span>
<span><span class="co">## 1 10327 hsa00010</span></span>
<span><span class="co">## 2   124 hsa00010</span></span>
<span><span class="co">## 3   125 hsa00010</span></span>
<span><span class="co">## 4   126 hsa00010</span></span>
<span><span class="co">## 5   127 hsa00010</span></span>
<span><span class="co">## 6   128 hsa00010</span></span></code></pre>
<p>We also retrieve the pathway names.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathway_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">pathway_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/structure.html" class="external-link">structure</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">" - Homo .*$"</span>, <span class="st">""</span>, <span class="va">pathway_names</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>, names <span class="op">=</span> <span class="va">pathway_names</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>Perform ORA analysis with <strong>clusterProfiler</strong>. There are 33 significant pathways.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## clusterProfiler v4.14.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Please cite:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.</span></span>
<span><span class="co">## clusterProfiler: an R package for comparing biological themes among</span></span>
<span><span class="co">## gene clusters. OMICS: A Journal of Integrative Biology. 2012,</span></span>
<span><span class="co">## 16(5):284-287</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'clusterProfiler'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html" class="external-link">enricher</a></span><span class="op">(</span><span class="va">diff_gene</span>, TERM2GENE <span class="op">=</span> <span class="va">df</span><span class="op">[</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">sig_pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span><span class="op">$</span><span class="va">ID</span></span>
<span></span>
<span><span class="va">gs_pathway</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">df</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">df</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">gs_pathway</span> <span class="op">=</span> <span class="va">gs_pathway</span><span class="op">[</span><span class="va">sig_pathways</span><span class="op">]</span></span></code></pre></div>
<p>There are four similarity methods which are based on the counts:</p>
<ul>
<li>Jaccard coeffcient</li>
<li>Dice coeffcient</li>
<li>Overlap coeffcient</li>
<li>Kappa coeffcient</li>
</ul>
<p>Since Dice coeffcient is just a simple transformation from the Jaccard coeffcient, we only look at the following three methods:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m1</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_pathway</span>, method <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span>
<span><span class="va">m2</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_pathway</span>, method <span class="op">=</span> <span class="st">"overlap"</span><span class="op">)</span></span>
<span><span class="va">m3</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_pathway</span>, method <span class="op">=</span> <span class="st">"kappa"</span><span class="op">)</span></span></code></pre></div>
<p>Values on diagonals of the three symmetric matrices are 1. Since other values in the matrices are much smaller than 1, we set the diagonals to NA to improve the heatmap visualization.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m2</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m3</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: grid</span></span></code></pre>
<pre><code><span><span class="co">## ========================================</span></span>
<span><span class="co">## ComplexHeatmap version 2.22.0</span></span>
<span><span class="co">## Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span><span class="co">## Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span><span class="co">## Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use it in published research, please cite either one:</span></span>
<span><span class="co">## - Gu, Z. Complex Heatmap Visualization. iMeta 2022.</span></span>
<span><span class="co">## - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span><span class="co">##     genomic data. Bioinformatics 2016.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The new InteractiveComplexHeatmap package can directly export static </span></span>
<span><span class="co">## complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## This message can be suppressed by:</span></span>
<span><span class="co">##   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span><span class="co">## ========================================</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m1</span>, name <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>pathway_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_text.html" class="external-link">anno_text</a></span><span class="op">(</span><span class="va">pathway_names</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-6-1.png" width="960"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m2</span>, name <span class="op">=</span> <span class="st">"overlap"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>pathway_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_text.html" class="external-link">anno_text</a></span><span class="op">(</span><span class="va">pathway_names</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-6-2.png" width="960"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m3</span>, name <span class="op">=</span> <span class="st">"kappa"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>pathway_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_text.html" class="external-link">anno_text</a></span><span class="op">(</span><span class="va">pathway_names</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-6-3.png" width="960"></p>
<p>In previous examples, we use all genes in the gene sets. Next we restrict only to the DE genes.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m1</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_pathway</span>, method <span class="op">=</span> <span class="st">"jaccard"</span>, all <span class="op">=</span> <span class="va">diff_gene</span><span class="op">)</span></span>
<span><span class="va">m2</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_pathway</span>, method <span class="op">=</span> <span class="st">"overlap"</span>, all <span class="op">=</span> <span class="va">diff_gene</span><span class="op">)</span></span>
<span><span class="va">m3</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_pathway</span>, method <span class="op">=</span> <span class="st">"kappa"</span>, all <span class="op">=</span> <span class="va">diff_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m2</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m3</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m1</span>, name <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>pathway_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_text.html" class="external-link">anno_text</a></span><span class="op">(</span><span class="va">pathway_names</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The automatically generated colors map from the 1^st and 99^th of the</span></span>
<span><span class="co">## values in the matrix. There are outliers in the matrix whose patterns</span></span>
<span><span class="co">## might be hidden by this color mapping. You can manually set the color</span></span>
<span><span class="co">## to `col` argument.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Use `suppressMessages()` to turn off this message.</span></span></code></pre>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m2</span>, name <span class="op">=</span> <span class="st">"overlap"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>pathway_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_text.html" class="external-link">anno_text</a></span><span class="op">(</span><span class="va">pathway_names</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-7-2.png" width="960"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m3</span>, name <span class="op">=</span> <span class="st">"kappa"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>pathway_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_text.html" class="external-link">anno_text</a></span><span class="op">(</span><span class="va">pathway_names</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-7-3.png" width="960"></p>
<p>Next we use a much larger gene set collection: the GO gene sets:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:clusterProfiler':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rename</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'IRanges'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:clusterProfiler':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     slice</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'AnnotationDbi'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:clusterProfiler':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     select</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sig_go</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span><span class="op">$</span><span class="va">ID</span></span></code></pre></div>
<p>There are &gt; 800 significant GO BP terms.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs_bp</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_GO_gene_sets_from_orgdb.html">get_GO_gene_sets_from_orgdb</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs_bp</span> <span class="op">=</span> <span class="va">gs_bp</span><span class="op">[</span><span class="va">sig_go</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m1</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_bp</span>, method <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span>
<span><span class="va">m2</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_bp</span>, method <span class="op">=</span> <span class="st">"overlap"</span><span class="op">)</span></span>
<span><span class="va">m3</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_bp</span>, method <span class="op">=</span> <span class="st">"kappa"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m2</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m3</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m1</span>, name <span class="op">=</span> <span class="st">"jaccard"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The automatically generated colors map from the 1^st and 99^th of the</span></span>
<span><span class="co">## values in the matrix. There are outliers in the matrix whose patterns</span></span>
<span><span class="co">## might be hidden by this color mapping. You can manually set the color</span></span>
<span><span class="co">## to `col` argument.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Use `suppressMessages()` to turn off this message.</span></span></code></pre>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m2</span>, name <span class="op">=</span> <span class="st">"overlap"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-10-2.png" width="700"></p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m3</span>, name <span class="op">=</span> <span class="st">"kappa"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The automatically generated colors map from the 1^st and 99^th of the</span></span>
<span><span class="co">## values in the matrix. There are outliers in the matrix whose patterns</span></span>
<span><span class="co">## might be hidden by this color mapping. You can manually set the color</span></span>
<span><span class="co">## to `col` argument.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Use `suppressMessages()` to turn off this message.</span></span></code></pre>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-10-3.png" width="700"></p>
<p>Or only restrict in DE genes</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m1</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_bp</span>, method <span class="op">=</span> <span class="st">"jaccard"</span>, all <span class="op">=</span> <span class="va">diff_gene</span><span class="op">)</span></span>
<span><span class="va">m2</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_bp</span>, method <span class="op">=</span> <span class="st">"overlap"</span>, all <span class="op">=</span> <span class="va">diff_gene</span><span class="op">)</span></span>
<span><span class="va">m3</span> <span class="op">=</span> <span class="fu"><a href="../reference/similarity.html">term_similarity</a></span><span class="op">(</span><span class="va">gs_bp</span>, method <span class="op">=</span> <span class="st">"kappa"</span>, all <span class="op">=</span> <span class="va">diff_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m1</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m2</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m3</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m1</span>, name <span class="op">=</span> <span class="st">"jaccard"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The automatically generated colors map from the 1^st and 99^th of the</span></span>
<span><span class="co">## values in the matrix. There are outliers in the matrix whose patterns</span></span>
<span><span class="co">## might be hidden by this color mapping. You can manually set the color</span></span>
<span><span class="co">## to `col` argument.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Use `suppressMessages()` to turn off this message.</span></span></code></pre>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m2</span>, name <span class="op">=</span> <span class="st">"overlap"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m3</span>, name <span class="op">=</span> <span class="st">"kappa"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The automatically generated colors map from the 1^st and 99^th of the</span></span>
<span><span class="co">## values in the matrix. There are outliers in the matrix whose patterns</span></span>
<span><span class="co">## might be hidden by this color mapping. You can manually set the color</span></span>
<span><span class="co">## to `col` argument.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Use `suppressMessages()` to turn off this message.</span></span></code></pre>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-11-3.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="semantic-similarity-based-on-ontology-structures">Semantic similarity based on ontology structures<a class="anchor" aria-label="anchor" href="#semantic-similarity-based-on-ontology-structures"></a>
</h2>
<p>There are also two major types of methods for semantic similarities.</p>
<ul>
<li>based on external gene annotations</li>
<li>based on internal topology structure</li>
</ul>
<div class="section level3">
<h3 id="based-on-gene-annotation">Based on gene annotation<a class="anchor" aria-label="anchor" href="#based-on-gene-annotation"></a>
</h3>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simona" class="external-link">simona</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ========================================</span></span>
<span><span class="co">## simona version 1.4.0</span></span>
<span><span class="co">## Bioconductor page: http://bioconductor.org/packages/simona/</span></span>
<span><span class="co">## Github page: https://github.com/jokergoo/simona</span></span>
<span><span class="co">## Documentation: https://jokergoo.github.io/simona/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use it in published research, please cite:</span></span>
<span><span class="co">## Gu, Z. simona: a Comprehensive R package for Semantic Similarity </span></span>
<span><span class="co">##   Analysis on Bio-Ontologies. BMC Genomics, 2024.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## This message can be suppressed by:</span></span>
<span><span class="co">##   suppressPackageStartupMessages(library(simona))</span></span>
<span><span class="co">## ========================================</span></span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simona_opt</span><span class="op">$</span><span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span></span></code></pre></div>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dag</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/simona/man/create_ontology_DAG_from_GO_db.html" class="external-link">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span>, org_db <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<p>The <em>Sim_Lin_1998</em> is one of the classic semantic similarity method.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/simona/man/term_sim.html" class="external-link">term_sim</a></span><span class="op">(</span><span class="va">dag</span>, <span class="va">sig_go</span>, method <span class="op">=</span> <span class="st">"Sim_Lin_1998"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## remove 20 terms with no annotation.</span></span></code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m</span>, name <span class="op">=</span> <span class="st">"Sim_Lin_1998"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="purely-based-on-the-topology">Purely based on the topology<a class="anchor" aria-label="anchor" href="#purely-based-on-the-topology"></a>
</h3>
<p>The _Sim_WP_1994__ is one of the classic semantic similarity method.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/simona/man/term_sim.html" class="external-link">term_sim</a></span><span class="op">(</span><span class="va">dag</span>, <span class="va">sig_go</span>, method <span class="op">=</span> <span class="st">"Sim_WP_1994"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m</span>, name <span class="op">=</span> <span class="st">"Sim_WP_1994"</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    show_row_dend <span class="op">=</span> <span class="cn">FALSE</span>, show_column_dend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualize-the-ontology">Visualize the ontology<a class="anchor" aria-label="anchor" href="#visualize-the-ontology"></a>
</h3>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/simona/man/dag_viz.html" class="external-link">dag_circular_viz</a></span><span class="op">(</span><span class="va">dag</span>, highlight <span class="op">=</span> <span class="va">sig_go</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic5_01_similarity_files/figure-html/unnamed-chunk-16-1.png" width="960"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
