<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 2-03: Use clusterProfiler for ORA • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 2-03: Use clusterProfiler for ORA">
<meta property="og:description" content="GSEAtraining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 2-03: Use clusterProfiler for ORA</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2024-01-25</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic2_03_clusterProfiler_ora.Rmd" class="external-link"><code>vignettes/topic2_03_clusterProfiler_ora.Rmd</code></a></small>
      <div class="hidden name"><code>topic2_03_clusterProfiler_ora.Rmd</code></div>

    </div>

    
    
<p>Let’s start with a vector of DE genes:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ora.rds"</span>, package <span class="op">=</span> <span class="st">"GSEAtraining"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">diff_gene</span> <span class="op">=</span> <span class="va">lt</span><span class="op">$</span><span class="va">diff_gene</span></span>
<span><span class="va">bg_gene</span> <span class="op">=</span> <span class="va">lt</span><span class="op">$</span><span class="va">bg_gene</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">diff_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "FGR"    "NIPAL3" "LAP3"   "CASP10" "CAMKK1" "PRSS22"</span></span></code></pre>
<p>Load the <strong>clusterProfiler</strong> package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span></code></pre></div>
<p><strong>clusterProfiler</strong> supports several major gene ID types in the input, but it is suggested to use Entrez IDs as the input because it is the “central gene ID type” in many databases/datasets.</p>
<p>We convert <code>diff_gene</code> to Entrez IDs. Note some genes may be lost due to the conversion.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/GSEAtraining" class="external-link">GSEAtraining</a></span><span class="op">)</span></span>
<span><span class="va">diff_gene</span> <span class="op">=</span> <span class="fu"><a href="../reference/convert_to_entrez_id.html">convert_to_entrez_id</a></span><span class="op">(</span><span class="va">diff_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">##   gene id might be SYMBOL (p =  1.000 )</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">diff_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     FGR  NIPAL3    LAP3  CASP10  CAMKK1  PRSS22 </span></span>
<span><span class="co">##  "2268" "57185" "51056"   "843" "84254" "64063"</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">diff_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 837</span></span></code></pre>
<p>Next we perform ORA on different gene set sources.</p>
<div class="section level2">
<h2 id="go-enrichment">GO enrichment<a class="anchor" aria-label="anchor" href="#go-enrichment"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: package 'S4Vectors' was built under R version 4.3.2</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                                Description GeneRatio</span></span>
<span><span class="co">## GO:0001819 GO:0001819 positive regulation of cytokine production    79/777</span></span>
<span><span class="co">## GO:0031349 GO:0031349    positive regulation of defense response    68/777</span></span>
<span><span class="co">## GO:0002237 GO:0002237   response to molecule of bacterial origin    61/777</span></span>
<span><span class="co">## GO:0002274 GO:0002274               myeloid leukocyte activation    49/777</span></span>
<span><span class="co">## GO:0032496 GO:0032496             response to lipopolysaccharide    59/777</span></span>
<span><span class="co">## GO:0019221 GO:0019221        cytokine-mediated signaling pathway    71/777</span></span>
<span><span class="co">##              BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## GO:0001819 489/18614 1.285130e-25 6.579865e-22 5.102642e-22</span></span>
<span><span class="co">## GO:0031349 441/18614 5.345810e-21 1.368527e-17 1.061284e-17</span></span>
<span><span class="co">## GO:0002237 366/18614 1.127385e-20 1.375774e-17 1.066904e-17</span></span>
<span><span class="co">## GO:0002274 240/18614 1.151399e-20 1.375774e-17 1.066904e-17</span></span>
<span><span class="co">## GO:0032496 345/18614 1.343530e-20 1.375774e-17 1.066904e-17</span></span>
<span><span class="co">## GO:0019221 492/18614 3.280167e-20 2.799076e-17 2.170665e-17</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                        geneID</span></span>
<span><span class="co">## GO:0001819 2268/4843/7305/6556/3082/11119/50943/8692/5743/3595/2219/58484/54209/5008/50487/51311/4210/80381/5054/59341/55801/3458/942/7474/3552/6772/9173/8809/22954/8013/29126/1958/729230/3659/3553/2867/10288/3620/1116/948/3569/7097/6653/834/116/5795/115362/26060/2207/10417/4332/5196/6850/26253/64127/3965/5724/136/929/1051/64581/10855/6288/4023/8877/247/353514/1604/246778/728/2214/3133/3134/93978/150372/11027/4321/338382/6348</span></span>
<span><span class="co">## GO:0031349                                                    7305/8692/3430/5743/2219/58484/54209/5008/50487/3055/51311/730249/4210/5054/59341/3458/7474/57142/9173/8807/729230/81030/3659/3553/2867/5819/3620/84034/8638/948/3569/7097/834/83666/84166/8547/6279/64332/7941/2209/115362/26060/1520/6280/6283/4332/6850/26253/64127/11314/136/929/2358/1051/64581/4023/353514/5320/5359/145741/3133/3134/388125/93978/11027/4321/338382/6348</span></span>
<span><span class="co">## GO:0002237                                                                                                               4843/6401/6556/50943/22904/5743/2920/3595/54209/3055/54/730249/5054/1440/6347/1594/57379/6648/942/7474/3552/29126/6590/6372/3553/10288/3620/948/3569/7097/10068/834/4283/6279/64332/133/10417/249/6280/9076/2921/6374/5196/2919/80149/717/64127/3965/3627/6373/5724/3576/2353/929/1051/9516/1604/728/6891/11027/6348</span></span>
<span><span class="co">## GO:0002274                                                                                                                                                                 2268/7305/6556/22904/8692/3430/54209/50487/3696/3458/7474/9173/8807/8013/729230/3598/55509/6372/30817/5819/8291/3569/7097/2207/6283/80149/81501/6850/338339/11314/116071/6338/3965/5724/3576/136/712/8877/2624/3579/2242/5320/5359/728/2214/57126/10537/11027/6348</span></span>
<span><span class="co">## GO:0032496                                                                                                                        4843/6401/6556/50943/22904/5743/2920/3595/54209/3055/54/730249/5054/1440/6347/1594/57379/6648/942/7474/3552/29126/6590/6372/3553/10288/3620/948/3569/7097/10068/834/4283/6279/64332/133/10417/249/6280/9076/2921/6374/5196/2919/80149/717/64127/3965/3627/6373/5724/3576/2353/929/1051/9516/1604/11027/6348</span></span>
<span><span class="co">## GO:0019221                                              608/51208/2920/3595/54209/5008/3055/11024/1440/6354/6347/3458/7474/3552/6772/7850/9173/8809/8807/22954/1441/1958/729230/3598/81030/6372/3659/3553/51554/10288/3601/8638/3569/3557/834/83666/4283/50506/84166/79092/10410/64332/5795/2180/26060/2207/8986/3577/2921/6374/5196/2919/6850/3429/3627/6373/3576/54625/8877/1237/3579/10581/8519/7293/8784/353514/11027/4321/6351/6349/6348</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:0001819    79</span></span>
<span><span class="co">## GO:0031349    68</span></span>
<span><span class="co">## GO:0002237    61</span></span>
<span><span class="co">## GO:0002274    49</span></span>
<span><span class="co">## GO:0032496    59</span></span>
<span><span class="co">## GO:0019221    71</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="kegg-enrichment">KEGG enrichment<a class="anchor" aria-label="anchor" href="#kegg-enrichment"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichKEGG.html" class="external-link">enrichKEGG</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, organism <span class="op">=</span> <span class="st">"hsa"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                ID                                                   Description</span></span>
<span><span class="co">## hsa04060 hsa04060                        Cytokine-cytokine receptor interaction</span></span>
<span><span class="co">## hsa04061 hsa04061 Viral protein interaction with cytokine and cytokine receptor</span></span>
<span><span class="co">## hsa04657 hsa04657                                       IL-17 signaling pathway</span></span>
<span><span class="co">## hsa04380 hsa04380                                    Osteoclast differentiation</span></span>
<span><span class="co">## hsa05323 hsa05323                                          Rheumatoid arthritis</span></span>
<span><span class="co">## hsa04668 hsa04668                                         TNF signaling pathway</span></span>
<span><span class="co">##          GeneRatio  BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## hsa04060    50/452 297/8662 7.690187e-14 2.399338e-11 2.072303e-11</span></span>
<span><span class="co">## hsa04061    27/452 100/8662 6.100715e-13 9.517115e-11 8.219910e-11</span></span>
<span><span class="co">## hsa04657    25/452  94/8662 6.469192e-12 6.727960e-10 5.810923e-10</span></span>
<span><span class="co">## hsa04380    29/452 135/8662 4.113147e-11 3.208254e-09 2.770962e-09</span></span>
<span><span class="co">## hsa05323    23/452  93/8662 2.278678e-10 1.421895e-08 1.228088e-08</span></span>
<span><span class="co">## hsa04668    25/452 114/8662 5.910829e-10 3.073631e-08 2.654688e-08</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                   geneID</span></span>
<span><span class="co">## hsa04060 53832/608/2920/3595/3589/5008/1440/6354/6347/55801/3458/3552/7850/9173/8809/8807/1441/729230/3624/3598/6372/3553/51554/9518/3601/3569/3557/4283/94/56477/58191/8741/3577/2921/6374/5196/2919/3627/6373/3576/8794/1237/3579/7293/8784/246778/6351/9560/6349/6348</span></span>
<span><span class="co">## hsa04061                                                                                                                     53832/2920/6354/6347/8809/8807/729230/6372/51554/3569/4283/56477/3577/2921/6374/5196/2919/3627/6373/3576/8794/1237/3579/6351/9560/6349/6348</span></span>
<span><span class="co">## hsa04657                                                                                                                                  5743/2920/1440/6354/6347/3458/727897/6372/3553/2354/3569/4322/6279/3934/4314/6280/2921/6374/2919/3627/3576/2353/1051/8061/4312</span></span>
<span><span class="co">## hsa04380                                                                                                       7305/54209/10326/54/11024/3458/3552/6772/2274/4688/8503/3553/2354/10288/7006/2212/2209/2215/6850/2353/126014/3726/8061/9021/353514/2214/11025/11027/79168</span></span>
<span><span class="co">## hsa05323                                                                                                                                                2920/3383/3589/54/6347/10312/3458/942/3552/6372/3553/3569/7097/4314/8741/2921/6374/2919/3576/2353/4312/6349/6348</span></span>
<span><span class="co">## hsa04668                                                                                                                                  843/6401/5743/2920/3383/6347/8809/8503/6372/3659/3553/3569/4314/4323/8986/2921/6374/2919/64127/197259/3627/2353/3726/1051/9021</span></span>
<span><span class="co">##          Count</span></span>
<span><span class="co">## hsa04060    50</span></span>
<span><span class="co">## hsa04061    27</span></span>
<span><span class="co">## hsa04657    25</span></span>
<span><span class="co">## hsa04380    29</span></span>
<span><span class="co">## hsa05323    23</span></span>
<span><span class="co">## hsa04668    25</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="reactome-enrichment">Reactome enrichment<a class="anchor" aria-label="anchor" href="#reactome-enrichment"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">ReactomePA</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ReactomePA v1.44.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use ReactomePA in published research, please cite:</span></span>
<span><span class="co">## Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization. Molecular BioSystems 2016, 12(2):477-479</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html" class="external-link">enrichPathway</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, organism <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                          ID</span></span>
<span><span class="co">## R-HSA-6798695 R-HSA-6798695</span></span>
<span><span class="co">## R-HSA-6783783 R-HSA-6783783</span></span>
<span><span class="co">## R-HSA-380108   R-HSA-380108</span></span>
<span><span class="co">## R-HSA-449147   R-HSA-449147</span></span>
<span><span class="co">## R-HSA-198933   R-HSA-198933</span></span>
<span><span class="co">## R-HSA-6785807 R-HSA-6785807</span></span>
<span><span class="co">##                                                                            Description</span></span>
<span><span class="co">## R-HSA-6798695                                                 Neutrophil degranulation</span></span>
<span><span class="co">## R-HSA-6783783                                                 Interleukin-10 signaling</span></span>
<span><span class="co">## R-HSA-380108                                       Chemokine receptors bind chemokines</span></span>
<span><span class="co">## R-HSA-449147                                                 Signaling by Interleukins</span></span>
<span><span class="co">## R-HSA-198933  Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell</span></span>
<span><span class="co">## R-HSA-6785807                               Interleukin-4 and Interleukin-13 signaling</span></span>
<span><span class="co">##               GeneRatio   BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## R-HSA-6798695    72/592 481/10955 1.196852e-15 1.345261e-12 1.277482e-12</span></span>
<span><span class="co">## R-HSA-6783783    21/592  47/10955 5.765865e-15 3.240416e-12 3.077151e-12</span></span>
<span><span class="co">## R-HSA-380108     22/592  59/10955 1.224584e-13 4.588108e-11 4.356940e-11</span></span>
<span><span class="co">## R-HSA-449147     64/592 473/10955 5.748719e-12 1.615390e-09 1.534000e-09</span></span>
<span><span class="co">## R-HSA-198933     28/592 134/10955 4.464142e-10 1.003539e-07 9.529767e-08</span></span>
<span><span class="co">## R-HSA-6785807    23/592 108/10955 1.113243e-08 2.085475e-06 1.980401e-06</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                          geneID</span></span>
<span><span class="co">## R-HSA-6798695 2268/64386/5329/7305/55/6556/2219/4680/4069/5836/10326/10562/3614/10493/10312/5768/5328/10549/7130/6590/27180/10288/1116/6947/948/10970/7097/6386/53831/2212/6279/3934/5795/2207/3101/2215/5284/1520/6282/6280/6283/3577/4332/2919/1508/338339/11314/6036/5724/929/126014/1084/2358/2357/116844/160364/10855/3310/4973/196527/3579/222487/199675/2204/1604/728/3107/11025/57126/3106/150372/81567</span></span>
<span><span class="co">## R-HSA-6783783                                                                                                                                                                                                                                                                                         5743/2920/3383/7076/1440/6347/942/3552/7850/729230/3553/3569/3557/2919/3627/5724/3576/2357/6351/6349/6348</span></span>
<span><span class="co">## R-HSA-380108                                                                                                                                                                                                                                                                                 2920/6354/6347/729230/6372/51554/4283/56477/58191/3577/2921/6374/5196/2919/3627/6373/3576/1237/3579/6351/6349/6348</span></span>
<span><span class="co">## R-HSA-449147                                                                4843/53832/3082/6196/5743/2920/3595/3383/3589/5008/3055/7076/1440/6347/595/55801/3458/6648/942/3552/6772/7850/9173/8809/8807/8503/1441/729230/3598/3553/759/3601/948/7006/3569/3557/10068/834/1848/84166/3934/4314/6283/2919/6850/64127/3965/3627/5724/3576/2353/2357/3726/6288/9021/4582/4312/246778/5696/5699/5698/6351/6349/6348</span></span>
<span><span class="co">## R-HSA-198933                                                                                                                                                                                                                                       7305/57823/1308/29992/3383/54209/11024/54210/27180/5819/10288/1280/2209/910/10871/11314/27036/126014/8519/342510/353514/2214/3107/11025/3133/3134/3106/11027</span></span>
<span><span class="co">## R-HSA-6785807                                                                                                                                                                                                                                                                                  4843/3082/5743/3383/5008/7076/6347/595/3552/6772/3598/3553/948/3569/3934/4314/3576/2353/3726/6288/9021/4582/4312</span></span>
<span><span class="co">##               Count</span></span>
<span><span class="co">## R-HSA-6798695    72</span></span>
<span><span class="co">## R-HSA-6783783    21</span></span>
<span><span class="co">## R-HSA-380108     22</span></span>
<span><span class="co">## R-HSA-449147     64</span></span>
<span><span class="co">## R-HSA-198933     28</span></span>
<span><span class="co">## R-HSA-6785807    23</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="do-enrichment">DO enrichment<a class="anchor" aria-label="anchor" href="#do-enrichment"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">DOSE</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DOSE v3.26.2  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use DOSE in published research, please cite:</span></span>
<span><span class="co">## Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/enrichDO.html" class="external-link">enrichDO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                  ID                     Description GeneRatio   BgRatio</span></span>
<span><span class="co">## DOID:824   DOID:824                   periodontitis    44/569 147/10312</span></span>
<span><span class="co">## DOID:3388 DOID:3388             periodontal disease    47/569 169/10312</span></span>
<span><span class="co">## DOID:403   DOID:403                   mouth disease    58/569 266/10312</span></span>
<span><span class="co">## DOID:104   DOID:104    bacterial infectious disease    68/569 359/10312</span></span>
<span><span class="co">## DOID:974   DOID:974 upper respiratory tract disease    49/569 202/10312</span></span>
<span><span class="co">## DOID:552   DOID:552                       pneumonia    44/569 164/10312</span></span>
<span><span class="co">##                 pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## DOID:824  2.827079e-21 2.025173e-18 1.005913e-18</span></span>
<span><span class="co">## DOID:3388 3.675451e-21 2.025173e-18 1.005913e-18</span></span>
<span><span class="co">## DOID:403  3.669020e-20 1.347753e-17 6.694352e-18</span></span>
<span><span class="co">## DOID:104  6.427655e-20 1.770819e-17 8.795738e-18</span></span>
<span><span class="co">## DOID:974  3.074551e-19 6.077593e-17 3.018768e-17</span></span>
<span><span class="co">## DOID:552  3.309034e-19 6.077593e-17 3.018768e-17</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                            geneID</span></span>
<span><span class="co">## DOID:824                                                                                                                        4843/6401/50943/5743/2920/3595/4069/3589/5008/7076/10135/5054/9126/6354/6347/3458/3552/7850/29126/729230/54210/6372/3553/3569/3557/7097/4322/2212/6279/133/4314/10630/2215/9076/64127/3576/929/2357/1051/3579/4312/2214/3106/4321</span></span>
<span><span class="co">## DOID:3388                                                                                                        4843/6401/3082/50943/5743/2920/3595/5243/4069/3589/5008/7076/10135/5054/9126/6354/6347/3458/3552/7850/29126/729230/5328/54210/6372/3553/3569/3557/7097/4322/2212/6279/133/4314/10630/2215/9076/64127/3576/929/2357/1051/3579/4312/2214/3106/4321</span></span>
<span><span class="co">## DOID:403                                                4843/6401/3082/50943/5743/2920/3595/5243/4069/3589/5008/7076/4210/10135/5054/9126/6354/6347/54757/3458/7474/3552/7850/29126/729230/5328/6590/6615/54210/6372/3553/80326/10970/3569/3557/7097/4322/2212/6279/3934/133/4314/10630/249/2215/9076/6469/64127/3576/929/2357/1051/3579/4312/7052/2214/3106/4321</span></span>
<span><span class="co">## DOID:104  4843/6401/5329/6556/50943/5743/5243/3383/54209/5008/7076/4210/5054/6347/7450/3458/1839/942/3552/6772/29126/729230/54210/3659/3553/3620/948/3569/3557/7097/4322/834/4283/50506/2212/3934/133/4314/2209/910/58191/2215/467/6469/64127/6890/3627/6373/5724/3576/929/2358/6288/9332/3579/338442/9021/4582/4312/246778/728/5696/6891/3106/5698/629/6351/6348</span></span>
<span><span class="co">## DOID:974                                                                                                    4843/5329/50943/5243/3383/4210/5054/6354/3458/942/3552/9173/727897/729230/5328/3598/6372/3553/3569/3557/7097/834/3067/2212/4585/3486/133/2209/6374/2919/64127/6890/3627/3576/929/4023/627/2205/3579/729238/4582/7293/728/2214/6891/3107/3106/629/6351</span></span>
<span><span class="co">## DOID:552                                                                                                                        5743/2920/3383/3589/5054/1440/6354/6347/595/4891/3458/3552/9173/727897/29126/5328/3598/6590/54210/3553/3569/3557/7097/4316/4283/3486/10630/2921/6374/2919/6469/64127/6890/3965/3576/929/627/9332/729238/4582/4312/57016/2214/6351</span></span>
<span><span class="co">##           Count</span></span>
<span><span class="co">## DOID:824     44</span></span>
<span><span class="co">## DOID:3388    47</span></span>
<span><span class="co">## DOID:403     58</span></span>
<span><span class="co">## DOID:104     68</span></span>
<span><span class="co">## DOID:974     49</span></span>
<span><span class="co">## DOID:552     44</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="msigdb-enrichment">MSigDB enrichment<a class="anchor" aria-label="anchor" href="#msigdb-enrichment"></a>
</h2>
<p>There is no built-in function specific for MSigDB gene sets, but there is a universal function <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html" class="external-link">enricher()</a></code> which accepts manually-specified gene sets. The gene sets object is simply a two-column data frame:</p>
<ul>
<li>the first column is the gene set ID</li>
<li>the second column is the gene ID</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span>
<span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="va">map</span> <span class="op">=</span> <span class="va">gene_sets</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"entrez_gene"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">map</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">##   gs_name               entrez_gene</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> HALLMARK_ADIPOGENESIS          19</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> HALLMARK_ADIPOGENESIS       <span style="text-decoration: underline;">11</span>194</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> HALLMARK_ADIPOGENESIS       <span style="text-decoration: underline;">10</span>449</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> HALLMARK_ADIPOGENESIS          33</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> HALLMARK_ADIPOGENESIS          34</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> HALLMARK_ADIPOGENESIS          35</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html" class="external-link">enricher</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, TERM2GENE <span class="op">=</span> <span class="va">map</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                    ID</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE HALLMARK_INTERFERON_GAMMA_RESPONSE</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE         HALLMARK_INFLAMMATORY_RESPONSE</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE HALLMARK_INTERFERON_ALPHA_RESPONSE</span></span>
<span><span class="co">## HALLMARK_TNFA_SIGNALING_VIA_NFKB     HALLMARK_TNFA_SIGNALING_VIA_NFKB</span></span>
<span><span class="co">## HALLMARK_COMPLEMENT                               HALLMARK_COMPLEMENT</span></span>
<span><span class="co">## HALLMARK_IL6_JAK_STAT3_SIGNALING     HALLMARK_IL6_JAK_STAT3_SIGNALING</span></span>
<span><span class="co">##                                                           Description GeneRatio</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE HALLMARK_INTERFERON_GAMMA_RESPONSE    55/347</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE         HALLMARK_INFLAMMATORY_RESPONSE    54/347</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE HALLMARK_INTERFERON_ALPHA_RESPONSE    33/347</span></span>
<span><span class="co">## HALLMARK_TNFA_SIGNALING_VIA_NFKB     HALLMARK_TNFA_SIGNALING_VIA_NFKB    49/347</span></span>
<span><span class="co">## HALLMARK_COMPLEMENT                               HALLMARK_COMPLEMENT    39/347</span></span>
<span><span class="co">## HALLMARK_IL6_JAK_STAT3_SIGNALING     HALLMARK_IL6_JAK_STAT3_SIGNALING    22/347</span></span>
<span><span class="co">##                                     BgRatio       pvalue     p.adjust</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE 200/4383 1.283631e-17 6.033064e-16</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE     200/4383 6.465947e-17 1.519498e-15</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE  97/4383 8.600159e-14 1.347358e-12</span></span>
<span><span class="co">## HALLMARK_TNFA_SIGNALING_VIA_NFKB   200/4383 1.359501e-13 1.597414e-12</span></span>
<span><span class="co">## HALLMARK_COMPLEMENT                200/4383 6.053975e-08 5.690736e-07</span></span>
<span><span class="co">## HALLMARK_IL6_JAK_STAT3_SIGNALING    87/4383 5.774922e-07 4.523689e-06</span></span>
<span><span class="co">##                                          qvalue</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE 4.594047e-16</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE     1.157064e-15</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE 1.025984e-12</span></span>
<span><span class="co">## HALLMARK_TNFA_SIGNALING_VIA_NFKB   1.216396e-12</span></span>
<span><span class="co">## HALLMARK_COMPLEMENT                4.333372e-07</span></span>
<span><span class="co">## HALLMARK_IL6_JAK_STAT3_SIGNALING   3.444690e-06</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                    geneID</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE 51056/57823/10797/3430/5743/3383/10135/6354/6347/6648/942/6772/29126/7130/81030/3659/3620/6737/94240/3601/8638/10906/3569/10068/834/10561/4283/7453/5371/84166/10410/2209/81894/9246/115361/3429/116071/6890/10791/3627/6373/2357/3669/54625/219285/9021/10581/7127/5359/5696/5699/10437/3106/5698/629</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE                                6401/5329/2769/490/3383/5008/7076/4210/366/3249/6324/10135/3696/5054/1440/6354/6347/4891/1839/3552/8809/8807/1441/3624/7130/3759/6372/3659/3553/2867/3601/3569/7097/9153/4283/7162/133/4323/10630/60675/64127/3627/6373/5724/3576/136/929/2357/4973/8877/8519/1604/3269/728</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE                                                                                                                  51056/3430/135112/3659/6737/94240/8638/10906/2766/834/10561/83666/7453/10410/81894/9246/2634/115361/3429/116071/6890/3627/6373/3669/54625/219285/10581/8519/5359/5696/3107/10437/5698</span></span>
<span><span class="co">## HALLMARK_TNFA_SIGNALING_VIA_NFKB                                                     5329/490/5743/2920/3383/10135/5054/6347/374/595/6648/1839/3552/5341/8013/1958/3624/5328/7130/50486/6372/3659/3553/2354/3601/3569/8870/7097/3491/4929/467/2921/2919/80149/6890/3627/6373/2353/3726/1051/4973/8061/8877/7262/9021/23764/7127/9516/6351</span></span>
<span><span class="co">## HALLMARK_COMPLEMENT                                                                                                                           51056/843/5329/2219/5251/7076/760/7980/5054/5341/725/3659/948/3569/4322/834/838/1848/7941/4323/2207/714/1520/6280/6283/2919/308/1508/717/255738/1051/712/4973/23764/5359/1604/5698/629/4321</span></span>
<span><span class="co">## HALLMARK_IL6_JAK_STAT3_SIGNALING                                                                                                                                                                                                6354/5967/6772/7850/8809/1441/3659/3553/3601/948/3569/7097/4283/94/2921/5196/2919/3627/6373/929/9021/5320</span></span>
<span><span class="co">##                                    Count</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE    55</span></span>
<span><span class="co">## HALLMARK_INFLAMMATORY_RESPONSE        54</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE    33</span></span>
<span><span class="co">## HALLMARK_TNFA_SIGNALING_VIA_NFKB      49</span></span>
<span><span class="co">## HALLMARK_COMPLEMENT                   39</span></span>
<span><span class="co">## HALLMARK_IL6_JAK_STAT3_SIGNALING      22</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="setting-background">Setting background<a class="anchor" aria-label="anchor" href="#setting-background"></a>
</h2>
<p>Note, in <strong>clusterProfiler</strong>, the background is intersect(all genes in the gene sets, user’s background genes).</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bg_gene</span> <span class="op">=</span> <span class="fu"><a href="../reference/convert_to_entrez_id.html">convert_to_entrez_id</a></span><span class="op">(</span><span class="va">bg_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   gene id might be SYMBOL (p =  0.990 )</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bg_gene</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">bg_gene</span>, <span class="fl">10000</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, universe <span class="op">=</span> <span class="va">bg_gene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                              Description GeneRatio</span></span>
<span><span class="co">## GO:0006954 GO:0006954                    inflammatory response    72/777</span></span>
<span><span class="co">## GO:0050764 GO:0050764               regulation of phagocytosis    18/777</span></span>
<span><span class="co">## GO:0071621 GO:0071621                   granulocyte chemotaxis    20/777</span></span>
<span><span class="co">## GO:0032496 GO:0032496           response to lipopolysaccharide    37/777</span></span>
<span><span class="co">## GO:0030593 GO:0030593                    neutrophil chemotaxis    18/777</span></span>
<span><span class="co">## GO:0002237 GO:0002237 response to molecule of bacterial origin    38/777</span></span>
<span><span class="co">##             BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## GO:0006954 418/8799 1.574385e-08 6.645479e-05 6.445035e-05</span></span>
<span><span class="co">## GO:0050764  51/8799 1.491784e-07 1.769292e-04 1.715926e-04</span></span>
<span><span class="co">## GO:0071621  62/8799 1.686835e-07 1.769292e-04 1.715926e-04</span></span>
<span><span class="co">## GO:0032496 170/8799 1.730502e-07 1.769292e-04 1.715926e-04</span></span>
<span><span class="co">## GO:0030593  52/8799 2.095821e-07 1.769292e-04 1.715926e-04</span></span>
<span><span class="co">## GO:0002237 181/8799 3.152967e-07 2.218113e-04 2.151209e-04</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                     geneID</span></span>
<span><span class="co">## GO:0006954 2268/4843/7305/6556/50943/22904/55737/7037/2920/54209/5008/25807/3055/54/5054/6354/6347/10312/59341/3552/7850/9173/729230/23780/9473/1116/9365/4907/3569/3557/8870/7097/834/64116/90527/64332/26060/56833/7791/2921/5196/80149/6850/64127/6338/3965/3627/6373/3576/136/2358/2357/1051/4973/6288/4023/8877/196527/9332/338557/7293/353514/2359/3269/728/145741/3133/388125/150372/6351/6349/6348</span></span>
<span><span class="co">## GO:0050764                                                                                                                                                                                                                                                                                                  2268/6556/54209/10326/3055/6347/3601/7097/8547/26060/2207/6850/717/64127/2358/8877/196527/7052</span></span>
<span><span class="co">## GO:0071621                                                                                                                                                                                                                                                                                            2920/6354/6347/59341/1441/2207/3577/2921/5196/6850/64127/3627/6373/3576/6288/3269/728/6351/6349/6348</span></span>
<span><span class="co">## GO:0032496                                                                                                                                                                                                    4843/6556/50943/22904/2920/3595/54209/3055/54/5054/1440/6347/1594/57379/6648/942/3552/29126/3569/7097/834/64332/10417/9076/2921/5196/80149/717/64127/3965/3627/6373/3576/1051/9516/1604/6348</span></span>
<span><span class="co">## GO:0030593                                                                                                                                                                                                                                                                                                       2920/6354/6347/1441/2207/3577/2921/5196/6850/64127/3627/6373/3576/6288/728/6351/6349/6348</span></span>
<span><span class="co">## GO:0002237                                                                                                                                                                                                4843/6556/50943/22904/2920/3595/54209/3055/54/5054/1440/6347/1594/57379/6648/942/3552/29126/3569/7097/834/64332/10417/9076/2921/5196/80149/717/64127/3965/3627/6373/3576/1051/9516/1604/728/6348</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:0006954    72</span></span>
<span><span class="co">## GO:0050764    18</span></span>
<span><span class="co">## GO:0071621    20</span></span>
<span><span class="co">## GO:0032496    37</span></span>
<span><span class="co">## GO:0030593    18</span></span>
<span><span class="co">## GO:0002237    38</span></span></code></pre>
<p>We can compare ORA with different backgrounds:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">tb2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, universe <span class="op">=</span> <span class="va">bg_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tb1</span> <span class="op">=</span> <span class="va">tb1</span><span class="op">@</span><span class="va">result</span></span>
<span><span class="va">tb2</span> <span class="op">=</span> <span class="va">tb2</span><span class="op">@</span><span class="va">result</span></span>
<span><span class="va">cn</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">tb1</span><span class="op">$</span><span class="va">ID</span>, <span class="va">tb2</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tb1</span><span class="op">[</span><span class="va">cn</span>, <span class="st">"pvalue"</span><span class="op">]</span>, <span class="va">tb2</span><span class="op">[</span><span class="va">cn</span>, <span class="st">"pvalue"</span><span class="op">]</span>, log <span class="op">=</span> <span class="st">"xy"</span>,</span>
<span>    xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1e-25</span>, <span class="fl">1</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1e-25</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"default background"</span>, ylab <span class="op">=</span> <span class="st">"self-defined background"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic2_03_clusterProfiler_ora_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>Or overlap between the significant GO terms:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jolars/eulerr" class="external-link">eulerr</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://jolars.github.io/eulerr/reference/euler.html" class="external-link">euler</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"default_background"</span> <span class="op">=</span> <span class="va">tb1</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">tb1</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span>,</span>
<span>                <span class="st">"bg_gene"</span> <span class="op">=</span> <span class="va">tb2</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">tb2</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    quantities <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic2_03_clusterProfiler_ora_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="example-for-other-organism">Example for other organism<a class="anchor" aria-label="anchor" href="#example-for-other-organism"></a>
</h2>
<ol style="list-style-type: decimal">
<li>GO enrichment</li>
</ol>
<p><code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO()</a></code> accepts an <code>OrgDb</code> object as the source of the GO gene sets. We take pig as an example. <code><a href="../reference/random_genes.html">random_genes()</a></code> is from <strong>GSEAtraining</strong> package.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Ss.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_gene</span> <span class="op">=</span> <span class="fu"><a href="../reference/random_genes.html">random_genes</a></span><span class="op">(</span><span class="va">org.Ss.eg.db</span>, <span class="fl">1000</span>, <span class="st">"ENTREZID"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<p>Here we set <code>pvalueCutoff = 1, qvalueCutoff = 1</code> because genes are random genes and it is expected there won’t be too much significant terms left.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org.Ss.eg.db</span>, </span>
<span>    pvalueCutoff <span class="op">=</span> <span class="fl">1</span>, qvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID</span></span>
<span><span class="co">## GO:1901019 GO:1901019</span></span>
<span><span class="co">## GO:0032413 GO:0032413</span></span>
<span><span class="co">## GO:0043112 GO:0043112</span></span>
<span><span class="co">## GO:1904063 GO:1904063</span></span>
<span><span class="co">## GO:0001881 GO:0001881</span></span>
<span><span class="co">## GO:0032410 GO:0032410</span></span>
<span><span class="co">##                                                              Description</span></span>
<span><span class="co">## GO:1901019  regulation of calcium ion transmembrane transporter activity</span></span>
<span><span class="co">## GO:0032413 negative regulation of ion transmembrane transporter activity</span></span>
<span><span class="co">## GO:0043112                                    receptor metabolic process</span></span>
<span><span class="co">## GO:1904063         negative regulation of cation transmembrane transport</span></span>
<span><span class="co">## GO:0001881                                            receptor recycling</span></span>
<span><span class="co">## GO:0032410                   negative regulation of transporter activity</span></span>
<span><span class="co">##            GeneRatio BgRatio       pvalue  p.adjust    qvalue</span></span>
<span><span class="co">## GO:1901019     6/322 20/7101 0.0001873522 0.1907751 0.1905821</span></span>
<span><span class="co">## GO:0032413     6/322 21/7101 0.0002523848 0.1907751 0.1905821</span></span>
<span><span class="co">## GO:0043112     6/322 22/7101 0.0003339304 0.1907751 0.1905821</span></span>
<span><span class="co">## GO:1904063     6/322 23/7101 0.0004347492 0.1907751 0.1905821</span></span>
<span><span class="co">## GO:0001881     5/322 16/7101 0.0005366868 0.1907751 0.1905821</span></span>
<span><span class="co">## GO:0032410     6/322 24/7101 0.0005578219 0.1907751 0.1905821</span></span>
<span><span class="co">##                                                                 geneID Count</span></span>
<span><span class="co">## GO:1901019    404772/100514123/100144505/100522131/100624857/100736934     6</span></span>
<span><span class="co">## GO:0032413    404772/100514123/100233201/100155879/100522131/100624857     6</span></span>
<span><span class="co">## GO:0043112 100155795/100513805/100513035/100153106/100737880/100154597     6</span></span>
<span><span class="co">## GO:1904063    404772/100514123/100233201/100155879/100522131/100624857     6</span></span>
<span><span class="co">## GO:0001881           100513805/100513035/100153106/100737880/100154597     5</span></span>
<span><span class="co">## GO:0032410    404772/100514123/100233201/100155879/100522131/100624857     6</span></span></code></pre>
<ol start="2" style="list-style-type: decimal">
<li>KEGG enrichment</li>
</ol>
<p>The KEGG code of a specific organism can be found at <a href="https://rest.kegg.jp/list/organism" class="external-link uri">https://rest.kegg.jp/list/organism</a></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichKEGG.html" class="external-link">enrichKEGG</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, organism <span class="op">=</span> <span class="st">"ssc"</span>, pvalueCutoff <span class="op">=</span> <span class="fl">1</span>, qvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                ID               Description GeneRatio  BgRatio     pvalue</span></span>
<span><span class="co">## ssc04720 ssc04720    Long-term potentiation     8/438  65/9240 0.01126773</span></span>
<span><span class="co">## ssc05417 ssc05417 Lipid and atherosclerosis    18/438 217/9240 0.01478149</span></span>
<span><span class="co">## ssc03050 ssc03050                Proteasome     6/438  46/9240 0.02058512</span></span>
<span><span class="co">## ssc05133 ssc05133                 Pertussis     8/438  74/9240 0.02331562</span></span>
<span><span class="co">## ssc04514 ssc04514   Cell adhesion molecules    13/438 155/9240 0.03241658</span></span>
<span><span class="co">## ssc02010 ssc02010          ABC transporters     6/438  51/9240 0.03252066</span></span>
<span><span class="co">##           p.adjust    qvalue</span></span>
<span><span class="co">## ssc04720 0.9692743 0.9692743</span></span>
<span><span class="co">## ssc05417 0.9692743 0.9692743</span></span>
<span><span class="co">## ssc03050 0.9692743 0.9692743</span></span>
<span><span class="co">## ssc05133 0.9692743 0.9692743</span></span>
<span><span class="co">## ssc04514 0.9692743 0.9692743</span></span>
<span><span class="co">## ssc02010 0.9692743 0.9692743</span></span>
<span><span class="co">##                                                                                                                                                                     geneID</span></span>
<span><span class="co">## ssc04720                                                                                         100516581/100515286/100153199/397184/100522174/100521797/733611/100519528</span></span>
<span><span class="co">## ssc05417 100513283/396925/100516581/100620416/100153199/100142665/397231/100037938/397184/100127478/397319/100156777/100739450/403103/100515857/751869/100156979/100519528</span></span>
<span><span class="co">## ssc03050                                                                                                                   497065/100155621/396975/733626/100153926/654294</span></span>
<span><span class="co">## ssc05133                                                                                               100520583/100037938/397319/100156777/751869/397503/396900/100519528</span></span>
<span><span class="co">## ssc04514                                             100620189/396925/100626921/397236/106504882/397231/100516913/100521039/100153273/100520970/100521555/100517508/397528</span></span>
<span><span class="co">## ssc02010                                                                                                          404699/100620416/110258083/100048963/100513231/100512445</span></span>
<span><span class="co">##          Count</span></span>
<span><span class="co">## ssc04720     8</span></span>
<span><span class="co">## ssc05417    18</span></span>
<span><span class="co">## ssc03050     6</span></span>
<span><span class="co">## ssc05133     8</span></span>
<span><span class="co">## ssc04514    13</span></span>
<span><span class="co">## ssc02010     6</span></span></code></pre>
<ol start="3" style="list-style-type: decimal">
<li>MSigDB</li>
</ol>
<p>Use <code><a href="https://igordot.github.io/msigdbr/reference/msigdbr_species.html" class="external-link">msigdbr::msigdbr_species()</a></code> to see which organisms are supported.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"pig"</span>, category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="va">map</span> <span class="op">=</span> <span class="va">gene_sets</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"entrez_gene"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">map</span><span class="op">$</span><span class="va">entrez_gene</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">map</span><span class="op">$</span><span class="va">entrez_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html" class="external-link">enricher</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, TERM2GENE <span class="op">=</span> <span class="va">map</span>, pvalueCutoff <span class="op">=</span> <span class="fl">1</span>, qvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                    ID</span></span>
<span><span class="co">## HALLMARK_HYPOXIA                                     HALLMARK_HYPOXIA</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE HALLMARK_INTERFERON_GAMMA_RESPONSE</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                       HALLMARK_MYC_TARGETS_V2</span></span>
<span><span class="co">## HALLMARK_DNA_REPAIR                               HALLMARK_DNA_REPAIR</span></span>
<span><span class="co">## HALLMARK_P53_PATHWAY                             HALLMARK_P53_PATHWAY</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE HALLMARK_INTERFERON_ALPHA_RESPONSE</span></span>
<span><span class="co">##                                                           Description GeneRatio</span></span>
<span><span class="co">## HALLMARK_HYPOXIA                                     HALLMARK_HYPOXIA    19/203</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE HALLMARK_INTERFERON_GAMMA_RESPONSE    15/203</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                       HALLMARK_MYC_TARGETS_V2     6/203</span></span>
<span><span class="co">## HALLMARK_DNA_REPAIR                               HALLMARK_DNA_REPAIR    11/203</span></span>
<span><span class="co">## HALLMARK_P53_PATHWAY                             HALLMARK_P53_PATHWAY    13/203</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE HALLMARK_INTERFERON_ALPHA_RESPONSE     7/203</span></span>
<span><span class="co">##                                     BgRatio      pvalue  p.adjust    qvalue</span></span>
<span><span class="co">## HALLMARK_HYPOXIA                   194/4191 0.002342544 0.1147847 0.1147847</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE 184/4191 0.031615932 0.6908646 0.6908646</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2             57/4191 0.055791459 0.6908646 0.6908646</span></span>
<span><span class="co">## HALLMARK_DNA_REPAIR                145/4191 0.091130973 0.6908646 0.6908646</span></span>
<span><span class="co">## HALLMARK_P53_PATHWAY               188/4191 0.121575318 0.6908646 0.6908646</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE  88/4191 0.132426113 0.6908646 0.6908646</span></span>
<span><span class="co">##                                                                                                                                                                                                geneID</span></span>
<span><span class="co">## HALLMARK_HYPOXIA                   110259313/100191062/448812/100154770/100156209/397273/397412/100522745/396677/397184/397159/106504129/100738344/100174961/100516442/110255858/494561/397528/406870</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE                                   396925/396975/100302538/100626247/100514123/100171401/733626/100621776/397319/100156777/397683/654294/751869/100519985/110255948</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                                                                                                                            595107/100512235/100329222/100155960/396953/494561</span></span>
<span><span class="co">## HALLMARK_DNA_REPAIR                                                                     100513720/100526021/100516955/100513971/100124384/100514398/100513197/100620801/100625874/100513963/100516196</span></span>
<span><span class="co">## HALLMARK_P53_PATHWAY                                                      100513283/100038006/100154770/102164801/100157103/397663/100625349/397319/100513197/100144589/100738912/100517839/102161427</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE                                                                                                          396975/100302538/100626247/397319/397683/654294/110255948</span></span>
<span><span class="co">##                                    Count</span></span>
<span><span class="co">## HALLMARK_HYPOXIA                      19</span></span>
<span><span class="co">## HALLMARK_INTERFERON_GAMMA_RESPONSE    15</span></span>
<span><span class="co">## HALLMARK_MYC_TARGETS_V2                6</span></span>
<span><span class="co">## HALLMARK_DNA_REPAIR                   11</span></span>
<span><span class="co">## HALLMARK_P53_PATHWAY                  13</span></span>
<span><span class="co">## HALLMARK_INTERFERON_ALPHA_RESPONSE     7</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="examples-for-less-well-studied-organisms">Examples for less well-studied organisms<a class="anchor" aria-label="anchor" href="#examples-for-less-well-studied-organisms"></a>
</h2>
<div class="section level3">
<h3 id="when-an-orgdb-object-is-avaiable">When an <code>OrgDb</code> object is avaiable<a class="anchor" aria-label="anchor" href="#when-an-orgdb-object-is-avaiable"></a>
</h3>
<p>You can use <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO()</a></code> as long as you have an <code>OrgDb</code> object:</p>
<p>Taking dolphin as an example.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: BiocFileCache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: dbplyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'AnnotationHub'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     cache</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ah</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">org_db</span> <span class="op">=</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH112418"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_gene</span> <span class="op">=</span> <span class="fu"><a href="../reference/random_genes.html">random_genes</a></span><span class="op">(</span><span class="va">org_db</span>, <span class="fl">1000</span>, <span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span><span class="va">diff_gene</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, OrgDb <span class="op">=</span> <span class="va">org_db</span>, </span>
<span>    pvalueCutoff <span class="op">=</span> <span class="fl">1</span>, qvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                                       Description</span></span>
<span><span class="co">## GO:1901575 GO:1901575               organic substance catabolic process</span></span>
<span><span class="co">## GO:0009056 GO:0009056                                 catabolic process</span></span>
<span><span class="co">## GO:0030163 GO:0030163                         protein catabolic process</span></span>
<span><span class="co">## GO:0016042 GO:0016042                           lipid catabolic process</span></span>
<span><span class="co">## GO:0051603 GO:0051603 proteolysis involved in protein catabolic process</span></span>
<span><span class="co">## GO:0009887 GO:0009887                        animal organ morphogenesis</span></span>
<span><span class="co">##            GeneRatio  BgRatio      pvalue  p.adjust    qvalue</span></span>
<span><span class="co">## GO:1901575    23/150 353/4458 0.001416669 0.6827016 0.6827016</span></span>
<span><span class="co">## GO:0009056    26/150 422/4458 0.001541087 0.6827016 0.6827016</span></span>
<span><span class="co">## GO:0030163    13/150 163/4458 0.003032529 0.8081516 0.8081516</span></span>
<span><span class="co">## GO:0016042     6/150  53/4458 0.008172339 0.8081516 0.8081516</span></span>
<span><span class="co">## GO:0051603    11/150 147/4458 0.010012634 0.8081516 0.8081516</span></span>
<span><span class="co">## GO:0009887     5/150  40/4458 0.010264565 0.8081516 0.8081516</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                         geneID</span></span>
<span><span class="co">## GO:1901575                               101321692/101330000/101321621/101329993/101337634/101328625/101327810/101321305/101330482/101330212/101333502/101335699/101325407/101332617/101327586/101322521/101329165/101328759/101322780/101331077/101324776/101334669/101329007</span></span>
<span><span class="co">## GO:0009056 101321692/101330000/101321621/101329993/101337634/101335932/101328625/101327810/101321305/101330482/101330212/101333502/101335699/101325407/101325691/101332617/101327586/101322521/101329165/101328759/101334905/101322780/101331077/101324776/101334669/101329007</span></span>
<span><span class="co">## GO:0030163                                                                                                                                   101321692/101321621/101329993/101328625/101330212/101335699/101325407/101332617/101327586/101322521/101329165/101331077/101329007</span></span>
<span><span class="co">## GO:0016042                                                                                                                                                                                                         101330000/101327810/101321305/101333502/101328759/101322780</span></span>
<span><span class="co">## GO:0051603                                                                                                                                                       101321692/101321621/101329993/101330212/101335699/101325407/101332617/101327586/101322521/101329165/101329007</span></span>
<span><span class="co">## GO:0009887                                                                                                                                                                                                                   101338909/101316954/101323188/101336945/101323943</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:1901575    23</span></span>
<span><span class="co">## GO:0009056    26</span></span>
<span><span class="co">## GO:0030163    13</span></span>
<span><span class="co">## GO:0016042     6</span></span>
<span><span class="co">## GO:0051603    11</span></span>
<span><span class="co">## GO:0009887     5</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="enrichkegg-also-supports-many-other-orgainsms">enrichKEGG() also supports many other orgainsms<a class="anchor" aria-label="anchor" href="#enrichkegg-also-supports-many-other-orgainsms"></a>
</h3>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichKEGG.html" class="external-link">enrichKEGG</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">diff_gene</span>, organism <span class="op">=</span> <span class="va">...</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="manually-construct-the-gene-sets">Manually construct the gene sets<a class="anchor" aria-label="anchor" href="#manually-construct-the-gene-sets"></a>
</h3>
<p>We have introduced many ways to obtain gene sets for less well-studies organisms. The only thing to do here is to convert gene sets to a two-column data frame where gene sets are in the first column and genes are in the second column. Then use <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html" class="external-link">enricher()</a></code>.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html" class="external-link">enricher</a></span><span class="op">(</span><span class="va">diff_gene</span>, TERM2GENE <span class="op">=</span> <span class="va">...</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="look-at-the-tb-object">Look at the <code>tb</code> object<a class="anchor" aria-label="anchor" href="#look-at-the-tb-object"></a>
</h2>
<p>We used the same variable name <code>tb</code> for the object returned by the various enrichment functions. They are all in the same format. It looks like a table, but be careful, it is actually not:</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "enrichResult"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "DOSE"</span></span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Formal class 'enrichResult' [package "DOSE"] with 15 slots</span></span>
<span><span class="co">##   ..@ result       :'data.frame':    886 obs. of  9 variables:</span></span>
<span><span class="co">##   .. ..$ ID         : chr [1:886] "GO:1901575" "GO:0009056" "GO:0030163" "GO:0016042" ...</span></span>
<span><span class="co">##   .. ..$ Description: chr [1:886] "organic substance catabolic process" "catabolic process" "protein catabolic process" "lipid catabolic process" ...</span></span>
<span><span class="co">##   .. ..$ GeneRatio  : chr [1:886] "23/150" "26/150" "13/150" "6/150" ...</span></span>
<span><span class="co">##   .. ..$ BgRatio    : chr [1:886] "353/4458" "422/4458" "163/4458" "53/4458" ...</span></span>
<span><span class="co">##   .. ..$ pvalue     : num [1:886] 0.00142 0.00154 0.00303 0.00817 0.01001 ...</span></span>
<span><span class="co">##   .. ..$ p.adjust   : num [1:886] 0.683 0.683 0.808 0.808 0.808 ...</span></span>
<span><span class="co">##   .. ..$ qvalue     : num [1:886] 0.683 0.683 0.808 0.808 0.808 ...</span></span>
<span><span class="co">##   .. ..$ geneID     : chr [1:886] "101321692/101330000/101321621/101329993/101337634/101328625/101327810/101321305/101330482/101330212/101333502/1"| __truncated__ "101321692/101330000/101321621/101329993/101337634/101335932/101328625/101327810/101321305/101330482/101330212/1"| __truncated__ "101321692/101321621/101329993/101328625/101330212/101335699/101325407/101332617/101327586/101322521/101329165/1"| __truncated__ "101330000/101327810/101321305/101333502/101328759/101322780" ...</span></span>
<span><span class="co">##   .. ..$ Count      : int [1:886] 23 26 13 6 11 5 4 14 6 3 ...</span></span>
<span><span class="co">##   ..@ pvalueCutoff : num 1</span></span>
<span><span class="co">##   ..@ pAdjustMethod: chr "BH"</span></span>
<span><span class="co">##   ..@ qvalueCutoff : num 1</span></span>
<span><span class="co">##   ..@ organism     : chr "Tursiops truncatus"</span></span>
<span><span class="co">##   ..@ ontology     : chr "BP"</span></span>
<span><span class="co">##   ..@ gene         : chr [1:1000] "101323162" "101338410" "101326117" "117307745" ...</span></span>
<span><span class="co">##   ..@ keytype      : chr "ENTREZID"</span></span>
<span><span class="co">##   ..@ universe     : chr [1:4458] "101319146" "101322275" "101323809" "101330306" ...</span></span>
<span><span class="co">##   ..@ gene2Symbol  : chr(0) </span></span>
<span><span class="co">##   ..@ geneSets     :List of 1397</span></span>
<span><span class="co">##   .. ..$ GO:0000003: chr [1:87] "101335701" "101315473" "101315779" "101315844" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000070: chr [1:26] "101315815" "101316537" "101316695" "101318454" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000075: chr [1:23] "101316695" "101316733" "101318513" "101318624" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000077: chr [1:14] "101316733" "101318513" "101319146" "101320252" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000082: chr [1:15] "101317019" "101319146" "101321188" "101322275" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000122: chr [1:35] "101316020" "101319446" "101320166" "101321309" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000165: chr [1:48] "101315703" "101315844" "101316074" "101316105" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000209: chr [1:33] "101315684" "101316554" "101319013" "101319502" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000226: chr [1:61] "101315537" "101316394" "101316518" "101317000" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000245: chr [1:9] "101321273" "101322182" "101327231" "101330237" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000266: chr [1:9] "101319797" "101320522" "101322601" "101323172" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000278: chr [1:95] "101315815" "101316323" "101316500" "101316537" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000280: chr [1:45] "101315815" "101316537" "101316695" "101316904" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000302: chr [1:15] "101316954" "101319583" "101321208" "101322317" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000375: chr [1:74] "101315667" "101316328" "101316554" "101317380" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000377: chr [1:74] "101315667" "101316328" "101316554" "101317380" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000387: chr [1:11] "101318020" "101318583" "101318698" "101319671" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000398: chr [1:74] "101315667" "101316328" "101316554" "101317380" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000724: chr [1:20] "101315841" "101315900" "101317036" "101319304" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000725: chr [1:21] "101315841" "101315900" "101317036" "101319304" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000819: chr [1:28] "101315815" "101316537" "101316695" "101318454" ...</span></span>
<span><span class="co">##   .. ..$ GO:0000902: chr [1:41] "101316391" "101316954" "101317123" "101317226" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001505: chr [1:9] "101317652" "101318996" "101320339" "101323784" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001525: chr [1:32] "101331207" "101317123" "101318052" "101319105" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001558: chr [1:27] "101315844" "101316274" "101318448" "101318509" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001568: chr [1:34] "101331207" "101315844" "101317123" "101318052" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001655: chr [1:3] "101315844" "101336945" "101338144"</span></span>
<span><span class="co">##   .. ..$ GO:0001659: chr [1:9] "101316053" "101318809" "101319101" "101322601" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001667: chr [1:30] "101337436" "101321208" "101321450" "101322398" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001701: chr [1:14] "101316053" "101319146" "101319991" "101322124" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001756: chr [1:6] "101316583" "101319146" "101323188" "101323484" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001775: chr [1:63] "101325136" "101336443" "101337436" "101338293" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001816: chr [1:52] "101318178" "101325136" "101328418" "101331207" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001817: chr [1:51] "101325136" "101328418" "101331207" "101335103" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001818: chr [1:17] "101331207" "101335103" "101337436" "101318757" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001821: chr "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0001890: chr [1:8] "101315779" "101317123" "101319991" "101322124" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001892: chr [1:4] "101319991" "101322124" "101325153" "101338909"</span></span>
<span><span class="co">##   .. ..$ GO:0001944: chr [1:36] "101331207" "101315844" "101317123" "101318052" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001959: chr [1:9] "101316698" "101319508" "101322100" "101322274" ...</span></span>
<span><span class="co">##   .. ..$ GO:0001960: chr [1:4] "101319508" "101322274" "101323172" "101336222"</span></span>
<span><span class="co">##   .. ..$ GO:0002009: chr [1:16] "101316583" "101318072" "101319991" "101321450" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002040: chr [1:10] "101320969" "101324963" "101326054" "101327652" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002042: chr [1:6] "101324963" "101326054" "101329517" "101333080" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002218: chr [1:31] "101325271" "101328418" "101316515" "101316747" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002220: chr [1:5] "101328418" "101319301" "101333343" "101336375" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002221: chr [1:26] "101325271" "101328418" "101316515" "101316747" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002237: chr [1:13] "101335103" "101317804" "101318757" "101319301" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002244: chr [1:12] "101315577" "101315602" "101316583" "101318719" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002252: chr [1:31] "101325136" "101337436" "101315662" "101317324" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002253: chr [1:45] "101325271" "101328418" "101315844" "101316515" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002263: chr [1:17] "101325136" "101337436" "101315662" "101318088" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002274: chr [1:16] "101337436" "101318080" "101318088" "101320899" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002275: chr [1:4] "101337436" "101318088" "101325904" "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002279: chr "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002349: chr "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002366: chr [1:16] "101325136" "101337436" "101315662" "101318088" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002376: chr [1:262] "101317873" "101318178" "101325136" "101325271" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002429: chr [1:15] "101328418" "101315844" "101318384" "101319301" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002441: chr "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002443: chr [1:15] "101319418" "101319446" "101319565" "101320115" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002444: chr [1:3] "101320910" "101326295" "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002448: chr "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002532: chr [1:4] "101319739" "101322970" "101329901" "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002553: chr "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0002682: chr [1:116] "101325136" "101325271" "101328418" "101336443" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002683: chr [1:27] "101318088" "101318384" "101318757" "101321302" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002684: chr [1:81] "101325136" "101325271" "101328418" "101336443" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002685: chr [1:17] "101315844" "101317019" "101318861" "101325904" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002688: chr [1:11] "101317019" "101318861" "101325904" "101326847" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002694: chr [1:37] "101325136" "101336443" "101337436" "101316111" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002695: chr [1:9] "101325073" "101329251" "101329691" "101332625" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002696: chr [1:24] "101325136" "101336443" "101337436" "101316111" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002697: chr [1:16] "101325136" "101337436" "101319418" "101319446" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002699: chr [1:13] "101325136" "101337436" "101319418" "101319446" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002703: chr [1:9] "101319418" "101319446" "101319565" "101320899" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002752: chr [1:5] "101328418" "101319301" "101333343" "101336375" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002757: chr [1:35] "101325271" "101328418" "101315844" "101316515" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002758: chr [1:26] "101325271" "101328418" "101316515" "101316747" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002764: chr [1:36] "101325271" "101328418" "101315844" "101316515" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002768: chr [1:16] "101328418" "101315844" "101318384" "101319301" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002831: chr [1:45] "101325271" "101328418" "101316515" "101316747" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002832: chr [1:11] "101318088" "101318757" "101322160" "101322274" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002833: chr [1:35] "101325271" "101328418" "101316515" "101316747" ...</span></span>
<span><span class="co">##   .. ..$ GO:0002886: chr "101335932"</span></span>
<span><span class="co">##   .. ..$ GO:0003002: chr [1:17] "101315844" "101316583" "101317452" "101319146" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003006: chr [1:49] "101335701" "101315473" "101315779" "101315844" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003007: chr [1:11] "101315844" "101316954" "101318912" "101319146" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003008: chr [1:111] "101316391" "101320374" "101323479" "101329622" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003012: chr [1:20] "101335701" "101316390" "101316752" "101316954" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003013: chr [1:36] "101331207" "101335701" "101316390" "101316752" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003015: chr [1:16] "101335701" "101316752" "101316954" "101316959" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003018: chr [1:18] "101316390" "101316838" "101316959" "101317132" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003205: chr [1:8] "101316954" "101318912" "101319146" "101319198" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003206: chr [1:6] "101316954" "101318912" "101319146" "101319198" ...</span></span>
<span><span class="co">##   .. ..$ GO:0003208: chr [1:4] "101316954" "101318912" "101319198" "101338002"</span></span>
<span><span class="co">##   .. ..$ GO:0003229: chr [1:4] "101316954" "101318912" "101324101" "101338002"</span></span>
<span><span class="co">##   .. ..$ GO:0003231: chr [1:7] "101316954" "101318912" "101319198" "101319991" ...</span></span>
<span><span class="co">##   .. ..$ GO:0005975: chr [1:128] "101315510" "101315634" "101316031" "101316078" ...</span></span>
<span><span class="co">##   .. .. [list output truncated]</span></span>
<span><span class="co">##   ..@ readable     : logi FALSE</span></span>
<span><span class="co">##   ..@ termsim      : num[0 , 0 ] </span></span>
<span><span class="co">##   ..@ method       : chr(0) </span></span>
<span><span class="co">##   ..@ dr           : list()</span></span></code></pre>
<p>To convert it into a “real” table, DO NOT use <code><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame()</a></code> which only returns the significant terms, use <code>tb@result</code>.</p>
<p>Also some columns in <code>tb@result</code> are not in the proper format, e.g. <code>"GeneRatio"</code> and <code>"BgRatio"</code> where numbers should be in numeric mode while not characters. In <strong>GSEAtraining</strong>, there is a <code><a href="../reference/add_more_columns.html">add_more_columns()</a></code> which adds more columns to the <code>tb@result</code> table.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb2</span> <span class="op">=</span> <span class="fu"><a href="../reference/add_more_columns.html">add_more_columns</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb2</span><span class="op">@</span><span class="va">result</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                                       Description</span></span>
<span><span class="co">## GO:1901575 GO:1901575               organic substance catabolic process</span></span>
<span><span class="co">## GO:0009056 GO:0009056                                 catabolic process</span></span>
<span><span class="co">## GO:0030163 GO:0030163                         protein catabolic process</span></span>
<span><span class="co">## GO:0016042 GO:0016042                           lipid catabolic process</span></span>
<span><span class="co">## GO:0051603 GO:0051603 proteolysis involved in protein catabolic process</span></span>
<span><span class="co">## GO:0009887 GO:0009887                        animal organ morphogenesis</span></span>
<span><span class="co">##            GeneRatio  BgRatio      pvalue  p.adjust    qvalue</span></span>
<span><span class="co">## GO:1901575    23/150 353/4458 0.001416669 0.6827016 0.6827016</span></span>
<span><span class="co">## GO:0009056    26/150 422/4458 0.001541087 0.6827016 0.6827016</span></span>
<span><span class="co">## GO:0030163    13/150 163/4458 0.003032529 0.8081516 0.8081516</span></span>
<span><span class="co">## GO:0016042     6/150  53/4458 0.008172339 0.8081516 0.8081516</span></span>
<span><span class="co">## GO:0051603    11/150 147/4458 0.010012634 0.8081516 0.8081516</span></span>
<span><span class="co">## GO:0009887     5/150  40/4458 0.010264565 0.8081516 0.8081516</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                         geneID</span></span>
<span><span class="co">## GO:1901575                               101321692/101330000/101321621/101329993/101337634/101328625/101327810/101321305/101330482/101330212/101333502/101335699/101325407/101332617/101327586/101322521/101329165/101328759/101322780/101331077/101324776/101334669/101329007</span></span>
<span><span class="co">## GO:0009056 101321692/101330000/101321621/101329993/101337634/101335932/101328625/101327810/101321305/101330482/101330212/101333502/101335699/101325407/101325691/101332617/101327586/101322521/101329165/101328759/101334905/101322780/101331077/101324776/101334669/101329007</span></span>
<span><span class="co">## GO:0030163                                                                                                                                   101321692/101321621/101329993/101328625/101330212/101335699/101325407/101332617/101327586/101322521/101329165/101331077/101329007</span></span>
<span><span class="co">## GO:0016042                                                                                                                                                                                                         101330000/101327810/101321305/101333502/101328759/101322780</span></span>
<span><span class="co">## GO:0051603                                                                                                                                                       101321692/101321621/101329993/101330212/101335699/101325407/101332617/101327586/101322521/101329165/101329007</span></span>
<span><span class="co">## GO:0009887                                                                                                                                                                                                                   101338909/101316954/101323188/101336945/101323943</span></span>
<span><span class="co">##            Count n_hits n_genes gs_size n_totle log2_fold_enrichment  z_score</span></span>
<span><span class="co">## GO:1901575    23     23     150     353    4458            0.9533998 3.420861</span></span>
<span><span class="co">## GO:0009056    26     26     150     422    4458            0.8727027 3.347787</span></span>
<span><span class="co">## GO:0030163    13     13     150     163    4458            1.2450738 3.325517</span></span>
<span><span class="co">## GO:0016042     6      6     150      53    4458            1.7504043 3.231012</span></span>
<span><span class="co">## GO:0051603    11     11     150     147    4458            1.1531215 2.815535</span></span>
<span><span class="co">## GO:0009887     5      5     150      40    4458            1.8933622 3.218223</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>Use the DE gene list in <code>webgestalt_example_gene_list.rds</code> to perform ORA on GO/KEGG/MSigDB gene sets.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"webgestalt_example_gene_list.rds"</span>, </span>
<span>    package <span class="op">=</span> <span class="st">"GSEAtraining"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
