<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 3-02: Local GREAT analysis • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 3-02: Local GREAT analysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 3-02: Local GREAT analysis</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2025-03-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic3_02_local_GREAT.Rmd" class="external-link"><code>vignettes/topic3_02_local_GREAT.Rmd</code></a></small>
      <div class="hidden name"><code>topic3_02_local_GREAT.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<p>First let’s load the <strong>rGREAT</strong> package and generate a set of random regions:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/rGREAT" class="external-link">rGREAT</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## ========================================</span></span>
<span><span class="co">## rGREAT version 2.8.0</span></span>
<span><span class="co">## Bioconductor page: http://bioconductor.org/packages/rGREAT/</span></span>
<span><span class="co">## Github page: https://github.com/jokergoo/rGREAT</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use it in published research, please cite:</span></span>
<span><span class="co">## Gu, Z. rGREAT: an R/Bioconductor package for functional enrichment </span></span>
<span><span class="co">##   on genomic regions. Bioinformatics 2023.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## This message can be suppressed by:</span></span>
<span><span class="co">##   suppressPackageStartupMessages(library(rGREAT))</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Note: On Aug 19 2019 GREAT released version 4 where it supports `hg38`</span></span>
<span><span class="co">## genome and removes some ontologies such pathways. `submitGreatJob()`</span></span>
<span><span class="co">## still takes `hg19` as default. `hg38` can be specified by the `genome</span></span>
<span><span class="co">## = 'hg38'` argument. To use the older versions such as 3.0.0, specify as</span></span>
<span><span class="co">## `submitGreatJob(..., version = '3.0.0')`.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## From rGREAT version 1.99.0, it also implements the GREAT algorithm and</span></span>
<span><span class="co">## it allows to integrate GREAT analysis with the Bioconductor annotation</span></span>
<span><span class="co">## ecosystem. By default it supports more than 500 organisms. Check the</span></span>
<span><span class="co">## new function `great()` and the new vignette.</span></span>
<span><span class="co">## ========================================</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">gr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/randomRegions.html" class="external-link">randomRegions</a></span><span class="op">(</span>nr <span class="op">=</span> <span class="fl">1000</span>, genome <span class="op">=</span> <span class="st">"hg19"</span><span class="op">)</span></span></code></pre></div>
<p>GREAT analysis needs two types of input data: the gene sets and TSS source. Thus the second argument in <code><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great()</a></code> corresponds to the gene sets, and the third argumetn corresponds to the source of TSS.</p>
<p><strong>rGREAT</strong> has integrated many gene sets and genomes.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"MSigDB:H"</span>, <span class="st">"hg19"</span><span class="op">)</span></span></code></pre></div>
<p>Following functions are very similar as in online GREAT.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getEnrichmentTable-dispatch.html" class="external-link">getEnrichmentTable</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotVolcano-dispatch.html" class="external-link">plotVolcano</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic3_02_local_GREAT_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotRegionGeneAssociations-dispatch.html" class="external-link">plotRegionGeneAssociations</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic3_02_local_GREAT_files/figure-html/unnamed-chunk-4-2.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getRegionGeneAssociations-dispatch.html" class="external-link">getRegionGeneAssociations</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 869 ranges and 2 metadata columns:</span></span>
<span><span class="co">##         seqnames            ranges strand | annotated_genes    dist_to_TSS</span></span>
<span><span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;CharacterList&gt;  &lt;IntegerList&gt;</span></span>
<span><span class="co">##     [1]     chr1   9204434-9208784      * |     GPR157,H6PD  -15205,-86079</span></span>
<span><span class="co">##     [2]     chr1   9853594-9859363      * |   PIK3CD,CLSTN1   141804,25187</span></span>
<span><span class="co">##     [3]     chr1 10862809-10871681      * |  CASZ1,C1orf127   -6076,170413</span></span>
<span><span class="co">##     [4]     chr1 12716970-12723206      * | AADACL4,AADACL3   12404,-52912</span></span>
<span><span class="co">##     [5]     chr1 13814692-13823250      * | PRAMEF17,LRRC38    98604,16992</span></span>
<span><span class="co">##     ...      ...               ...    ... .             ...            ...</span></span>
<span><span class="co">##   [865]     chrY 16384799-16389852      * |   TMSB4Y,NLGN4Y 569352,-244636</span></span>
<span><span class="co">##   [866]     chrY 16842593-16846893      * |          NLGN4Y         208105</span></span>
<span><span class="co">##   [867]     chrY 16998750-17003225      * |          NLGN4Y         364262</span></span>
<span><span class="co">##   [868]     chrY 24401612-24406252      * |   RBMY1E,RBMY1F  -72523,-48754</span></span>
<span><span class="co">##   [869]     chrY 26206997-26215445      * |            DAZ2         841416</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 26 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/plotRegionGeneAssociations-dispatch.html" class="external-link">plotRegionGeneAssociations</a></span><span class="op">(</span><span class="va">res</span>, term_id <span class="op">=</span> <span class="st">"HALLMARK_APOPTOSIS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic3_02_local_GREAT_files/figure-html/unnamed-chunk-4-3.png" width="700"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># shinyReport(res)</span></span></code></pre></div>
<p>The following gene sets are directly supported:</p>
<ul>
<li>
<code>"GO:BP"</code>: Biological Process.</li>
<li>
<code>"GO:CC"</code>: Cellular Component.</li>
<li>
<code>"GO:MP"</code>: Molecular Function.</li>
</ul>
<p>Note MSigDB genes are only for human.</p>
<ul>
<li>
<code>"msigdb:H"</code> Hallmark gene sets.</li>
<li>
<code>"msigdb:C1"</code> Positional gene sets.</li>
<li>
<code>"msigdb:C2"</code> Curated gene sets.</li>
<li>
<code>"msigdb:C2:CGP"</code> C2 subcategory: chemical and genetic perturbations gene sets.</li>
<li>
<code>"msigdb:C2:CP"</code> C2 subcategory: canonical pathways gene sets.</li>
<li>
<code>"msigdb:C2:CP:KEGG"</code> C2 subcategory: KEGG subset of CP.</li>
<li>
<code>"msigdb:C2:CP:PID"</code> C2 subcategory: PID subset of CP.</li>
<li>
<code>"msigdb:C2:CP:REACTOME"</code> C2 subcategory: REACTOME subset of CP.</li>
<li>
<code>"msigdb:C2:CP:WIKIPATHWAYS"</code> C2 subcategory: WIKIPATHWAYS subset of CP.</li>
<li>
<code>"msigdb:C3"</code> Regulatory target gene sets.</li>
<li>
<code>"msigdb:C3:MIR:MIRDB"</code> miRDB of microRNA targets gene sets.</li>
<li>
<code>"msigdb:C3:MIR:MIR_LEGACY"</code> MIR_Legacy of MIRDB.</li>
<li>
<code>"msigdb:C3:TFT:GTRD"</code> GTRD transcription factor targets gene sets.</li>
<li>
<code>"msigdb:C3:TFT:TFT_LEGACY"</code> TFT_Legacy.</li>
<li>
<code>"msigdb:C4"</code> Computational gene sets.</li>
<li>
<code>"msigdb:C4:CGN"</code> C4 subcategory: cancer gene neighborhoods gene sets.</li>
<li>
<code>"msigdb:C4:CM"</code> C4 subcategory: cancer modules gene sets.</li>
<li>
<code>"msigdb:C5"</code> Ontology gene sets.</li>
<li>
<code>"msigdb:C5:GO:BP"</code> C5 subcategory: BP subset.</li>
<li>
<code>"msigdb:C5:GO:CC"</code> C5 subcategory: CC subset.</li>
<li>
<code>"msigdb:C5:GO:MF"</code> C5 subcategory: MF subset.</li>
<li>
<code>"msigdb:C5:HPO"</code> C5 subcategory: human phenotype ontology gene sets.</li>
<li>
<code>"msigdb:C6"</code> Oncogenic signature gene sets.</li>
<li>
<code>"msigdb:C7"</code> Immunologic signature gene sets.</li>
<li>
<code>"msigdb:C7:IMMUNESIGDB"</code> ImmuneSigDB subset of C7.</li>
<li>
<code>"msigdb:C7:VAX"</code> C7 subcategory: vaccine response gene sets.</li>
<li>
<code>"msigdb:C8"</code> Cell type signature gene sets.</li>
</ul>
<p>The <code>"GO:"</code> and “msigdb:” prefix can be removed when specifying in <code><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great()</a></code>.</p>
</div>
<div class="section level2">
<h2 id="setting-genome-background">Setting genome background<a class="anchor" aria-label="anchor" href="#setting-genome-background"></a>
</h2>
<p>Background can be set via <code>backgroud</code> and <code>excluce</code> arguments.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getGapFromUCSC.html" class="external-link">getGapFromUCSC</a></span><span class="op">(</span><span class="st">"hg19"</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">22</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"MSigDB:H"</span>, <span class="st">"hg19"</span>, exclude <span class="op">=</span> <span class="va">gap</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"GO:BP"</span>, <span class="st">"hg19"</span>, background <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">22</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"GO:BP"</span>, <span class="st">"hg19"</span>, exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chrX"</span>, <span class="st">"chrY"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Restricting the anallysis in a certain type of genome background helps to reduce the false positives. Examples are:</p>
<ul>
<li>excluding unmappable regions</li>
<li>regions with certain level of CG density</li>
<li>regions with certain level of gene densities</li>
<li>exons (coding regions)</li>
</ul>
</div>
<div class="section level2">
<h2 id="extend-from-the-complete-gene">Extend from the complete gene<a class="anchor" aria-label="anchor" href="#extend-from-the-complete-gene"></a>
</h2>
<p>This feature is only available in the devel version of <strong>rGREAT</strong>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"BP"</span>, <span class="st">"hg19"</span>, extend_from <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="using-other-gene-sets">Using other gene sets<a class="anchor" aria-label="anchor" href="#using-other-gene-sets"></a>
</h2>
<p>More gene sets can be set as a list of gene vectors. For most organisms, EntreZ IDs should be used as the gene ID type in gene sets.</p>
<p>From a <code>.gmt</code> file:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/read_gmt.html" class="external-link">read_gmt</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"http://dsigdb.tanlab.org/Downloads/D2_LINCS.gmt"</span><span class="op">)</span>, </span>
<span>    from <span class="op">=</span> <span class="st">"SYMBOL"</span>, to <span class="op">=</span> <span class="st">"ENTREZ"</span>, orgdb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="va">gs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $GSK429286A</span></span>
<span><span class="co">##  [1] "5566"  "9475"  "81788" "6093"  "5562"  "26524" "5592"  "5979"  "640"  </span></span>
<span><span class="co">## [10] "5567"  "3717"  "5613"  "23012" "695"   "3718" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`BS-181`</span></span>
<span><span class="co">##  [1] "122011" "1452"   "1022"   "7272"   "1454"   "1453"   "65975"  "147746"</span></span>
<span><span class="co">##  [9] "1859"   "1195"   "1196"   "9149"   "57396"  "5261"</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">gs</span>, <span class="st">"hg19"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 916 regions are associated to 1400 genes' extended TSSs.</span></span>
<span><span class="co">##   TSS source: TxDb.Hsapiens.UCSC.hg19.knownGene</span></span>
<span><span class="co">##   Genome: hg19</span></span>
<span><span class="co">##   OrgDb: org.Hs.eg.db</span></span>
<span><span class="co">##   Gene sets: self-provided</span></span>
<span><span class="co">##   Background: whole genome excluding gaps</span></span>
<span><span class="co">## Mode: Basal plus extension</span></span>
<span><span class="co">##   Proximal: 5000 bp upstream, 1000 bp downstream,</span></span>
<span><span class="co">##   plus Distal: up to 1000000 bp</span></span></code></pre>
<p>KEGG pathways:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/link/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"hsa:"</span>, <span class="st">""</span>, <span class="va">df</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"path:"</span>,  <span class="st">""</span>, <span class="va">df</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">gs_kegg</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">df</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">df</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">gs_kegg</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $hsa00010</span></span>
<span><span class="co">##  [1] "10327"  "124"    "125"    "126"    "127"    "128"    "130"    "130589"</span></span>
<span><span class="co">##  [9] "131"    "160287" "1737"   "1738"   "2023"   "2026"   "2027"   "217"   </span></span>
<span><span class="co">## [17] "218"    "219"    "2203"   "221"    "222"    "223"    "224"    "226"   </span></span>
<span><span class="co">## [25] "229"    "230"    "2538"   "2597"   "26330"  "2645"   "2821"   "3098"  </span></span>
<span><span class="co">## [33] "3099"   "3101"   "387712" "3939"   "3945"   "3948"   "441531" "501"   </span></span>
<span><span class="co">## [41] "5105"   "5106"   "5160"   "5161"   "5162"   "5211"   "5213"   "5214"  </span></span>
<span><span class="co">## [49] "5223"   "5224"   "5230"   "5232"   "5236"   "5313"   "5315"   "55276" </span></span>
<span><span class="co">## [57] "55902"  "57818"  "669"    "7167"   "80201"  "83440"  "84532"  "8789"  </span></span>
<span><span class="co">## [65] "92483"  "92579"  "9562"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $hsa00020</span></span>
<span><span class="co">##  [1] "1431"  "1737"  "1738"  "1743"  "2271"  "3417"  "3418"  "3419"  "3420" </span></span>
<span><span class="co">## [10] "3421"  "4190"  "4191"  "47"    "48"    "4967"  "50"    "5091"  "5105" </span></span>
<span><span class="co">## [19] "5106"  "5160"  "5161"  "5162"  "55753" "6389"  "6390"  "6391"  "6392" </span></span>
<span><span class="co">## [28] "8801"  "8802"  "8803"</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">gs_kegg</span>, <span class="st">"hg19"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 916 regions are associated to 1400 genes' extended TSSs.</span></span>
<span><span class="co">##   TSS source: TxDb.Hsapiens.UCSC.hg19.knownGene</span></span>
<span><span class="co">##   Genome: hg19</span></span>
<span><span class="co">##   OrgDb: org.Hs.eg.db</span></span>
<span><span class="co">##   Gene sets: self-provided</span></span>
<span><span class="co">##   Background: whole genome excluding gaps</span></span>
<span><span class="co">## Mode: Basal plus extension</span></span>
<span><span class="co">##   Proximal: 5000 bp upstream, 1000 bp downstream,</span></span>
<span><span class="co">##   plus Distal: up to 1000000 bp</span></span></code></pre>
<p>Reactome pathways:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">reactome.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs_reactome</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">reactomePATHID2EXTID</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">gs_reactome</span>, <span class="st">"hg19"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 916 regions are associated to 1400 genes' extended TSSs.</span></span>
<span><span class="co">##   TSS source: TxDb.Hsapiens.UCSC.hg19.knownGene</span></span>
<span><span class="co">##   Genome: hg19</span></span>
<span><span class="co">##   OrgDb: org.Hs.eg.db</span></span>
<span><span class="co">##   Gene sets: self-provided</span></span>
<span><span class="co">##   Background: whole genome excluding gaps</span></span>
<span><span class="co">## Mode: Basal plus extension</span></span>
<span><span class="co">##   Proximal: 5000 bp upstream, 1000 bp downstream,</span></span>
<span><span class="co">##   plus Distal: up to 1000000 bp</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="other-organisms-who-have-txdb-packages">Other organisms who have txdb packages<a class="anchor" aria-label="anchor" href="#other-organisms-who-have-txdb-packages"></a>
</h2>
<p>On Bioconductor, there are a family of annotation packages with name <strong>txdb.*.db</strong>. These packages contain sources of TSS definitions.</p>
<p>All supported <strong>txdb.*.db</strong> packages:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu">rGREAT</span><span class="fu">:::</span><span class="va">BIOC_ANNO_PKGS</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">tb</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">tb</span><span class="op">$</span><span class="va">genome_version_in_txdb</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="1%">
<col width="6%">
<col width="13%">
<col width="4%">
<col width="20%">
<col width="8%">
<col width="12%">
<col width="8%">
<col width="13%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">species_name</th>
<th align="left">species_latin</th>
<th align="right">taxon_id</th>
<th align="left">txdb</th>
<th align="left">gene_id_in_txdb</th>
<th align="left">genome_version_in_txdb</th>
<th align="left">orgdb</th>
<th align="left">primary_gene_id_in_orgdb</th>
<th align="left">chromosome_prefix</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">human</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">TxDb.Hsapiens.UCSC.hg18.knownGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">hg18</td>
<td align="left">org.Hs.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">human</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">TxDb.Hsapiens.UCSC.hg19.knownGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">hg19</td>
<td align="left">org.Hs.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">human</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">TxDb.Hsapiens.UCSC.hg38.knownGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">hg38</td>
<td align="left">org.Hs.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">mouse</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">TxDb.Mmusculus.UCSC.mm10.knownGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">mm10</td>
<td align="left">org.Mm.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">mouse</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">TxDb.Mmusculus.UCSC.mm39.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">mm39</td>
<td align="left">org.Mm.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">mouse</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">TxDb.Mmusculus.UCSC.mm9.knownGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">mm9</td>
<td align="left">org.Mm.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">rat</td>
<td align="left">Rattus norvegicus</td>
<td align="right">10116</td>
<td align="left">TxDb.Rnorvegicus.UCSC.rn4.ensGene</td>
<td align="left">Ensembl gene ID</td>
<td align="left">rn4</td>
<td align="left">org.Rn.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">rat</td>
<td align="left">Rattus norvegicus</td>
<td align="right">10116</td>
<td align="left">TxDb.Rnorvegicus.UCSC.rn5.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">rn5</td>
<td align="left">org.Rn.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="left">rat</td>
<td align="left">Rattus norvegicus</td>
<td align="right">10116</td>
<td align="left">TxDb.Rnorvegicus.UCSC.rn6.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">rn6</td>
<td align="left">org.Rn.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">12</td>
<td align="left">rat</td>
<td align="left">Rattus norvegicus</td>
<td align="right">10116</td>
<td align="left">TxDb.Rnorvegicus.UCSC.rn7.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">rn7</td>
<td align="left">org.Rn.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">13</td>
<td align="left">chicken</td>
<td align="left">Gallus gallus</td>
<td align="right">9031</td>
<td align="left">TxDb.Ggallus.UCSC.galGal4.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">galGal4</td>
<td align="left">org.Gg.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">14</td>
<td align="left">chicken</td>
<td align="left">Gallus gallus</td>
<td align="right">9031</td>
<td align="left">TxDb.Ggallus.UCSC.galGal5.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">galGal5</td>
<td align="left">org.Gg.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">15</td>
<td align="left">chicken</td>
<td align="left">Gallus gallus</td>
<td align="right">9031</td>
<td align="left">TxDb.Ggallus.UCSC.galGal6.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">galGal6</td>
<td align="left">org.Gg.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">16</td>
<td align="left">rhesus</td>
<td align="left">Macaca mulatta</td>
<td align="right">9544</td>
<td align="left">TxDb.Mmulatta.UCSC.rheMac10.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">rheMac10</td>
<td align="left">org.Mmu.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">17</td>
<td align="left">rhesus</td>
<td align="left">Macaca mulatta</td>
<td align="right">9544</td>
<td align="left">TxDb.Mmulatta.UCSC.rheMac3.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">rheMac3</td>
<td align="left">org.Mmu.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">18</td>
<td align="left">rhesus</td>
<td align="left">Macaca mulatta</td>
<td align="right">9544</td>
<td align="left">TxDb.Mmulatta.UCSC.rheMac8.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">rheMac8</td>
<td align="left">org.Mmu.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">19</td>
<td align="left">worm</td>
<td align="left">Caenorhabditis elegans</td>
<td align="right">6239</td>
<td align="left">TxDb.Celegans.UCSC.ce11.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">ce11</td>
<td align="left">org.Ce.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">21</td>
<td align="left">dog</td>
<td align="left">Canis familiari</td>
<td align="right">9615</td>
<td align="left">TxDb.Cfamiliaris.UCSC.canFam3.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">canFam3</td>
<td align="left">org.Cf.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">22</td>
<td align="left">dog</td>
<td align="left">Canis familiari</td>
<td align="right">9615</td>
<td align="left">TxDb.Cfamiliaris.UCSC.canFam4.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">canFam4</td>
<td align="left">org.Cf.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">23</td>
<td align="left">dog</td>
<td align="left">Canis familiari</td>
<td align="right">9615</td>
<td align="left">TxDb.Cfamiliaris.UCSC.canFam5.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">canFam5</td>
<td align="left">org.Cf.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">24</td>
<td align="left">pig</td>
<td align="left">Sus scrofa</td>
<td align="right">9823</td>
<td align="left">TxDb.Sscrofa.UCSC.susScr11.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">susScr11</td>
<td align="left">org.Ss.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">25</td>
<td align="left">pig</td>
<td align="left">Sus scrofa</td>
<td align="right">9823</td>
<td align="left">TxDb.Sscrofa.UCSC.susScr3.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">susScr3</td>
<td align="left">org.Ss.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">26</td>
<td align="left">yeast</td>
<td align="left">Saccharomyces cerevisiae</td>
<td align="right">4932</td>
<td align="left">TxDb.Scerevisiae.UCSC.sacCer2.sgdGene</td>
<td align="left">SGD Gene ID</td>
<td align="left">sacCer2</td>
<td align="left">org.Sc.sgd.db</td>
<td align="left">SGD Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">27</td>
<td align="left">yeast</td>
<td align="left">Saccharomyces cerevisiae</td>
<td align="right">4932</td>
<td align="left">TxDb.Scerevisiae.UCSC.sacCer3.sgdGene</td>
<td align="left">SGD Gene ID</td>
<td align="left">sacCer3</td>
<td align="left">org.Sc.sgd.db</td>
<td align="left">SGD Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">28</td>
<td align="left">chimp</td>
<td align="left">Pan troglodytes</td>
<td align="right">9598</td>
<td align="left">TxDb.Ptroglodytes.UCSC.panTro4.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">panTro4</td>
<td align="left">org.Pt.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">29</td>
<td align="left">chimp</td>
<td align="left">Pan troglodytes</td>
<td align="right">9598</td>
<td align="left">TxDb.Ptroglodytes.UCSC.panTro5.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">panTro5</td>
<td align="left">org.Pt.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">30</td>
<td align="left">chimp</td>
<td align="left">Pan troglodytes</td>
<td align="right">9598</td>
<td align="left">TxDb.Ptroglodytes.UCSC.panTro6.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">panTro6</td>
<td align="left">org.Pt.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">31</td>
<td align="left">fruitfly</td>
<td align="left">Drosophila melanogaster</td>
<td align="right">7227</td>
<td align="left">TxDb.Dmelanogaster.UCSC.dm3.ensGene</td>
<td align="left">Ensembl gene ID</td>
<td align="left">dm3</td>
<td align="left">org.Dm.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">32</td>
<td align="left">fruitfly</td>
<td align="left">Drosophila melanogaster</td>
<td align="right">7227</td>
<td align="left">TxDb.Dmelanogaster.UCSC.dm6.ensGene</td>
<td align="left">Ensembl gene ID</td>
<td align="left">dm6</td>
<td align="left">org.Dm.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">33</td>
<td align="left">zebrafish</td>
<td align="left">Danio rerio</td>
<td align="right">7955</td>
<td align="left">TxDb.Drerio.UCSC.danRer10.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">danRer10</td>
<td align="left">org.Dr.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">34</td>
<td align="left">zebrafish</td>
<td align="left">Danio rerio</td>
<td align="right">7955</td>
<td align="left">TxDb.Drerio.UCSC.danRer11.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">danRer11</td>
<td align="left">org.Dr.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">35</td>
<td align="left">cattle</td>
<td align="left">Bos taurus</td>
<td align="right">9913</td>
<td align="left">TxDb.Btaurus.UCSC.bosTau8.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">bosTau8</td>
<td align="left">org.Bt.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="odd">
<td align="left">36</td>
<td align="left">cattle</td>
<td align="left">Bos taurus</td>
<td align="right">9913</td>
<td align="left">TxDb.Btaurus.UCSC.bosTau9.refGene</td>
<td align="left">Entrez Gene ID</td>
<td align="left">bosTau9</td>
<td align="left">org.Bt.eg.db</td>
<td align="left">Entrez Gene ID</td>
<td align="left">chr</td>
</tr>
<tr class="even">
<td align="left">37</td>
<td align="left">thale cress</td>
<td align="left">Arabidopsis thaliana</td>
<td align="right">3702</td>
<td align="left">TxDb.Athaliana.BioMart.plantsmart51</td>
<td align="left">TAIR ID</td>
<td align="left">araTha</td>
<td align="left">org.At.tair.db</td>
<td align="left">TAIR ID</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>For these packages, users only need to specify the “genome version”:</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"BP"</span>, <span class="st">"mm10"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"BP"</span>, <span class="st">"rn7"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="use-msigdb-gene-sets">Use MSigDB gene sets<a class="anchor" aria-label="anchor" href="#use-msigdb-gene-sets"></a>
</h2>
<p>The <strong>msigdbr</strong> package provides gene sets by mapping to several other organisms.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_species.html" class="external-link">msigdbr_species</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 20 × 2</span></span></span>
<span><span class="co">##    species_name                    species_common_name                          </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Anolis carolinensis             Carolina anole, green anole                  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Bos taurus                      bovine, cattle, cow, dairy cow, domestic cat…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Caenorhabditis elegans          <span style="color: #BB0000;">NA</span>                                           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Canis lupus familiaris          dog, dogs                                    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> Danio rerio                     leopard danio, zebra danio, zebra fish, zebr…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Drosophila melanogaster         fruit fly                                    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Equus caballus                  domestic horse, equine, horse                </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Felis catus                     cat, cats, domestic cat                      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Gallus gallus                   bantam, chicken, chickens, Gallus domesticus </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Homo sapiens                    human                                        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> Macaca mulatta                  rhesus macaque, rhesus macaques, Rhesus monk…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> Monodelphis domestica           gray short-tailed opossum                    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">13</span> Mus musculus                    house mouse, mouse                           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">14</span> Ornithorhynchus anatinus        duck-billed platypus, duckbill platypus, pla…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">15</span> Pan troglodytes                 chimpanzee                                   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">16</span> Rattus norvegicus               brown rat, Norway rat, rat, rats             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">17</span> Saccharomyces cerevisiae        baker's yeast, brewer's yeast, S. cerevisiae </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">18</span> Schizosaccharomyces pombe 972h- <span style="color: #BB0000;">NA</span>                                           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">19</span> Sus scrofa                      pig, pigs, swine, wild boar                  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">20</span> Xenopus tropicalis              tropical clawed frog, western clawed frog</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"chimpanzee"</span>, category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">h_gene_sets</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 18</span></span></span>
<span><span class="co">##   gs_cat gs_subcat gs_name               gene_symbol entrez_gene ensembl_gene   </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> H      <span style="color: #949494;">""</span>        HALLMARK_ADIPOGENESIS ABCA1            <span style="text-decoration: underline;">464</span>630 ENSPTRG0000002…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> H      <span style="color: #949494;">""</span>        HALLMARK_ADIPOGENESIS ABCB8            <span style="text-decoration: underline;">463</span>892 ENSPTRG0000001…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> H      <span style="color: #949494;">""</span>        HALLMARK_ADIPOGENESIS ACAA2            <span style="text-decoration: underline;">455</span>414 ENSPTRG0000001…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> H      <span style="color: #949494;">""</span>        HALLMARK_ADIPOGENESIS ACADL            <span style="text-decoration: underline;">459</span>914 ENSPTRG0000001…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> H      <span style="color: #949494;">""</span>        HALLMARK_ADIPOGENESIS ACADM            <span style="text-decoration: underline;">469</span>356 ENSPTRG0000000…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> H      <span style="color: #949494;">""</span>        HALLMARK_ADIPOGENESIS ACADS            <span style="text-decoration: underline;">742</span>921 ENSPTRG0000000…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 12 more variables: human_gene_symbol &lt;chr&gt;, human_entrez_gene &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   human_ensembl_gene &lt;chr&gt;, gs_id &lt;chr&gt;, gs_pmid &lt;chr&gt;, gs_geoid &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;, gs_description &lt;chr&gt;, taxon_id &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;</span></span></span></code></pre>
<p>Convert from the original format to a list of gene vectors.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">h_gene_sets</span><span class="op">$</span><span class="va">entrez_gene</span>, <span class="va">h_gene_sets</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="va">h_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">h_gene_sets</span>, <span class="va">as.character</span><span class="op">)</span>  <span class="co"># just to make sure gene IDs are all in character.</span></span>
<span><span class="va">h_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">h_gene_sets</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="co"># remove NA and duplicated genes</span></span>
<span><span class="va">h_gene_sets</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $HALLMARK_ADIPOGENESIS</span></span>
<span><span class="co">##   [1] "464630"    "463892"    "455414"    "459914"    "469356"    "742921"   </span></span>
<span><span class="co">##   [7] "454672"    "104003784" "454895"    "451866"    "737339"    "471032"   </span></span>
<span><span class="co">##  [13] "451742"    "737305"    "100615914" "456723"    "107967644" "454362"   </span></span>
<span><span class="co">##  [19] "464334"    "743667"    "741867"    "449586"    "100614256" "741708"   </span></span>
<span><span class="co">##  [25] "459164"    "746692"    "473976"    "452433"    "468889"    "745443"   </span></span>
<span><span class="co">##  [31] "460926"    "455644"    "451116"    "454684"    "744890"    "461229"   </span></span>
<span><span class="co">##  [37] "740513"    "104005232" "463949"    "469319"    "748673"    "450673"   </span></span>
<span><span class="co">##  [43] "468605"    "471455"    "456837"    "464611"    "452659"    "472079"   </span></span>
<span><span class="co">##  [49] "452307"    "454118"    "100616508" "465727"    "742828"    "737945"   </span></span>
<span><span class="co">##  [55] "107976794" "746229"    "472893"    "456557"    "457056"    "747265"   </span></span>
<span><span class="co">##  [61] "736777"    "464460"    "451393"    "745691"    "454512"    "466780"   </span></span>
<span><span class="co">##  [67] "463861"    "744984"    "452566"    "457117"    "747936"    "459360"   </span></span>
<span><span class="co">##  [73] "461436"    "464353"    "464074"    "466651"    "451984"    "456243"   </span></span>
<span><span class="co">##  [79] "464255"    "467738"    "466732"    "461244"    "456929"    "460520"   </span></span>
<span><span class="co">##  [85] "450562"    "450738"    "464140"    "459670"    "452976"    "471703"   </span></span>
<span><span class="co">##  [91] "741876"    "471135"    "461424"    "459828"    "452295"    "460113"   </span></span>
<span><span class="co">##  [97] "453565"    "741179"    "747276"    "470423"    "451967"    "450290"   </span></span>
<span><span class="co">## [103] "473975"    "460157"    "462946"    "449638"    "738797"    "456076"   </span></span>
<span><span class="co">## [109] "451807"    "464031"    "739986"    "459173"    "460872"    "463484"   </span></span>
<span><span class="co">## [115] "462853"    "739167"    "457477"    "742027"    "746245"    "472764"   </span></span>
<span><span class="co">## [121] "747387"    "744096"    "101057233" "744811"    "463686"    "744435"   </span></span>
<span><span class="co">## [127] "468748"    "451175"    "460227"    "454744"    "739996"    "450735"   </span></span>
<span><span class="co">## [133] "454478"    "457929"    "738397"    "458602"    "456908"    "451591"   </span></span>
<span><span class="co">## [139] "450310"    "107970333" "465012"    "463481"    "460178"    "470365"   </span></span>
<span><span class="co">## [145] "742092"    "741184"    "459094"    "459374"    "456940"    "745779"   </span></span>
<span><span class="co">## [151] "454531"    "737918"    "107973114" "742100"    "470420"    "468499"   </span></span>
<span><span class="co">## [157] "467657"    "100608935" "462416"    "451281"    "470281"    "452359"   </span></span>
<span><span class="co">## [163] "456862"    "456526"    "747462"    "474051"    "456155"    "458647"   </span></span>
<span><span class="co">## [169] "744390"    "455841"    "459096"    "459031"    "450574"    "449637"   </span></span>
<span><span class="co">## [175] "450628"    "470477"    "471247"    "453405"    "739128"    "454681"   </span></span>
<span><span class="co">## [181] "464707"    "470417"    "450933"    "459685"    "460443"    "468406"   </span></span>
<span><span class="co">## [187] "458803"    "467151"    "464550"    "745004"    "451416"    "735808"   </span></span>
<span><span class="co">## [193] "743144"    "460348"    "107974864" "471631"    "741897"    "463489"   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">##   [1] "454210"    "461523"    "450363"    "100609296" "459646"    "740898"   </span></span>
<span><span class="co">##   [7] "466415"    "450170"    "465345"    "456984"    "744209"    "449497"   </span></span>
<span><span class="co">##  [13] "100608992" "459361"    "741390"    "468208"    "748142"    "473220"   </span></span>
<span><span class="co">##  [19] "748205"    "736543"    "454593"    "747004"    "454579"    "747123"   </span></span>
<span><span class="co">##  [25] "462689"    "460323"    "740071"    "450128"    "469524"    "449512"   </span></span>
<span><span class="co">##  [31] "748272"    "470617"    "451584"    "742330"    "451585"    "450124"   </span></span>
<span><span class="co">##  [37] "100615583" "473802"    "460569"    "745293"    "462191"    "740560"   </span></span>
<span><span class="co">##  [43] "470892"    "470900"    "735755"    "470426"    "460577"    "465021"   </span></span>
<span><span class="co">##  [49] "465607"    "736196"    "457127"    "453745"    "457277"    "738275"   </span></span>
<span><span class="co">##  [55] "471200"    "459634"    "457770"    "469142"    "463415"    "466216"   </span></span>
<span><span class="co">##  [61] "458797"    "453714"    "469204"    "750603"    "100615835" "740028"   </span></span>
<span><span class="co">##  [67] "451695"    "451158"    "471510"    "738331"    "469584"    "739516"   </span></span>
<span><span class="co">##  [73] "465940"    "461906"    "468521"    "457003"    "472959"    "457020"   </span></span>
<span><span class="co">##  [79] "467610"    "461873"    "452825"    "460623"    "463280"    "746195"   </span></span>
<span><span class="co">##  [85] "750725"    "100608816" "449592"    "471979"    "471977"    "471974"   </span></span>
<span><span class="co">##  [91] "462591"    "462540"    "494187"    "450196"    "474132"    "473965"   </span></span>
<span><span class="co">##  [97] "449517"    "747276"    "470077"    "743102"    "472749"    "469657"   </span></span>
<span><span class="co">## [103] "736204"    "460816"    "471723"    "455851"    "449564"    "737808"   </span></span>
<span><span class="co">## [109] "449644"    "739011"    "744277"    "450200"    "461472"    "456370"   </span></span>
<span><span class="co">## [115] "450884"    "470203"    "101059843" "449565"    "454005"    "463288"   </span></span>
<span><span class="co">## [121] "464245"    "745517"    "463371"    "470524"    "462386"    "450927"   </span></span>
<span><span class="co">## [127] "454294"    "454045"    "458607"    "735556"    "464979"    "450156"   </span></span>
<span><span class="co">## [133] "738375"    "456715"    "471734"    "736309"    "744486"    "459682"   </span></span>
<span><span class="co">## [139] "745667"    "472771"    "462888"    "748652"    "449582"    "458294"   </span></span>
<span><span class="co">## [145] "460699"    "459239"    "741196"    "460720"    "100322885" "469743"   </span></span>
<span><span class="co">## [151] "455026"    "740704"    "740477"    "450512"    "453993"    "456276"   </span></span>
<span><span class="co">## [157] "743176"    "748032"    "457607"    "462249"    "464277"    "737451"   </span></span>
<span><span class="co">## [163] "746600"    "737526"    "456065"    "461536"    "107966305" "746721"   </span></span>
<span><span class="co">## [169] "737070"    "459209"    "451169"    "450503"    "461971"    "461023"   </span></span>
<span><span class="co">## [175] "459834"    "100610925" "471978"    "746399"    "746814"    "456060"   </span></span>
<span><span class="co">## [181] "457742"    "451611"    "107971092" "461167"    "471325"    "471374"   </span></span>
<span><span class="co">## [187] "471167"    "494186"    "748737"    "464876"    "741922"    "745141"   </span></span>
<span><span class="co">## [193] "452125"    "453161"    "743187"    "459427"</span></span></code></pre>
<p>Remember to pick the correct genome version.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">h_gene_sets</span>, <span class="st">"panTro6"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="other-organisms-who-have-txdb-object-on-annotationhub">Other organisms who have txdb object on AnnotationHub<a class="anchor" aria-label="anchor" href="#other-organisms-who-have-txdb-object-on-annotationhub"></a>
</h2>
<p>For some other organisms, although they don’t have a <strong>txdb</strong> package on Bioconductor, but they have a <code>txdb</code> object on <strong>AnnotationHub</strong>:</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: BiocFileCache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: dbplyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'AnnotationHub'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     cache</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ah</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="st">"TxDb"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AnnotationHub with 511 records</span></span>
<span><span class="co">## # snapshotDate(): 2024-10-28</span></span>
<span><span class="co">## # $dataprovider: UCSC, FungiDB, TriTrypDB, PlasmoDB, MicrosporidiaDB, ToxoDB...</span></span>
<span><span class="co">## # $species: Homo sapiens, Rattus norvegicus, Macaca mulatta, Gallus gallus, ...</span></span>
<span><span class="co">## # $rdataclass: TxDb, SQLiteFile, FaFile, ChainFile</span></span>
<span><span class="co">## # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">## #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">## # retrieve records with, e.g., 'object[["AH52245"]]' </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##              title                                     </span></span>
<span><span class="co">##   AH52245  | TxDb.Athaliana.BioMart.plantsmart22.sqlite</span></span>
<span><span class="co">##   AH52246  | TxDb.Athaliana.BioMart.plantsmart25.sqlite</span></span>
<span><span class="co">##   AH52247  | TxDb.Athaliana.BioMart.plantsmart28.sqlite</span></span>
<span><span class="co">##   AH52248  | TxDb.Btaurus.UCSC.bosTau8.refGene.sqlite  </span></span>
<span><span class="co">##   AH52249  | TxDb.Celegans.UCSC.ce11.refGene.sqlite    </span></span>
<span><span class="co">##   ...        ...                                       </span></span>
<span><span class="co">##   AH114096 | TxDb.Mmusculus.UCSC.mm39.refGene.sqlite   </span></span>
<span><span class="co">##   AH116719 | TxDb.Hsapiens.UCSC.hg38.refGene.sqlite    </span></span>
<span><span class="co">##   AH116720 | TxDb.Mmusculus.UCSC.mm39.refGene.sqlite   </span></span>
<span><span class="co">##   AH117076 | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite  </span></span>
<span><span class="co">##   AH117077 | TxDb.Mmusculus.UCSC.mm39.knownGene.sqlite</span></span></code></pre>
<p>But it seems many of those organisms are microbes. I will not introduce it here.</p>
</div>
<div class="section level2">
<h2 id="other-organisms-who-do-not-have-txdb-objects">Other organisms who do not have txdb objects<a class="anchor" aria-label="anchor" href="#other-organisms-who-do-not-have-txdb-objects"></a>
</h2>
<p>Then we need to obtain the TSS definition manually. Obtaining gene sets for an organism is relatively easy, but it is sometimes difficult to find a proper genome version because for these “rare” genomes, their genome data evolves very fast.</p>
<p>There are mainly two sources to obtain TSS definitions: 1. from NCBI, 2. from Ensembl.</p>
<div class="section level3">
<h3 id="get-tss-from-ncbi">Get TSS from NCBI<a class="anchor" aria-label="anchor" href="#get-tss-from-ncbi"></a>
</h3>
<p><strong>Currently some service from NCBI is not available, so we skip this and the next section</strong>.</p>
<p>We need to first know the “accession number” of an organism’s genome. Taking dolphin as an example, its web page on NCBI Genome is <a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_011762595.1/" class="external-link uri">https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_011762595.1/</a>. In <strong>rGREAT</strong>, there is a <code><a href="https://rdrr.io/pkg/rGREAT/man/getGenomeDataFromNCBI.html" class="external-link">getGenomeDataFromNCBI()</a></code> function which can automatically retrieve gene definitions from NCBI.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getGenomeDataFromNCBI.html" class="external-link">getGenomeDataFromNCBI</a></span><span class="op">(</span><span class="st">"GCF_011762595.1"</span>, return_granges <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">genes</span></span></code></pre></div>
<p>As we have already introduced, retrieve GO gene sets for an organism is easy:</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span>
<span><span class="va">ah</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">orgdb</span> <span class="op">=</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH112418"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/GSEAtraining" class="external-link">GSEAtraining</a></span><span class="op">)</span></span>
<span><span class="va">gs</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_GO_gene_sets_from_orgdb.html">get_GO_gene_sets_from_orgdb</a></span><span class="op">(</span><span class="va">orgdb</span>, <span class="st">"BP"</span><span class="op">)</span></span>
<span><span class="va">gs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<p>Note <code>genes</code> is set to the argument <code>tss_source</code>:</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">gs</span>, tss_source <span class="op">=</span> <span class="va">genes</span><span class="op">)</span></span></code></pre></div>
<p>Remember to double check the format of chromosome names in <code>gr</code> and in <code>genes</code>.</p>
</div>
<div class="section level3">
<h3 id="kegg">KEGG<a class="anchor" aria-label="anchor" href="#kegg"></a>
</h3>
<p>KEGG also links to NCBI genomes, but the genome version for an organism is not always synchronized to NCBI. The function <code><a href="https://rdrr.io/pkg/rGREAT/man/getKEGGGenome.html" class="external-link">getKEGGGenome()</a></code> returns the accession number of the genome recorded on KEGG, and this accession number can be directly used in <code><a href="https://rdrr.io/pkg/rGREAT/man/getGenomeDataFromNCBI.html" class="external-link">getGenomeDataFromNCBI()</a></code>.</p>
<p><code><a href="https://rdrr.io/pkg/rGREAT/man/getKEGGPathways.html" class="external-link">getKEGGPathways()</a></code> is a helper function which downloads KEGG pathways.</p>
<p>In the following example, <code>"mgp"</code> is the organism code for Meleagris gallopavo (turkey).</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getGenomeDataFromNCBI.html" class="external-link">getGenomeDataFromNCBI</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getKEGGGenome.html" class="external-link">getKEGGGenome</a></span><span class="op">(</span><span class="st">"mgp"</span><span class="op">)</span>, return_granges <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getKEGGPathways.html" class="external-link">getKEGGPathways</a></span><span class="op">(</span><span class="st">"mgp"</span><span class="op">)</span></span>
<span><span class="va">gene_sets</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">gene_sets</span>, <span class="va">genes</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="use-biomart-go-gene-sets">Use BioMart GO gene sets<a class="anchor" aria-label="anchor" href="#use-biomart-go-gene-sets"></a>
</h3>
<p>The <strong>BioMartGOGeneSets</strong> package not only contains GO gene sets for &gt; 700 organisms, but also the gene definitions. <code><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great()</a></code> has already integrate <strong>BioMartGOGeneSets</strong> and you only need to specify the <code>biomart_dataset</code> argument.</p>
<p>Note, the genes from biomart may not come from the genome with the same version as on NCBI, normally older than NCBI. You need to double check the genome version on biomart and in your <code>gr</code>.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/great.html" class="external-link">great</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"GO:BP"</span>, biomart_dataset <span class="op">=</span> <span class="st">"amelanoleuca_gene_ensembl"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>Run local GREAT analysis with a list of TFBS from UCSC table browser:</p>
<pre><code>clade: Mammal
genome: Human
assembly: GRCh37/hg19
group: Regulation
track: ENCODE 3 TFBS
table: 22Rv1 ZFX (encTfChipRkENCFF445OFX)</code></pre>
<p>In the “output format” field, select “BED - browser extensible data”, then click the button “get output”. In the next page, click the button “get BED”.</p>
<p>Use 1. the GO BP gene sets and 2. KEGG pathways.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
