<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 1-02: Get pathways from KEGG • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 1-02: Get pathways from KEGG">
<meta property="og:description" content="GSEAtraining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 1-02: Get pathways from KEGG</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2024-01-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic1_02_kegg.Rmd" class="external-link"><code>vignettes/topic1_02_kegg.Rmd</code></a></small>
      <div class="hidden name"><code>topic1_02_kegg.Rmd</code></div>

    </div>

    
    
<style>
blockquote {
    padding: 10px 20px;
    margin: 0 0 20px;
    font-size: 14px;
    background: #eee;
    border-left: 6px #CCCCCC solid;
}
</style>
<p>KEGG is a comprehensive database which not only contains pathways but also a large variaty of other information. The entrence of getting KEGG data programmatically is via the KEGG REST API.</p>
<p>Please note the restrictions of using KEGG data (<a href="https://www.kegg.jp/kegg/legal.html" class="external-link uri">https://www.kegg.jp/kegg/legal.html</a>):</p>
<blockquote>
<h3 id="academic-use-of-kegg">Academic use of KEGG</h3>
<p>Academic users may freely use the KEGG website at <a href="https://www.kegg.jp/" class="external-link uri">https://www.kegg.jp/</a> or its mirror site at GenomeNet <a href="https://www.genome.jp/kegg/" class="external-link uri">https://www.genome.jp/kegg/</a>. Academic users who utilize KEGG for providing academic services are requested to obtain an academic service provider license, which is included in the KEGG FTP academic subscription. The KEGG FTP academic subscription, which is a paid service (see background information), may also be obtained to conveniently download the entire KEGG database.</p>
<h3 id="non-academic-use-of-kegg">Non-academic use of KEGG</h3>
<p>Non-academic users must understand that KEGG is not a public database and non-academic use of KEGG generally requires a commercial license. There are two types of commercial licenses available: end user and business. The end user license includes access rights to the FTP site and the website, while the business license includes access rights to the FTP site only. Please contact Pathway Solutions for more details.</p>
</blockquote>
<div class="section level2">
<h2 id="directly-read-from-the-kegg-api">Directly read from the KEGG API<a class="anchor" aria-label="anchor" href="#directly-read-from-the-kegg-api"></a>
</h2>
<p>KEGG provides its data via a REST API (<a href="https://rest.kegg.jp/" class="external-link uri">https://rest.kegg.jp/</a>). There are several commands that can be used to retrieve specific types of data. The URL form of the request is</p>
<pre><code>https://rest.kegg.jp/&lt;operation&gt;/&lt;argument&gt;[/&lt;argument2[/&lt;argument3&gt; ...]]</code></pre>
<p>To get the KEGG pathway gene sets, we will use the following two operators:</p>
<ul>
<li>
<code>link</code>: get the mapping between two sources.</li>
<li>
<code>list</code>: get the details of a list of items.</li>
</ul>
<p>The <code>link</code> operator returns the mapping between two sources of information. We can use the following command to get the mappings between genes and pathways for human.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/link/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          V1            V2</span></span>
<span><span class="co">## 1 hsa:10327 path:hsa00010</span></span>
<span><span class="co">## 2   hsa:124 path:hsa00010</span></span>
<span><span class="co">## 3   hsa:125 path:hsa00010</span></span>
<span><span class="co">## 4   hsa:126 path:hsa00010</span></span>
<span><span class="co">## 5   hsa:127 path:hsa00010</span></span>
<span><span class="co">## 6   hsa:128 path:hsa00010</span></span></code></pre>
<p>In the example, <code><a href="https://rdrr.io/r/base/connections.html" class="external-link">url()</a></code> constructs a connection object that directly transfer data from the remote URL. You can also first download the output from KEGG into a local file, read it and later delete the temporary file. <code><a href="https://rdrr.io/r/base/connections.html" class="external-link">url()</a></code> basically automates such steps.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">temp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/link/pathway/hsa"</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span>
<span><span class="va">df1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">temp</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span></span></code></pre></div>
<p>Also the URL <code>"https://rest.kegg.jp/link/hsa/pathway"</code> returns identical results which only switches the two columns in the table.</p>
<p>In the output, the first column contains Entrez ID (users may remove the <code>"hsa:"</code> prefix for downstream analysis) and the second columncontains KEGG pathways IDs (users may remove the <code>"path:"</code> previx).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df1</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"hsa:"</span>, <span class="st">""</span>, <span class="va">df1</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">df1</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"path:"</span>,  <span class="st">""</span>, <span class="va">df1</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      V1       V2</span></span>
<span><span class="co">## 1 10327 hsa00010</span></span>
<span><span class="co">## 2   124 hsa00010</span></span>
<span><span class="co">## 3   125 hsa00010</span></span>
<span><span class="co">## 4   126 hsa00010</span></span>
<span><span class="co">## 5   127 hsa00010</span></span>
<span><span class="co">## 6   128 hsa00010</span></span></code></pre>
<p>To get the full name of pathways, use the <code>list</code> command:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         V1                                                     V2</span></span>
<span><span class="co">## 1 hsa01100              Metabolic pathways - Homo sapiens (human)</span></span>
<span><span class="co">## 2 hsa01200               Carbon metabolism - Homo sapiens (human)</span></span>
<span><span class="co">## 3 hsa01210 2-Oxocarboxylic acid metabolism - Homo sapiens (human)</span></span>
<span><span class="co">## 4 hsa01212           Fatty acid metabolism - Homo sapiens (human)</span></span>
<span><span class="co">## 5 hsa01230     Biosynthesis of amino acids - Homo sapiens (human)</span></span>
<span><span class="co">## 6 hsa01232           Nucleotide metabolism - Homo sapiens (human)</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df2</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">" - Homo.*$"</span>, <span class="st">""</span>, <span class="va">df2</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         V1                              V2</span></span>
<span><span class="co">## 1 hsa01100              Metabolic pathways</span></span>
<span><span class="co">## 2 hsa01200               Carbon metabolism</span></span>
<span><span class="co">## 3 hsa01210 2-Oxocarboxylic acid metabolism</span></span>
<span><span class="co">## 4 hsa01212           Fatty acid metabolism</span></span>
<span><span class="co">## 5 hsa01230     Biosynthesis of amino acids</span></span>
<span><span class="co">## 6 hsa01232           Nucleotide metabolism</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="keggrest">KEGGREST<a class="anchor" aria-label="anchor" href="#keggrest"></a>
</h2>
<p>There is a <strong>KEGGREST</strong> package which implements a full interface to access KEGG data in R. All the APIs from KEGG REST service are suppoted in <strong>KEGGREST</strong>. The names of functions in <strong>KEGGREST</strong> are consistent to the operators in the KEGG API. For example, to get the mapping between genes and pathways, the function <code><a href="https://rdrr.io/pkg/KEGGREST/man/keggLink.html" class="external-link">keggLink()</a></code> can be used.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">KEGGREST</span><span class="op">)</span></span>
<span><span class="va">gene2pathway</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/KEGGREST/man/keggLink.html" class="external-link">keggLink</a></span><span class="op">(</span><span class="st">"pathway"</span>, <span class="st">"hsa"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene2pathway</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       hsa:10327         hsa:124         hsa:125         hsa:126         hsa:127 </span></span>
<span><span class="co">## "path:hsa00010" "path:hsa00010" "path:hsa00010" "path:hsa00010" "path:hsa00010" </span></span>
<span><span class="co">##         hsa:128 </span></span>
<span><span class="co">## "path:hsa00010"</span></span></code></pre>
<p>This named vector is not commonly used for downstream gene set analysis. A more used format is a data frame. We can simply converted <code>gene2pathway</code> to a data frame.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df3</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    gene_id    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"hsa:"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene2pathway</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    pathway_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"path:"</span>, <span class="st">""</span>, <span class="va">gene2pathway</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   gene_id pathway_id</span></span>
<span><span class="co">## 1   10327   hsa00010</span></span>
<span><span class="co">## 2     124   hsa00010</span></span>
<span><span class="co">## 3     125   hsa00010</span></span>
<span><span class="co">## 4     126   hsa00010</span></span>
<span><span class="co">## 5     127   hsa00010</span></span>
<span><span class="co">## 6     128   hsa00010</span></span></code></pre>
<p>Similar as the <code>list</code> operator in KEGG API, <code><a href="https://rdrr.io/pkg/KEGGREST/man/keggList.html" class="external-link">keggList()</a></code> function can be used to retrieve pathway IDs and their names. The function returns a named vector.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/KEGGREST/man/keggList.html" class="external-link">keggList</a></span><span class="op">(</span><span class="st">"pathway"</span>, <span class="st">"hsa"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                 hsa01100 </span></span>
<span><span class="co">##              "Metabolic pathways - Homo sapiens (human)" </span></span>
<span><span class="co">##                                                 hsa01200 </span></span>
<span><span class="co">##               "Carbon metabolism - Homo sapiens (human)" </span></span>
<span><span class="co">##                                                 hsa01210 </span></span>
<span><span class="co">## "2-Oxocarboxylic acid metabolism - Homo sapiens (human)" </span></span>
<span><span class="co">##                                                 hsa01212 </span></span>
<span><span class="co">##           "Fatty acid metabolism - Homo sapiens (human)" </span></span>
<span><span class="co">##                                                 hsa01230 </span></span>
<span><span class="co">##     "Biosynthesis of amino acids - Homo sapiens (human)" </span></span>
<span><span class="co">##                                                 hsa01232 </span></span>
<span><span class="co">##           "Nucleotide metabolism - Homo sapiens (human)"</span></span></code></pre>
<div class="section level3">
<h3 id="highlight-genes-on-the-pathway-image">Highlight genes on the pathway image<a class="anchor" aria-label="anchor" href="#highlight-genes-on-the-pathway-image"></a>
</h3>
<style>
.delete {
    text-decoration: line-through;
}
</style>
<p>Users may want to highlight some genes on the pathway image, e.g. to highlight differentially expressed genes. <span class="delete">The function <code><a href="https://rdrr.io/pkg/KEGGREST/man/mark.pathway.by.objects.html" class="external-link">color.pathway.by.objects()</a></code> in <strong>KEGGREST</strong> accepts a vector of genes and assigns them with a set of colors. In the following example, I randomly pick 10 genes from the pathway “hsa04911”.</span> <strong>It does not work any more because KEGG has changed its API.</strong></p>
<p>Let’s manually color genes for a specific pathway on KEGG web site.</p>
<p>Open <a href="https://www.genome.jp/pathway/hsa04911" class="external-link uri">https://www.genome.jp/pathway/hsa04911</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene2pathway</span><span class="op">[</span><span class="va">gene2pathway</span> <span class="op">==</span> <span class="st">"path:hsa04911"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">diff_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">genes</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">settings</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">diff_genes</span>, </span>
<span>    colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"orange"</span>, <span class="fl">10</span><span class="op">)</span>,<span class="st">","</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="co"># background,labels</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">settings</span>, file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/showConnections.html" class="external-link">stdout</a></span><span class="op">(</span><span class="op">)</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span></code></pre></div>
<pre><code>hsa:6514    orange,blue
hsa:775 orange,blue
hsa:2776    orange,blue
hsa:6833    orange,blue
hsa:139760  orange,blue
hsa:114 orange,blue
hsa:6513    orange,blue
hsa:886 orange,blue
hsa:3781    orange,blue
hsa:5864    orange,blue</code></pre>
<p>How to set gene colors: <a href="https://www.genome.jp/kegg/webapp/color_gui.html" class="external-link uri">https://www.genome.jp/kegg/webapp/color_gui.html</a></p>
<p>The package <strong>ggkegg</strong> also supports customizing KEGG graph.</p>
</div>
</div>
<div class="section level2">
<h2 id="clusterprofiler">clusterProfiler<a class="anchor" aria-label="anchor" href="#clusterprofiler"></a>
</h2>
<p>The second Bioconductor pacakge <strong>clusterProfiler</strong> has a simple function <code>download_KEGG()</code> which accepts the prefix of a organism and returns a list of two data frames, one for the mapping between genes and pathways and the other for the full name of pathways.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/download_KEGG.html" class="external-link">download_KEGG</a></span><span class="op">(</span><span class="st">"hsa"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lt</span><span class="op">$</span><span class="va">KEGGPATHID2EXTID</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       from    to</span></span>
<span><span class="co">## 1 hsa00010 10327</span></span>
<span><span class="co">## 2 hsa00010   124</span></span>
<span><span class="co">## 3 hsa00010   125</span></span>
<span><span class="co">## 4 hsa00010   126</span></span>
<span><span class="co">## 5 hsa00010   127</span></span>
<span><span class="co">## 6 hsa00010   128</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lt</span><span class="op">$</span><span class="va">KEGGPATHID2NAME</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       from                              to</span></span>
<span><span class="co">## 1 hsa01100              Metabolic pathways</span></span>
<span><span class="co">## 2 hsa01200               Carbon metabolism</span></span>
<span><span class="co">## 3 hsa01210 2-Oxocarboxylic acid metabolism</span></span>
<span><span class="co">## 4 hsa01212           Fatty acid metabolism</span></span>
<span><span class="co">## 5 hsa01230     Biosynthesis of amino acids</span></span>
<span><span class="co">## 6 hsa01232           Nucleotide metabolism</span></span></code></pre>
<blockquote>
<p>Note: all the methods we introduced can not be used to retrieve the network data from KEGG.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>In the previous example, we obtained the mapping between genes and KEGG pathways for human, using the organism code of <code>"hsa"</code>. Since human is very well-studied and everyone knows <code>"hsa"</code> is for human, what if you are studying dogs? Try to find out the organism code for dog and get the gene-pathway mappings for dog.</p>
<p>Hint: use the <code>list</code> operator in KEGG API and search for dog.</p>
</div>
<div class="section level4">
<h4 id="solution">Solution<a class="anchor" aria-label="anchor" href="#solution"></a>
</h4>
<p>You can get organism codes for all organisms by <a href="https://rest.kegg.jp/list/organism" class="external-link uri">https://rest.kegg.jp/list/organism</a>. Then you will find the code for dog is <code>"cfa"</code>. Use one of the following three ways to download the data.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/link/pathway/cfa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">pathway2gene</span> <span class="op">=</span> <span class="fu">KEGGREST</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/KEGGREST/man/keggLink.html" class="external-link">keggLink</a></span><span class="op">(</span><span class="st">"pathway"</span>, <span class="st">"cfa"</span><span class="op">)</span></span>
<span><span class="va">lt</span> <span class="op">=</span> <span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/download_KEGG.html" class="external-link">download_KEGG</a></span><span class="op">(</span><span class="st">"cfa"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3 tabset">
<h3 class="tabset" id="section-1"></h3>
<div class="section level4">
<h4 id="practice-2">Practice 2<a class="anchor" aria-label="anchor" href="#practice-2"></a>
</h4>
<p>Make the distribution of the numbers of genes in KEGG pathways (for human).</p>
<p>There is a pathway that contains far more genes than all other pathways. What is the pathway ID of it and what is its full name? Go to the web page of this largest pathway to see its real structure.</p>
</div>
<div class="section level4">
<h4 id="solution-1">Solution<a class="anchor" aria-label="anchor" href="#solution-1"></a>
</h4>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/link/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">n_gene</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">df1</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">n_gene</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_02_kegg_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>Making the intervals smaller is better:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">n_gene</span>, nc <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p><img src="topic1_02_kegg_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>The largest pathway:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">n_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## path:hsa01100 </span></span>
<span><span class="co">##            86</span></span></code></pre>
<p>Its name:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">df2</span><span class="op">[</span><span class="va">df2</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">==</span> <span class="st">"hsa01100"</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##         V1                                        V2</span></span>
<span><span class="co">## 1 hsa01100 Metabolic pathways - Homo sapiens (human)</span></span></code></pre>
<p>And its url on KEGG: <a href="https://www.genome.jp/pathway/hsa01100" class="external-link uri">https://www.genome.jp/pathway/hsa01100</a>.</p>
</div>
</div>
<div class="section level3 tabset">
<h3 class="tabset" id="section-2"></h3>
<div class="section level4">
<h4 id="practice-3">Practice 3<a class="anchor" aria-label="anchor" href="#practice-3"></a>
</h4>
<p>Highlight the following genes “5527”, “7534”, “2033”, “1111”, “5885” on the pathway image of “Cell cycle”. Choose colors you like.</p>
</div>
<div class="section level4">
<h4 id="solution-2">Solution<a class="anchor" aria-label="anchor" href="#solution-2"></a>
</h4>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://rest.kegg.jp/list/pathway/hsa"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">df2</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">" - Homo .*$"</span>, <span class="st">""</span>, <span class="va">df2</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pathway_id</span> <span class="op">=</span> <span class="va">df2</span><span class="op">[</span><span class="va">df2</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Cell cycle"</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">pathway_id</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "hsa04110"</span></span></code></pre>
<p>Prepare the color table:</p>
<pre><code>5527    red,blue
7534    blue,red
2033    yello,darkgreen
1111    grey,red
5885    purple,blue</code></pre>
<p>Then go to <a href="https://www.genome.jp/pathway/hsa04110" class="external-link uri">https://www.genome.jp/pathway/hsa04110</a> and paste the color table there.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
