<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Topic 1-03: Get pathways from MSigDB • GSEAtraining</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Topic 1-03: Get pathways from MSigDB">
<meta property="og:description" content="GSEAtraining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GSEAtraining</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/topic1_00_format.html">Topic 1-00: Representation of gene sets in R</a>
    </li>
    <li>
      <a href="../articles/topic1_01_GO.html">Topic 1-01: Get GO gene sets</a>
    </li>
    <li>
      <a href="../articles/topic1_02_kegg.html">Topic 1-02: Get pathways from KEGG</a>
    </li>
    <li>
      <a href="../articles/topic1_03_msigdbr.html">Topic 1-03: Get pathways from MSigDB</a>
    </li>
    <li>
      <a href="../articles/topic1_04_Reactome.html">Topic 1-04: Get pathways from Reactome</a>
    </li>
    <li>
      <a href="../articles/topic1_05_BioCarta.html">Topic 1-05: Get pathways from BioCarta</a>
    </li>
    <li>
      <a href="../articles/topic1_06_UniProtKeywords.html">Topic 1-06: Get gene sets from UniProt Keywords</a>
    </li>
    <li>
      <a href="../articles/topic1_07_more_gene_sets.html">Topic 1-07: Get GO/KEGG gene sets for other organisms</a>
    </li>
    <li>
      <a href="../articles/topic1_08_ortholog.html">Topic 1-08: Generate gene sets for other organisms by mapping to orthologues</a>
    </li>
    <li>
      <a href="../articles/topic1_09_gene_id.html">Topic 1-09: Gene ID conversion</a>
    </li>
    <li>
      <a href="../articles/topic2_01_ora_single_gene_set.html">Topic 2-01: ORA on a single gene set</a>
    </li>
    <li>
      <a href="../articles/topic2_02_implement_ora.html">Topic 2-02: Implement ORA from stratch</a>
    </li>
    <li>
      <a href="../articles/topic2_03_clusterProfiler_ora.html">Topic 2-03: Use clusterProfiler for ORA</a>
    </li>
    <li>
      <a href="../articles/topic2_04_implement_gsea.html">Topic 2-04: Implement GSEA from scratch</a>
    </li>
    <li>
      <a href="../articles/topic2_05_clusterProfiler_gsea.html">Topic 2-05: Use clusterProfiler for GSEA</a>
    </li>
    <li>
      <a href="../articles/topic2_06_compare_ORA_GSEA.html">Topic 2-06: Compare ORA and GSEA</a>
    </li>
    <li>
      <a href="../articles/topic3_01_online_GREAT.html">Topic 3-01: Online GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_02_local_GREAT.html">Topic 3-02: Local GREAT analysis</a>
    </li>
    <li>
      <a href="../articles/topic3_03_goseq.html">Topic 3-03: GOseq</a>
    </li>
    <li>
      <a href="../articles/topic4_01_CePa.html">Topic 4-01: CePa</a>
    </li>
    <li>
      <a href="../articles/topic4_02_GSVA.html">Topic 4-02: GSVA</a>
    </li>
    <li>
      <a href="../articles/topic5_01_similarity.html">Topic 5-01: Similarities of terms</a>
    </li>
    <li>
      <a href="../articles/topic5_02_simplifyEnrichment.html">Topic 5-02: simplifyEnrichment</a>
    </li>
    <li>
      <a href="../articles/topic6_01_visualization.html">Topic 6-01: Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/GSEAtraining/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Topic 1-03: Get pathways from MSigDB</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>
</h4>
            
            <h4 data-toc-skip class="date">2024-01-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/GSEAtraining/blob/HEAD/vignettes/topic1_03_msigdbr.Rmd" class="external-link"><code>vignettes/topic1_03_msigdbr.Rmd</code></a></small>
      <div class="hidden name"><code>topic1_03_msigdbr.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="the--gmt-data-format-for-gene-sets">The .gmt data format for gene sets<a class="anchor" aria-label="anchor" href="#the--gmt-data-format-for-gene-sets"></a>
</h2>
<p>MSigDB defines a simple <code>.gmt</code> format for storing gene sets. It has the following format:</p>
<pre><code>gene_set_1  gene_set_description    gene1   gene2   gene3
gene_set_2  gene_set_description    gene4   gene5
...</code></pre>
<p>The <code>.gmt</code> format is used more and more for sharing gene set data, e.g. <a href="https://maayanlab.cloud/Enrichr/#libraries" class="external-link uri">https://maayanlab.cloud/Enrichr/#libraries</a>.</p>
<p>Let’s try to read a <code>.gmt</code> file. We use the <code>.gmt</code> file for the Hallmark gene set.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://data.broadinstitute.org/gsea-msigdb/msigdb/release/2023.2.Hs/h.all.v2023.2.Hs.symbols.gmt"</span>,</span>
<span>  destfile <span class="op">=</span> <span class="st">"h.all.v2023.2.Hs.symbols.gmt"</span><span class="op">)</span></span></code></pre></div>
<p>Since it is not a table, we have to read the file by lines.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ln</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readLines.html" class="external-link">readLines</a></span><span class="op">(</span><span class="st">"h.all.v2023.2.Hs.symbols.gmt"</span><span class="op">)</span></span>
<span><span class="va">ln</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">ln</span>, <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">gs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">ln</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gs</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ln</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">gs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $HALLMARK_ADIPOGENESIS</span></span>
<span><span class="co">##   [1] "ABCA1"    "ABCB8"    "ACAA2"    "ACADL"    "ACADM"    "ACADS"   </span></span>
<span><span class="co">##   [7] "ACLY"     "ACO2"     "ACOX1"    "ADCY6"    "ADIG"     "ADIPOQ"  </span></span>
<span><span class="co">##  [13] "ADIPOR2"  "AGPAT3"   "AIFM1"    "AK2"      "ALDH2"    "ALDOA"   </span></span>
<span><span class="co">##  [19] "ANGPT1"   "ANGPTL4"  "APLP2"    "APOE"     "ARAF"     "ARL4A"   </span></span>
<span><span class="co">##  [25] "ATL2"     "ATP1B3"   "ATP5PO"   "BAZ2A"    "BCKDHA"   "BCL2L13" </span></span>
<span><span class="co">##  [31] "BCL6"     "C3"       "CAT"      "CAVIN1"   "CAVIN2"   "CCNG2"   </span></span>
<span><span class="co">##  [37] "CD151"    "CD302"    "CD36"     "CDKN2C"   "CHCHD10"  "CHUK"    </span></span>
<span><span class="co">##  [43] "CIDEA"    "CMBL"     "CMPK1"    "COL15A1"  "COL4A1"   "COQ3"    </span></span>
<span><span class="co">##  [49] "COQ5"     "COQ9"     "COX6A1"   "COX7B"    "COX8A"    "CPT2"    </span></span>
<span><span class="co">##  [55] "CRAT"     "CS"       "CYC1"     "CYP4B1"   "DBT"      "DDT"     </span></span>
<span><span class="co">##  [61] "DECR1"    "DGAT1"    "DHCR7"    "DHRS7"    "DHRS7B"   "DLAT"    </span></span>
<span><span class="co">##  [67] "DLD"      "DNAJB9"   "DNAJC15"  "DRAM2"    "ECH1"     "ECHS1"   </span></span>
<span><span class="co">##  [73] "ELMOD3"   "ELOVL6"   "ENPP2"    "EPHX2"    "ESRRA"    "ESYT1"   </span></span>
<span><span class="co">##  [79] "ETFB"     "FABP4"    "FAH"      "FZD4"     "G3BP2"    "GADD45A" </span></span>
<span><span class="co">##  [85] "GBE1"     "GHITM"    "GPAM"     "GPAT4"    "GPD2"     "GPHN"    </span></span>
<span><span class="co">##  [91] "GPX3"     "GPX4"     "GRPEL1"   "HADH"     "HIBCH"    "HSPB8"   </span></span>
<span><span class="co">##  [97] "IDH1"     "IDH3A"    "IDH3G"    "IFNGR1"   "IMMT"     "ITGA7"   </span></span>
<span><span class="co">## [103] "ITIH5"    "ITSN1"    "JAGN1"    "LAMA4"    "LEP"      "LIFR"    </span></span>
<span><span class="co">## [109] "LIPE"     "LPCAT3"   "LPL"      "LTC4S"    "MAP4K3"   "MCCC1"   </span></span>
<span><span class="co">## [115] "MDH2"     "ME1"      "MGLL"     "MGST3"    "MIGA2"    "MRAP"    </span></span>
<span><span class="co">## [121] "MRPL15"   "MTARC2"   "MTCH2"    "MYLK"     "NABP1"    "NDUFA5"  </span></span>
<span><span class="co">## [127] "NDUFAB1"  "NDUFB7"   "NDUFS3"   "NKIRAS1"  "NMT1"     "OMD"     </span></span>
<span><span class="co">## [133] "ORM1"     "PDCD4"    "PEMT"     "PEX14"    "PFKFB3"   "PFKL"    </span></span>
<span><span class="co">## [139] "PGM1"     "PHLDB1"   "PHYH"     "PIM3"     "PLIN2"    "POR"     </span></span>
<span><span class="co">## [145] "PPARG"    "PPM1B"    "PPP1R15B" "PRDX3"    "PREB"     "PTCD3"   </span></span>
<span><span class="co">## [151] "PTGER3"   "QDPR"     "RAB34"    "REEP5"    "REEP6"    "RETN"    </span></span>
<span><span class="co">## [157] "RETSAT"   "RIOK3"    "RMDN3"    "RNF11"    "RREB1"    "RTN3"    </span></span>
<span><span class="co">## [163] "SAMM50"   "SCARB1"   "SCP2"     "SDHB"     "SDHC"     "SLC19A1" </span></span>
<span><span class="co">## [169] "SLC1A5"   "SLC25A1"  "SLC25A10" "SLC27A1"  "SLC5A6"   "SLC66A3" </span></span>
<span><span class="co">## [175] "SNCG"     "SOD1"     "SORBS1"   "SOWAHC"   "SPARCL1"  "SQOR"    </span></span>
<span><span class="co">## [181] "SSPN"     "STAT5A"   "STOM"     "SUCLG1"   "SULT1A1"  "TALDO1"  </span></span>
<span><span class="co">## [187] "TANK"     "TKT"      "TOB1"     "TST"      "UBC"      "UBQLN1"  </span></span>
<span><span class="co">## [193] "UCK1"     "UCP2"     "UQCR10"   "UQCR11"   "UQCRC1"   "UQCRQ"   </span></span>
<span><span class="co">## [199] "VEGFB"    "YWHAG"   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">##   [1] "AARS1"    "ABCE1"    "ABI1"     "ACHE"     "ACVR2A"   "AKT1"    </span></span>
<span><span class="co">##   [7] "APBB1"    "B2M"      "BCAT1"    "BCL10"    "BCL3"     "BRCA1"   </span></span>
<span><span class="co">##  [13] "C2"       "CAPG"     "CARTPT"   "CCL11"    "CCL13"    "CCL19"   </span></span>
<span><span class="co">##  [19] "CCL2"     "CCL22"    "CCL4"     "CCL5"     "CCL7"     "CCND2"   </span></span>
<span><span class="co">##  [25] "CCND3"    "CCR1"     "CCR2"     "CCR5"     "CD1D"     "CD2"     </span></span>
<span><span class="co">##  [31] "CD247"    "CD28"     "CD3D"     "CD3E"     "CD3G"     "CD4"     </span></span>
<span><span class="co">##  [37] "CD40"     "CD40LG"   "CD47"     "CD7"      "CD74"     "CD79A"   </span></span>
<span><span class="co">##  [43] "CD80"     "CD86"     "CD8A"     "CD8B"     "CD96"     "CDKN2A"  </span></span>
<span><span class="co">##  [49] "CFP"      "CRTAM"    "CSF1"     "CSK"      "CTSS"     "CXCL13"  </span></span>
<span><span class="co">##  [55] "CXCL9"    "CXCR3"    "DARS1"    "DEGS1"    "DYRK3"    "EGFR"    </span></span>
<span><span class="co">##  [61] "EIF3A"    "EIF3D"    "EIF3J"    "EIF4G3"   "EIF5A"    "ELANE"   </span></span>
<span><span class="co">##  [67] "ELF4"     "EREG"     "ETS1"     "F2"       "F2R"      "FAS"     </span></span>
<span><span class="co">##  [73] "FASLG"    "FCGR2B"   "FGR"      "FLNA"     "FYB1"     "GALNT1"  </span></span>
<span><span class="co">##  [79] "GBP2"     "GCNT1"    "GLMN"     "GPR65"    "GZMA"     "GZMB"    </span></span>
<span><span class="co">##  [85] "HCLS1"    "HDAC9"    "HIF1A"    "HLA-A"    "HLA-DMA"  "HLA-DMB" </span></span>
<span><span class="co">##  [91] "HLA-DOA"  "HLA-DOB"  "HLA-DQA1" "HLA-DRA"  "HLA-E"    "HLA-G"   </span></span>
<span><span class="co">##  [97] "ICAM1"    "ICOSLG"   "IFNAR2"   "IFNG"     "IFNGR1"   "IFNGR2"  </span></span>
<span><span class="co">## [103] "IGSF6"    "IKBKB"    "IL10"     "IL11"     "IL12A"    "IL12B"   </span></span>
<span><span class="co">## [109] "IL12RB1"  "IL13"     "IL15"     "IL16"     "IL18"     "IL18RAP" </span></span>
<span><span class="co">## [115] "IL1B"     "IL2"      "IL27RA"   "IL2RA"    "IL2RB"    "IL2RG"   </span></span>
<span><span class="co">## [121] "IL4"      "IL4R"     "IL6"      "IL7"      "IL9"      "INHBA"   </span></span>
<span><span class="co">## [127] "INHBB"    "IRF4"     "IRF7"     "IRF8"     "ITGAL"    "ITGB2"   </span></span>
<span><span class="co">## [133] "ITK"      "JAK2"     "KLRD1"    "KRT1"     "LCK"      "LCP2"    </span></span>
<span><span class="co">## [139] "LIF"      "LTB"      "LY75"     "LY86"     "LYN"      "MAP3K7"  </span></span>
<span><span class="co">## [145] "MAP4K1"   "MBL2"     "MMP9"     "MRPL3"    "MTIF2"    "NCF4"    </span></span>
<span><span class="co">## [151] "NCK1"     "NCR1"     "NLRP3"    "NME1"     "NOS2"     "NPM1"    </span></span>
<span><span class="co">## [157] "PF4"      "PRF1"     "PRKCB"    "PRKCG"    "PSMB10"   "PTPN6"   </span></span>
<span><span class="co">## [163] "PTPRC"    "RARS1"    "RIPK2"    "RPL39"    "RPL3L"    "RPL9"    </span></span>
<span><span class="co">## [169] "RPS19"    "RPS3A"    "RPS9"     "SIT1"     "SOCS1"    "SOCS5"   </span></span>
<span><span class="co">## [175] "SPI1"     "SRGN"     "ST8SIA4"  "STAB1"    "STAT1"    "STAT4"   </span></span>
<span><span class="co">## [181] "TAP1"     "TAP2"     "TAPBP"    "TGFB1"    "TGFB2"    "THY1"    </span></span>
<span><span class="co">## [187] "TIMP1"    "TLR1"     "TLR2"     "TLR3"     "TLR6"     "TNF"     </span></span>
<span><span class="co">## [193] "TPD52"    "TRAF2"    "TRAT1"    "UBE2D1"   "UBE2N"    "WARS1"   </span></span>
<span><span class="co">## [199] "WAS"      "ZAP70"</span></span></code></pre>
<p>In <strong>rGREAT</strong> package, there is a helper function <code><a href="https://rdrr.io/pkg/rGREAT/man/read_gmt.html" class="external-link">read_gmt()</a></code> which can read gene sets from <code>.gmt</code> file and also does gene ID conversion.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/rGREAT" class="external-link">rGREAT</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span></span>
<span><span class="co">##     as.data.frame, basename, cbind, colnames, dirname, do.call,</span></span>
<span><span class="co">##     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span></span>
<span><span class="co">##     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span></span>
<span><span class="co">##     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span></span>
<span><span class="co">##     tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'S4Vectors' was built under R version 4.3.2</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     I, expand.grid, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## ========================================</span></span>
<span><span class="co">## rGREAT version 2.5.1</span></span>
<span><span class="co">## Bioconductor page: http://bioconductor.org/packages/rGREAT/</span></span>
<span><span class="co">## Github page: https://github.com/jokergoo/rGREAT</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use it in published research, please cite:</span></span>
<span><span class="co">## Gu, Z. rGREAT: an R/Bioconductor package for functional enrichment </span></span>
<span><span class="co">##   on genomic regions. Bioinformatics 2023.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## This message can be suppressed by:</span></span>
<span><span class="co">##   suppressPackageStartupMessages(library(rGREAT))</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Note: On Aug 19 2019 GREAT released version 4 where it supports `hg38`</span></span>
<span><span class="co">## genome and removes some ontologies such pathways. `submitGreatJob()`</span></span>
<span><span class="co">## still takes `hg19` as default. `hg38` can be specified by the `species</span></span>
<span><span class="co">## = 'hg38'` argument. To use the older versions such as 3.0.0, specify as</span></span>
<span><span class="co">## `submitGreatJob(..., version = '3.0.0')`.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## From rGREAT version 1.99.0, it also implements the GREAT algorithm and</span></span>
<span><span class="co">## it allows to integrate GREAT analysis with the Bioconductor annotation</span></span>
<span><span class="co">## ecosystem. By default it supports more than 500 organisms. Check the</span></span>
<span><span class="co">## new function `great()` and the new vignette.</span></span>
<span><span class="co">## ========================================</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/read_gmt.html" class="external-link">read_gmt</a></span><span class="op">(</span><span class="st">"h.all.v2023.2.Hs.symbols.gmt"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gs2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/read_gmt.html" class="external-link">read_gmt</a></span><span class="op">(</span><span class="st">"h.all.v2023.2.Hs.symbols.gmt"</span>, </span>
<span>    from <span class="op">=</span> <span class="st">"SYMBOL"</span>, to <span class="op">=</span> <span class="st">"ENSEMBL"</span>, orgdb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="va">gs2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $HALLMARK_ADIPOGENESIS</span></span>
<span><span class="co">##   [1] "ENSG00000165029" "ENSG00000197150" "ENSG00000167315" "ENSG00000115361"</span></span>
<span><span class="co">##   [5] "ENSG00000117054" "ENSG00000122971" "ENSG00000131473" "ENSG00000100412"</span></span>
<span><span class="co">##   [9] "ENSG00000161533" "ENSG00000174233" "ENSG00000182035" "ENSG00000181092"</span></span>
<span><span class="co">##  [13] "ENSG00000160216" "ENSG00000156709" "ENSG00000004455" "ENSG00000111275"</span></span>
<span><span class="co">##  [17] "ENSG00000149925" "ENSG00000154188" "ENSG00000167772" "ENSG00000084234"</span></span>
<span><span class="co">##  [21] "ENSG00000130203" "ENSG00000078061" "ENSG00000122644" "ENSG00000119787"</span></span>
<span><span class="co">##  [25] "ENSG00000069849" "ENSG00000241837" "ENSG00000076108" "ENSG00000248098"</span></span>
<span><span class="co">##  [29] "ENSG00000099968" "ENSG00000113916" "ENSG00000125730" "ENSG00000121691"</span></span>
<span><span class="co">##  [33] "ENSG00000177469" "ENSG00000168497" "ENSG00000138764" "ENSG00000177697"</span></span>
<span><span class="co">##  [37] "ENSG00000241399" "ENSG00000135218" "ENSG00000123080" "ENSG00000213341"</span></span>
<span><span class="co">##  [41] "ENSG00000176194" "ENSG00000164237" "ENSG00000162368" "ENSG00000204291"</span></span>
<span><span class="co">##  [45] "ENSG00000187498" "ENSG00000132423" "ENSG00000110871" "ENSG00000088682"</span></span>
<span><span class="co">##  [49] "ENSG00000111775" "ENSG00000131174" "ENSG00000176340" "ENSG00000157184"</span></span>
<span><span class="co">##  [53] "ENSG00000095321" "ENSG00000062485" "ENSG00000179091" "ENSG00000142973"</span></span>
<span><span class="co">##  [57] "ENSG00000137992" "ENSG00000104325" "ENSG00000172893" "ENSG00000100612"</span></span>
<span><span class="co">##  [61] "ENSG00000109016" "ENSG00000150768" "ENSG00000091140" "ENSG00000128590"</span></span>
<span><span class="co">##  [65] "ENSG00000120675" "ENSG00000156171" "ENSG00000127884" "ENSG00000115459"</span></span>
<span><span class="co">##  [69] "ENSG00000170522" "ENSG00000136960" "ENSG00000120915" "ENSG00000173153"</span></span>
<span><span class="co">##  [73] "ENSG00000139641" "ENSG00000105379" "ENSG00000170323" "ENSG00000103876"</span></span>
<span><span class="co">##  [77] "ENSG00000174804" "ENSG00000138757" "ENSG00000116717" "ENSG00000114480"</span></span>
<span><span class="co">##  [81] "ENSG00000165678" "ENSG00000119927" "ENSG00000158669" "ENSG00000115159"</span></span>
<span><span class="co">##  [85] "ENSG00000171723" "ENSG00000211445" "ENSG00000167468" "ENSG00000109519"</span></span>
<span><span class="co">##  [89] "ENSG00000138796" "ENSG00000198130" "ENSG00000152137" "ENSG00000138413"</span></span>
<span><span class="co">##  [93] "ENSG00000166411" "ENSG00000067829" "ENSG00000027697" "ENSG00000132305"</span></span>
<span><span class="co">##  [97] "ENSG00000135424" "ENSG00000123243" "ENSG00000205726" "ENSG00000171135"</span></span>
<span><span class="co">## [101] "ENSG00000112769" "ENSG00000174697" "ENSG00000113594" "ENSG00000079435"</span></span>
<span><span class="co">## [105] "ENSG00000111684" "ENSG00000175445" "ENSG00000011566" "ENSG00000078070"</span></span>
<span><span class="co">## [109] "ENSG00000146701" "ENSG00000065833" "ENSG00000074416" "ENSG00000143198"</span></span>
<span><span class="co">## [113] "ENSG00000148343" "ENSG00000170262" "ENSG00000137547" "ENSG00000117791"</span></span>
<span><span class="co">## [117] "ENSG00000065534" "ENSG00000173559" "ENSG00000128609" "ENSG00000004779"</span></span>
<span><span class="co">## [121] "ENSG00000099795" "ENSG00000197885" "ENSG00000136448" "ENSG00000127083"</span></span>
<span><span class="co">## [125] "ENSG00000229314" "ENSG00000150593" "ENSG00000133027" "ENSG00000142655"</span></span>
<span><span class="co">## [129] "ENSG00000170525" "ENSG00000141959" "ENSG00000079739" "ENSG00000019144"</span></span>
<span><span class="co">## [133] "ENSG00000107537" "ENSG00000198355" "ENSG00000147872" "ENSG00000127948"</span></span>
<span><span class="co">## [137] "ENSG00000132170" "ENSG00000138032" "ENSG00000158615" "ENSG00000165672"</span></span>
<span><span class="co">## [141] "ENSG00000138073" "ENSG00000132300" "ENSG00000050628" "ENSG00000151552"</span></span>
<span><span class="co">## [145] "ENSG00000109113" "ENSG00000129625" "ENSG00000115255" "ENSG00000104918"</span></span>
<span><span class="co">## [149] "ENSG00000042445" "ENSG00000101782" "ENSG00000137824" "ENSG00000123091"</span></span>
<span><span class="co">## [153] "ENSG00000124782" "ENSG00000133318" "ENSG00000100347" "ENSG00000073060"</span></span>
<span><span class="co">## [157] "ENSG00000116171" "ENSG00000117118" "ENSG00000143252" "ENSG00000173638"</span></span>
<span><span class="co">## [161] "ENSG00000105281" "ENSG00000100075" "ENSG00000183048" "ENSG00000130304"</span></span>
<span><span class="co">## [165] "ENSG00000138074" "ENSG00000162976" "ENSG00000173267" "ENSG00000142168"</span></span>
<span><span class="co">## [169] "ENSG00000095637" "ENSG00000198142" "ENSG00000152583" "ENSG00000137767"</span></span>
<span><span class="co">## [173] "ENSG00000123096" "ENSG00000126561" "ENSG00000148175" "ENSG00000163541"</span></span>
<span><span class="co">## [177] "ENSG00000177156" "ENSG00000136560" "ENSG00000163931" "ENSG00000141232"</span></span>
<span><span class="co">## [181] "ENSG00000128311" "ENSG00000150991" "ENSG00000135018" "ENSG00000130717"</span></span>
<span><span class="co">## [185] "ENSG00000175567" "ENSG00000184076" "ENSG00000127540" "ENSG00000010256"</span></span>
<span><span class="co">## [189] "ENSG00000164405" "ENSG00000173511" "ENSG00000170027"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">##   [1] "ENSG00000090861" "ENSG00000164163" "ENSG00000136754" "ENSG00000087085"</span></span>
<span><span class="co">##   [5] "ENSG00000121989" "ENSG00000142208" "ENSG00000166313" "ENSG00000060982"</span></span>
<span><span class="co">##   [9] "ENSG00000142867" "ENSG00000069399" "ENSG00000012048" "ENSG00000042493"</span></span>
<span><span class="co">##  [13] "ENSG00000164326" "ENSG00000172156" "ENSG00000181374" "ENSG00000172724"</span></span>
<span><span class="co">##  [17] "ENSG00000108691" "ENSG00000102962" "ENSG00000108688" "ENSG00000118971"</span></span>
<span><span class="co">##  [21] "ENSG00000112576" "ENSG00000163823" "ENSG00000121807" "ENSG00000160791"</span></span>
<span><span class="co">##  [25] "ENSG00000158473" "ENSG00000116824" "ENSG00000198821" "ENSG00000178562"</span></span>
<span><span class="co">##  [29] "ENSG00000167286" "ENSG00000198851" "ENSG00000160654" "ENSG00000010610"</span></span>
<span><span class="co">##  [33] "ENSG00000101017" "ENSG00000102245" "ENSG00000196776" "ENSG00000173762"</span></span>
<span><span class="co">##  [37] "ENSG00000019582" "ENSG00000105369" "ENSG00000121594" "ENSG00000114013"</span></span>
<span><span class="co">##  [41] "ENSG00000153563" "ENSG00000172116" "ENSG00000153283" "ENSG00000147889"</span></span>
<span><span class="co">##  [45] "ENSG00000126759" "ENSG00000109943" "ENSG00000184371" "ENSG00000103653"</span></span>
<span><span class="co">##  [49] "ENSG00000163131" "ENSG00000156234" "ENSG00000138755" "ENSG00000186810"</span></span>
<span><span class="co">##  [53] "ENSG00000115866" "ENSG00000143753" "ENSG00000143479" "ENSG00000146648"</span></span>
<span><span class="co">##  [57] "ENSG00000170099" "ENSG00000100353" "ENSG00000104131" "ENSG00000075151"</span></span>
<span><span class="co">##  [61] "ENSG00000204472" "ENSG00000102034" "ENSG00000124882" "ENSG00000134954"</span></span>
<span><span class="co">##  [65] "ENSG00000180210" "ENSG00000181104" "ENSG00000026103" "ENSG00000117560"</span></span>
<span><span class="co">##  [69] "ENSG00000072694" "ENSG00000000938" "ENSG00000196924" "ENSG00000082074"</span></span>
<span><span class="co">##  [73] "ENSG00000141429" "ENSG00000162645" "ENSG00000187210" "ENSG00000174842"</span></span>
<span><span class="co">##  [77] "ENSG00000140030" "ENSG00000145649" "ENSG00000100453" "ENSG00000180353"</span></span>
<span><span class="co">##  [81] "ENSG00000048052" "ENSG00000100644" "ENSG00000090339" "ENSG00000160223"</span></span>
<span><span class="co">##  [85] "ENSG00000159110" "ENSG00000111537" "ENSG00000027697" "ENSG00000104365"</span></span>
<span><span class="co">##  [89] "ENSG00000136634" "ENSG00000095752" "ENSG00000168811" "ENSG00000113302"</span></span>
<span><span class="co">##  [93] "ENSG00000096996" "ENSG00000169194" "ENSG00000164136" "ENSG00000172349"</span></span>
<span><span class="co">##  [97] "ENSG00000150782" "ENSG00000115607" "ENSG00000125538" "ENSG00000109471"</span></span>
<span><span class="co">## [101] "ENSG00000134460" "ENSG00000100385" "ENSG00000147168" "ENSG00000113520"</span></span>
<span><span class="co">## [105] "ENSG00000077238" "ENSG00000136244" "ENSG00000104432" "ENSG00000145839"</span></span>
<span><span class="co">## [109] "ENSG00000122641" "ENSG00000163083" "ENSG00000137265" "ENSG00000140968"</span></span>
<span><span class="co">## [113] "ENSG00000005844" "ENSG00000160255" "ENSG00000113263" "ENSG00000096968"</span></span>
<span><span class="co">## [117] "ENSG00000134539" "ENSG00000167768" "ENSG00000182866" "ENSG00000043462"</span></span>
<span><span class="co">## [121] "ENSG00000128342" "ENSG00000054219" "ENSG00000112799" "ENSG00000254087"</span></span>
<span><span class="co">## [125] "ENSG00000135341" "ENSG00000165471" "ENSG00000100985" "ENSG00000114686"</span></span>
<span><span class="co">## [129] "ENSG00000085760" "ENSG00000158092" "ENSG00000162711" "ENSG00000239672"</span></span>
<span><span class="co">## [133] "ENSG00000007171" "ENSG00000181163" "ENSG00000163737" "ENSG00000180644"</span></span>
<span><span class="co">## [137] "ENSG00000166501" "ENSG00000126583" "ENSG00000205220" "ENSG00000111679"</span></span>
<span><span class="co">## [141] "ENSG00000113643" "ENSG00000104312" "ENSG00000198918" "ENSG00000140986"</span></span>
<span><span class="co">## [145] "ENSG00000163682" "ENSG00000105372" "ENSG00000145425" "ENSG00000137078"</span></span>
<span><span class="co">## [149] "ENSG00000185338" "ENSG00000171150" "ENSG00000066336" "ENSG00000122862"</span></span>
<span><span class="co">## [153] "ENSG00000113532" "ENSG00000010327" "ENSG00000115415" "ENSG00000138378"</span></span>
<span><span class="co">## [157] "ENSG00000105329" "ENSG00000092969" "ENSG00000154096" "ENSG00000102265"</span></span>
<span><span class="co">## [161] "ENSG00000174125" "ENSG00000137462" "ENSG00000164342" "ENSG00000174130"</span></span>
<span><span class="co">## [165] "ENSG00000076554" "ENSG00000127191" "ENSG00000163519" "ENSG00000072401"</span></span>
<span><span class="co">## [169] "ENSG00000177889" "ENSG00000140105" "ENSG00000015285" "ENSG00000115085"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="the-msigdbr-package">The msigdbr package<a class="anchor" aria-label="anchor" href="#the-msigdbr-package"></a>
</h2>
<p>There is already an R package <strong>msigdbr</strong> which has integerated all MSigDB gene sets.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span></code></pre></div>
<p>The major advantage is, MSigDB only supports human and mouse, <strong>msigdbr</strong> supports more organisms by mapping to the orthologe genes.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_species.html" class="external-link">msigdbr_species</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 20 × 2</span></span></span>
<span><span class="co">##    species_name                    species_common_name                          </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Anolis carolinensis             Carolina anole, green anole                  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Bos taurus                      bovine, cattle, cow, dairy cow, domestic cat…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Caenorhabditis elegans          <span style="color: #BB0000;">NA</span>                                           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Canis lupus familiaris          dog, dogs                                    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> Danio rerio                     leopard danio, zebra danio, zebra fish, zebr…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Drosophila melanogaster         fruit fly                                    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Equus caballus                  domestic horse, equine, horse                </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Felis catus                     cat, cats, domestic cat                      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Gallus gallus                   bantam, chicken, chickens, Gallus domesticus </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Homo sapiens                    human                                        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> Macaca mulatta                  rhesus macaque, rhesus macaques, Rhesus monk…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> Monodelphis domestica           gray short-tailed opossum                    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">13</span> Mus musculus                    house mouse, mouse                           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">14</span> Ornithorhynchus anatinus        duck-billed platypus, duckbill platypus, pla…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">15</span> Pan troglodytes                 chimpanzee                                   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">16</span> Rattus norvegicus               brown rat, Norway rat, rat, rats             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">17</span> Saccharomyces cerevisiae        baker's yeast, brewer's yeast, S. cerevisiae </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">18</span> Schizosaccharomyces pombe 972h- <span style="color: #BB0000;">NA</span>                                           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">19</span> Sus scrofa                      pig, pigs, swine, wild boar                  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">20</span> Xenopus tropicalis              tropical clawed frog, western clawed frog</span></span></code></pre>
<p>All categories of gene sets:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_collections.html" class="external-link">msigdbr_collections</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    gs_cat       gs_subcat num_genesets</span></span>
<span><span class="co">## 1      C1                          299</span></span>
<span><span class="co">## 2      C2             CGP         3384</span></span>
<span><span class="co">## 3      C2              CP           29</span></span>
<span><span class="co">## 4      C2     CP:BIOCARTA          292</span></span>
<span><span class="co">## 5      C2         CP:KEGG          186</span></span>
<span><span class="co">## 6      C2          CP:PID          196</span></span>
<span><span class="co">## 7      C2     CP:REACTOME         1615</span></span>
<span><span class="co">## 8      C2 CP:WIKIPATHWAYS          664</span></span>
<span><span class="co">## 9      C3       MIR:MIRDB         2377</span></span>
<span><span class="co">## 10     C3  MIR:MIR_Legacy          221</span></span>
<span><span class="co">## 11     C3        TFT:GTRD          518</span></span>
<span><span class="co">## 12     C3  TFT:TFT_Legacy          610</span></span>
<span><span class="co">## 13     C4             CGN          427</span></span>
<span><span class="co">## 14     C4              CM          431</span></span>
<span><span class="co">## 15     C5           GO:BP         7658</span></span>
<span><span class="co">## 16     C5           GO:CC         1006</span></span>
<span><span class="co">## 17     C5           GO:MF         1738</span></span>
<span><span class="co">## 18     C5             HPO         5071</span></span>
<span><span class="co">## 19     C6                          189</span></span>
<span><span class="co">## 20     C7     IMMUNESIGDB         4872</span></span>
<span><span class="co">## 21     C7             VAX          347</span></span>
<span><span class="co">## 22     C8                          700</span></span>
<span><span class="co">## 23      H                           50</span></span></code></pre>
<p>Values in the <code>gs_cat</code> column and the <code>gs_subcat</code> column can be used to retrieve gene sets in a specific category. E.g., we want to extract genesets in C2 category and CP:KEGG sub-category:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"C2"</span>, subcategory <span class="op">=</span> <span class="st">"CP:KEGG"</span><span class="op">)</span></span>
<span><span class="va">gene_sets</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 16,283 × 15</span></span></span>
<span><span class="co">##    gs_cat gs_subcat gs_name               gene_symbol entrez_gene ensembl_gene  </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA1                19 ENSG000001650…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA10            <span style="text-decoration: underline;">10</span>349 ENSG000001542…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA12            <span style="text-decoration: underline;">26</span>154 ENSG000001444…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA13           <span style="text-decoration: underline;">154</span>664 ENSG000001798…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA2                20 ENSG000001073…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA3                21 ENSG000001679…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA4                24 ENSG000001986…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA5             <span style="text-decoration: underline;">23</span>461 ENSG000001542…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA6             <span style="text-decoration: underline;">23</span>460 ENSG000001542…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA7             <span style="text-decoration: underline;">10</span>347 ENSG000000646…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 16,273 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 9 more variables: human_gene_symbol &lt;chr&gt;, human_entrez_gene &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   human_ensembl_gene &lt;chr&gt;, gs_id &lt;chr&gt;, gs_pmid &lt;chr&gt;, gs_geoid &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;, gs_description &lt;chr&gt;</span></span></span></code></pre>
<p>If you don’t like the <code>tibble</code> format, you can change it to a data frame.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gene_sets</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_sets</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   gs_cat gs_subcat               gs_name gene_symbol entrez_gene</span></span>
<span><span class="co">## 1     C2   CP:KEGG KEGG_ABC_TRANSPORTERS       ABCA1          19</span></span>
<span><span class="co">## 2     C2   CP:KEGG KEGG_ABC_TRANSPORTERS      ABCA10       10349</span></span>
<span><span class="co">## 3     C2   CP:KEGG KEGG_ABC_TRANSPORTERS      ABCA12       26154</span></span>
<span><span class="co">## 4     C2   CP:KEGG KEGG_ABC_TRANSPORTERS      ABCA13      154664</span></span>
<span><span class="co">## 5     C2   CP:KEGG KEGG_ABC_TRANSPORTERS       ABCA2          20</span></span>
<span><span class="co">## 6     C2   CP:KEGG KEGG_ABC_TRANSPORTERS       ABCA3          21</span></span>
<span><span class="co">##      ensembl_gene human_gene_symbol human_entrez_gene human_ensembl_gene  gs_id</span></span>
<span><span class="co">## 1 ENSG00000165029             ABCA1                19    ENSG00000165029 M11911</span></span>
<span><span class="co">## 2 ENSG00000154263            ABCA10             10349    ENSG00000154263 M11911</span></span>
<span><span class="co">## 3 ENSG00000144452            ABCA12             26154    ENSG00000144452 M11911</span></span>
<span><span class="co">## 4 ENSG00000179869            ABCA13            154664    ENSG00000179869 M11911</span></span>
<span><span class="co">## 5 ENSG00000107331             ABCA2                20    ENSG00000107331 M11911</span></span>
<span><span class="co">## 6 ENSG00000167972             ABCA3                21    ENSG00000167972 M11911</span></span>
<span><span class="co">##   gs_pmid gs_geoid gs_exact_source</span></span>
<span><span class="co">## 1                         hsa02010</span></span>
<span><span class="co">## 2                         hsa02010</span></span>
<span><span class="co">## 3                         hsa02010</span></span>
<span><span class="co">## 4                         hsa02010</span></span>
<span><span class="co">## 5                         hsa02010</span></span>
<span><span class="co">## 6                         hsa02010</span></span>
<span><span class="co">##                                                gs_url   gs_description</span></span>
<span><span class="co">## 1 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters</span></span>
<span><span class="co">## 2 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters</span></span>
<span><span class="co">## 3 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters</span></span>
<span><span class="co">## 4 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters</span></span>
<span><span class="co">## 5 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters</span></span>
<span><span class="co">## 6 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters</span></span></code></pre>
<style>
blockquote {
    padding: 10px 20px;
    margin: 0 0 20px;
    font-size: 14px;
    background: #eee;
    border-left: 6px #CCCCCC solid;
}
</style>
<blockquote>
<p><strong>Danger zone:</strong> The <code>entrez_gene</code> column is saved in integer. This may cause unseeable problems. Convert it explicitely to character: <code>gene_set$entrez_gene = as.character(gene_set$entrez_gene)</code>.</p>
</blockquote>
<p>Another advantage of <strong>misgdbr</strong> is that it provides all three major gene ID types. For example, if you need Ensembl gene IDs, you can obtain the gene sets by:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sets</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"ensembl_gene"</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 gs_name    ensembl_gene</span></span>
<span><span class="co">## 1 KEGG_ABC_TRANSPORTERS ENSG00000165029</span></span>
<span><span class="co">## 2 KEGG_ABC_TRANSPORTERS ENSG00000154263</span></span>
<span><span class="co">## 3 KEGG_ABC_TRANSPORTERS ENSG00000144452</span></span>
<span><span class="co">## 4 KEGG_ABC_TRANSPORTERS ENSG00000179869</span></span>
<span><span class="co">## 5 KEGG_ABC_TRANSPORTERS ENSG00000107331</span></span>
<span><span class="co">## 6 KEGG_ABC_TRANSPORTERS ENSG00000167972</span></span></code></pre>
<p>By default it retrieves data for human. The <code>species</code> argument can be used for other supported species.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"dog"</span>, category <span class="op">=</span> <span class="st">"C2"</span>, subcategory <span class="op">=</span> <span class="st">"CP:BIOCARTA"</span><span class="op">)</span></span>
<span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gene_sets</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_sets</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   gs_cat   gs_subcat               gs_name gene_symbol entrez_gene</span></span>
<span><span class="co">## 1     C2 CP:BIOCARTA BIOCARTA_41BB_PATHWAY        ATF2      478806</span></span>
<span><span class="co">## 2     C2 CP:BIOCARTA BIOCARTA_41BB_PATHWAY        CHUK      477796</span></span>
<span><span class="co">## 3     C2 CP:BIOCARTA BIOCARTA_41BB_PATHWAY        IFNG      403801</span></span>
<span><span class="co">## 4     C2 CP:BIOCARTA BIOCARTA_41BB_PATHWAY       IKBKB      482839</span></span>
<span><span class="co">## 5     C2 CP:BIOCARTA BIOCARTA_41BB_PATHWAY         IL2      403989</span></span>
<span><span class="co">## 6     C2 CP:BIOCARTA BIOCARTA_41BB_PATHWAY         IL4      403785</span></span>
<span><span class="co">##         ensembl_gene human_gene_symbol human_entrez_gene human_ensembl_gene</span></span>
<span><span class="co">## 1 ENSCAFG00845017708              ATF2              1386    ENSG00000115966</span></span>
<span><span class="co">## 2 ENSCAFG00845016257              CHUK              1147    ENSG00000213341</span></span>
<span><span class="co">## 3 ENSCAFG00845013680              IFNG              3458    ENSG00000111537</span></span>
<span><span class="co">## 4 ENSCAFG00845006197             IKBKB              3551    ENSG00000104365</span></span>
<span><span class="co">## 5 ENSCAFG00845020168               IL2              3558    ENSG00000109471</span></span>
<span><span class="co">## 6 ENSCAFG00845010991               IL4              3565    ENSG00000113520</span></span>
<span><span class="co">##   gs_id gs_pmid gs_geoid gs_exact_source</span></span>
<span><span class="co">## 1 M2064                                 </span></span>
<span><span class="co">## 2 M2064                                 </span></span>
<span><span class="co">## 3 M2064                                 </span></span>
<span><span class="co">## 4 M2064                                 </span></span>
<span><span class="co">## 5 M2064                                 </span></span>
<span><span class="co">## 6 M2064                                 </span></span>
<span><span class="co">##                                                                                gs_url</span></span>
<span><span class="co">## 1 https://data.broadinstitute.org/gsea-msigdb/msigdb/biocarta/human/h_41BBPathway.gif</span></span>
<span><span class="co">## 2 https://data.broadinstitute.org/gsea-msigdb/msigdb/biocarta/human/h_41BBPathway.gif</span></span>
<span><span class="co">## 3 https://data.broadinstitute.org/gsea-msigdb/msigdb/biocarta/human/h_41BBPathway.gif</span></span>
<span><span class="co">## 4 https://data.broadinstitute.org/gsea-msigdb/msigdb/biocarta/human/h_41BBPathway.gif</span></span>
<span><span class="co">## 5 https://data.broadinstitute.org/gsea-msigdb/msigdb/biocarta/human/h_41BBPathway.gif</span></span>
<span><span class="co">## 6 https://data.broadinstitute.org/gsea-msigdb/msigdb/biocarta/human/h_41BBPathway.gif</span></span>
<span><span class="co">##                        gs_description taxon_id</span></span>
<span><span class="co">## 1 The 4-1BB-dependent immune response     9615</span></span>
<span><span class="co">## 2 The 4-1BB-dependent immune response     9615</span></span>
<span><span class="co">## 3 The 4-1BB-dependent immune response     9615</span></span>
<span><span class="co">## 4 The 4-1BB-dependent immune response     9615</span></span>
<span><span class="co">## 5 The 4-1BB-dependent immune response     9615</span></span>
<span><span class="co">## 6 The 4-1BB-dependent immune response     9615</span></span>
<span><span class="co">##                             ortholog_sources num_ortholog_sources</span></span>
<span><span class="co">## 1 Ensembl|HomoloGene|Inparanoid|NCBI|OrthoDB                    5</span></span>
<span><span class="co">## 2 Ensembl|HomoloGene|Inparanoid|NCBI|OrthoDB                    5</span></span>
<span><span class="co">## 3 Ensembl|HomoloGene|Inparanoid|NCBI|OrthoDB                    5</span></span>
<span><span class="co">## 4 Ensembl|HomoloGene|Inparanoid|NCBI|OrthoDB                    5</span></span>
<span><span class="co">## 5         Ensembl|HomoloGene|Inparanoid|NCBI                    4</span></span>
<span><span class="co">## 6 Ensembl|HomoloGene|Inparanoid|NCBI|OrthoDB                    5</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="a-safer-check">A safer check<a class="anchor" aria-label="anchor" href="#a-safer-check"></a>
</h2>
<p>The table contains mappings between gene sets and all the three gene ID types. Since Symbol-EntreZ-Ensembl mapping is not 1:1, it is possible rows are duplicated if only taking two columns from the complete table:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sets</span> <span class="op">=</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"GO:BP"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="va">gene_sets</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"gene_symbol"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 721379      2</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 633008      2</span></span></code></pre>
<p>So, additionally applying <code><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique()</a></code> is always safe.</p>
</div>
<div class="section level2">
<h2 id="functions-in-the-gseatraining-package">Functions in the GSEAtraining package<a class="anchor" aria-label="anchor" href="#functions-in-the-gseatraining-package"></a>
</h2>
<p>We also have several functions in <strong>GSEAtraining</strong> to retrieve gene sets from MSigDB. It directly downloads data from MSigDB web server.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/GSEAtraining" class="external-link">GSEAtraining</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/msigdb.html">list_msigdb_versions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "6.0"       "6.1"       "6.2"       "7.0"       "7.1"       "7.2"      </span></span>
<span><span class="co">##  [7] "7.3"       "7.4"       "7.5.1"     "7.5"       "2022.1.Hs" "2022.1.Mm"</span></span>
<span><span class="co">## [13] "2023.1.Hs" "2023.1.Mm" "2023.2.Hs" "2023.2.Mm"</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/msigdb.html">list_msigdb_collections</a></span><span class="op">(</span><span class="st">"2023.2.Hs"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "c1.all"             "c2.all"             "c2.cgp"            </span></span>
<span><span class="co">##  [4] "c2.cp.biocarta"     "c2.cp.kegg_legacy"  "c2.cp.kegg_medicus"</span></span>
<span><span class="co">##  [7] "c2.cp.pid"          "c2.cp.reactome"     "c2.cp"             </span></span>
<span><span class="co">## [10] "c2.cp.wikipathways" "c3.all"             "c3.mir.mir_legacy" </span></span>
<span><span class="co">## [13] "c3.mir.mirdb"       "c3.mir"             "c3.tft.gtrd"       </span></span>
<span><span class="co">## [16] "c3.tft.tft_legacy"  "c3.tft"             "c4.3ca"            </span></span>
<span><span class="co">## [19] "c4.all"             "c4.cgn"             "c4.cm"             </span></span>
<span><span class="co">## [22] "c5.all"             "c5.go.bp"           "c5.go.cc"          </span></span>
<span><span class="co">## [25] "c5.go.mf"           "c5.go"              "c5.hpo"            </span></span>
<span><span class="co">## [28] "c6.all"             "c7.all"             "c7.immunesigdb"    </span></span>
<span><span class="co">## [31] "c7.vax"             "c8.all"             "h.all"</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">=</span> <span class="fu"><a href="../reference/msigdb.html">get_msigdb</a></span><span class="op">(</span>version <span class="op">=</span> <span class="st">"2023.2.Hs"</span>, collection <span class="op">=</span> <span class="st">"h.all"</span><span class="op">)</span></span>
<span><span class="va">lt</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $HALLMARK_ADIPOGENESIS</span></span>
<span><span class="co">##   [1] "19"     "11194"  "10449"  "33"     "34"     "35"     "47"     "50"    </span></span>
<span><span class="co">##   [9] "51"     "112"    "149685" "9370"   "79602"  "56894"  "9131"   "204"   </span></span>
<span><span class="co">##  [17] "217"    "226"    "284"    "51129"  "334"    "348"    "369"    "10124" </span></span>
<span><span class="co">##  [25] "64225"  "483"    "539"    "11176"  "593"    "23786"  "604"    "718"   </span></span>
<span><span class="co">##  [33] "847"    "284119" "8436"   "901"    "977"    "9936"   "948"    "1031"  </span></span>
<span><span class="co">##  [41] "400916" "1147"   "1149"   "134147" "51727"  "1306"   "1282"   "51805" </span></span>
<span><span class="co">##  [49] "84274"  "57017"  "1337"   "1349"   "1351"   "1376"   "1384"   "1431"  </span></span>
<span><span class="co">##  [57] "1537"   "1580"   "1629"   "1652"   "1666"   "8694"   "1717"   "51635" </span></span>
<span><span class="co">##  [65] "25979"  "1737"   "1738"   "4189"   "29103"  "128338" "1891"   "1892"  </span></span>
<span><span class="co">##  [73] "84173"  "79071"  "5168"   "2053"   "2101"   "23344"  "2109"   "2167"  </span></span>
<span><span class="co">##  [81] "2184"   "8322"   "9908"   "1647"   "2632"   "27069"  "57678"  "137964"</span></span>
<span><span class="co">##  [89] "2820"   "10243"  "2878"   "2879"   "80273"  "3033"   "26275"  "26353" </span></span>
<span><span class="co">##  [97] "3417"   "3419"   "3421"   "3459"   "10989"  "3679"   "80760"  "6453"  </span></span>
<span><span class="co">## [105] "84522"  "3910"   "3952"   "3977"   "3991"   "10162"  "4023"   "4056"  </span></span>
<span><span class="co">## [113] "8491"   "56922"  "4191"   "4199"   "11343"  "4259"   "84895"  "56246" </span></span>
<span><span class="co">## [121] "29088"  "54996"  "23788"  "4638"   "64859"  "4698"   "4706"   "4713"  </span></span>
<span><span class="co">## [129] "4722"   "28512"  "4836"   "4958"   "5004"   "27250"  "10400"  "5195"  </span></span>
<span><span class="co">## [137] "5209"   "5211"   "5236"   "23187"  "5264"   "415116" "123"    "5447"  </span></span>
<span><span class="co">## [145] "5468"   "5495"   "84919"  "10935"  "10113"  "55037"  "5733"   "5860"  </span></span>
<span><span class="co">## [153] "83871"  "7905"   "92840"  "56729"  "54884"  "8780"   "55177"  "26994" </span></span>
<span><span class="co">## [161] "6239"   "10313"  "25813"  "949"    "6342"   "6390"   "6391"   "6573"  </span></span>
<span><span class="co">## [169] "6510"   "6576"   "1468"   "376497" "8884"   "130814" "6623"   "6647"  </span></span>
<span><span class="co">## [177] "10580"  "65124"  "8404"   "58472"  "8082"   "6776"   "2040"   "8802"  </span></span>
<span><span class="co">## [185] "6817"   "6888"   "10010"  "7086"   "10140"  "7263"   "7316"   "29979" </span></span>
<span><span class="co">## [193] "83549"  "7351"   "29796"  "10975"  "7384"   "27089"  "7423"   "7532"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">##   [1] "16"     "6059"   "10006"  "43"     "92"     "207"    "322"    "567"   </span></span>
<span><span class="co">##   [9] "586"    "8915"   "602"    "672"    "717"    "822"    "9607"   "6356"  </span></span>
<span><span class="co">##  [17] "6357"   "6363"   "6347"   "6367"   "6351"   "6352"   "6354"   "894"   </span></span>
<span><span class="co">##  [25] "896"    "1230"   "729230" "1234"   "912"    "914"    "919"    "940"   </span></span>
<span><span class="co">##  [33] "915"    "916"    "917"    "920"    "958"    "959"    "961"    "924"   </span></span>
<span><span class="co">##  [41] "972"    "973"    "941"    "942"    "925"    "926"    "10225"  "1029"  </span></span>
<span><span class="co">##  [49] "5199"   "56253"  "1435"   "1445"   "1520"   "10563"  "4283"   "2833"  </span></span>
<span><span class="co">##  [57] "1615"   "8560"   "8444"   "1956"   "8661"   "8664"   "8669"   "8672"  </span></span>
<span><span class="co">##  [65] "1984"   "1991"   "2000"   "2069"   "2113"   "2147"   "2149"   "355"   </span></span>
<span><span class="co">##  [73] "356"    "2213"   "2268"   "2316"   "2533"   "2589"   "2634"   "2650"  </span></span>
<span><span class="co">##  [81] "11146"  "8477"   "3001"   "3002"   "3059"   "9734"   "3091"   "3105"  </span></span>
<span><span class="co">##  [89] "3108"   "3109"   "3111"   "3112"   "3117"   "3122"   "3133"   "3135"  </span></span>
<span><span class="co">##  [97] "3383"   "23308"  "3455"   "3458"   "3459"   "3460"   "10261"  "3551"  </span></span>
<span><span class="co">## [105] "3586"   "3589"   "3592"   "3593"   "3594"   "3596"   "3600"   "3603"  </span></span>
<span><span class="co">## [113] "3606"   "8807"   "3553"   "3558"   "9466"   "3559"   "3560"   "3561"  </span></span>
<span><span class="co">## [121] "3565"   "3566"   "3569"   "3574"   "3578"   "3624"   "3625"   "3662"  </span></span>
<span><span class="co">## [129] "3665"   "3394"   "3683"   "3689"   "3702"   "3717"   "3824"   "3848"  </span></span>
<span><span class="co">## [137] "3932"   "3937"   "3976"   "4050"   "4065"   "9450"   "4067"   "6885"  </span></span>
<span><span class="co">## [145] "11184"  "4153"   "4318"   "11222"  "4528"   "4689"   "4690"   "9437"  </span></span>
<span><span class="co">## [153] "114548" "4830"   "4843"   "4869"   "5196"   "5551"   "5579"   "5582"  </span></span>
<span><span class="co">## [161] "5699"   "5777"   "5788"   "5917"   "8767"   "6170"   "6123"   "6133"  </span></span>
<span><span class="co">## [169] "6223"   "6189"   "6203"   "27240"  "8651"   "9655"   "6688"   "5552"  </span></span>
<span><span class="co">## [177] "7903"   "23166"  "6772"   "6775"   "6890"   "6891"   "6892"   "7040"  </span></span>
<span><span class="co">## [185] "7042"   "7070"   "7076"   "7096"   "7097"   "7098"   "10333"  "7124"  </span></span>
<span><span class="co">## [193] "7163"   "7186"   "50852"  "7321"   "7334"   "7453"   "7454"   "7535"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="practice">Practice<a class="anchor" aria-label="anchor" href="#practice"></a>
</h2>
<div class="section level3 tabset">
<h3 class="tabset" id="section"></h3>
<div class="section level4">
<h4 id="practice-1">Practice 1<a class="anchor" aria-label="anchor" href="#practice-1"></a>
</h4>
<p>The gene set resource on <a href="https://maayanlab.cloud/Enrichr/#libraries" class="external-link uri">https://maayanlab.cloud/Enrichr/#libraries</a> is very useful. You may want to use it some say in the future. Take one gene set collection (e.g. the COVID-19 related gene sets), download the corresponding gene set file (in <code>.gmt</code> format) and try to read into R as a list or a two-column data frame.</p>
</div>
<div class="section level4">
<h4 id="solution">Solution<a class="anchor" aria-label="anchor" href="#solution"></a>
</h4>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&amp;libraryName=COVID-19_Related_Gene_Sets"</span>, destfile <span class="op">=</span> <span class="st">"covid-19.gmt"</span><span class="op">)</span></span>
<span><span class="va">ln</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readLines.html" class="external-link">readLines</a></span><span class="op">(</span><span class="st">"covid-19.gmt"</span><span class="op">)</span></span>
<span><span class="va">ln</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">ln</span>, <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">gs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">ln</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gs</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ln</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">gs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`COVID19-E protein host PPI from Krogan`</span></span>
<span><span class="co">## [1] "BRD4"    "BRD2"    "SLC44A2" "ZC3H18"  "AP3B1"   "CWC27"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`COVID19-M protein host PPI from Krogan`</span></span>
<span><span class="co">##  [1] "AAR2"     "AASS"     "SLC30A7"  "SLC30A9"  "INTS4"    "SAAL1"   </span></span>
<span><span class="co">##  [7] "ANO6"     "ATP1B1"   "YIF1A"    "REEP6"    "GGCX"     "REEP5"   </span></span>
<span><span class="co">## [13] "COQ8B"    "TARS2"    "FAM8A1"   "ATP6V1A"  "RTN4"     "TUBGCP2" </span></span>
<span><span class="co">## [19] "TUBGCP3"  "AKAP8L"   "FASTKD5"  "ETFA"     "BZW2"     "PSMD8"   </span></span>
<span><span class="co">## [25] "ACADM"    "PITRM1"   "STOM"     "PMPCB"    "PMPCA"    "SLC25A21"</span></span></code></pre>
<p>Convert <code>gs</code> to a data frame</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    geneset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gs</span><span class="op">)</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gs</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">gs</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                        geneset</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan1 COVID19-E protein host PPI from Krogan</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan2 COVID19-E protein host PPI from Krogan</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan3 COVID19-E protein host PPI from Krogan</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan4 COVID19-E protein host PPI from Krogan</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan5 COVID19-E protein host PPI from Krogan</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan6 COVID19-E protein host PPI from Krogan</span></span>
<span><span class="co">##                                            gene</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan1    BRD4</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan2    BRD2</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan3 SLC44A2</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan4  ZC3H18</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan5   AP3B1</span></span>
<span><span class="co">## COVID19-E protein host PPI from Krogan6   CWC27</span></span></code></pre>
<p>Or use the <code><a href="../reference/list_to_dataframe.html">list_to_dataframe()</a></code> function in <strong>GSEAtraining</strong>:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="../reference/list_to_dataframe.html">list_to_dataframe</a></span><span class="op">(</span><span class="va">gs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                 gene_set    gene</span></span>
<span><span class="co">## 1 COVID19-E protein host PPI from Krogan    BRD4</span></span>
<span><span class="co">## 2 COVID19-E protein host PPI from Krogan    BRD2</span></span>
<span><span class="co">## 3 COVID19-E protein host PPI from Krogan SLC44A2</span></span>
<span><span class="co">## 4 COVID19-E protein host PPI from Krogan  ZC3H18</span></span>
<span><span class="co">## 5 COVID19-E protein host PPI from Krogan   AP3B1</span></span>
<span><span class="co">## 6 COVID19-E protein host PPI from Krogan   CWC27</span></span></code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
